<!DOCTYPE html>
<!-- saved from url=(0043)http://www.iqtree.org/doc/Advanced-Tutorial -->
<html lang="en" class="csstransforms csstransforms3d csstransitions"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1"><meta name="description" content=""><meta name="author" content=""><title>Advanced Tutorial</title><link href="./Advanced Tutorial_files/bootstrap.min.css" rel="stylesheet"><link href="./Advanced Tutorial_files/style.css" rel="stylesheet"><link href="./Advanced Tutorial_files/font-awesome.min.css" rel="stylesheet"><link rel="stylesheet" href="./Advanced Tutorial_files/cookiealert.css"><script async="" src="./Advanced Tutorial_files/async-ads.js.download"></script><script async="" src="./Advanced Tutorial_files/analytics.js.download"></script><script src="./Advanced Tutorial_files/cse_element__en.js.download" type="text/javascript"></script><link type="text/css" rel="stylesheet" href="./Advanced Tutorial_files/default+en.css"><link type="text/css" rel="stylesheet" href="./Advanced Tutorial_files/default.css"><style type="text/css">.gsc-control-cse{font-family:times new roman, arial, serif}.gsc-control-cse .gsc-table-result{font-family:times new roman, arial, serif}.gsc-refinementsGradient{background:linear-gradient(to left,rgba(255,255,255,1),rgba(255,255,255,0))}.gsc-control-cse{border-color:#337AB7;background-color:#337AB7}input.gsc-input,.gsc-input-box,.gsc-input-box-hover,.gsc-input-box-focus{border-color:#FFFFFF}.gsc-search-button-v2,.gsc-search-button-v2:hover,.gsc-search-button-v2:focus{border-color:#4CAE4C;background-color:#5CB85C;background-image:none;filter:none}.gsc-search-button-v2 svg{fill:#FFFFFF}.gsc-tabHeader.gsc-tabhActive,.gsc-refinementHeader.gsc-refinementhActive{color:#CCCCCC;border-color:#CCCCCC;background-color:#FFFFFF}.gsc-tabHeader.gsc-tabhInactive,.gsc-refinementHeader.gsc-refinementhInactive{color:#CCCCCC;border-color:#CCCCCC;background-color:#FFFFFF}.gsc-webResult.gsc-result,.gsc-results .gsc-imageResult{border-color:#F5F5F5;background-color:#F5F5F5}.gsc-webResult.gsc-result:hover{border-color:#F5F5F5;background-color:#F5F5F5}.gs-webResult.gs-result a.gs-title:link,.gs-webResult.gs-result a.gs-title:link b,.gs-imageResult a.gs-title:link,.gs-imageResult a.gs-title:link b{color:#0000CC}.gs-webResult.gs-result a.gs-title:visited,.gs-webResult.gs-result a.gs-title:visited b,.gs-imageResult a.gs-title:visited,.gs-imageResult a.gs-title:visited b{color:#0000CC}.gs-webResult.gs-result a.gs-title:hover,.gs-webResult.gs-result a.gs-title:hover b,.gs-imageResult a.gs-title:hover,.gs-imageResult a.gs-title:hover b{color:#0000CC}.gs-webResult.gs-result a.gs-title:active,.gs-webResult.gs-result a.gs-title:active b,.gs-imageResult a.gs-title:active,.gs-imageResult a.gs-title:active b{color:#0000CC}.gsc-cursor-page{color:#0000CC}a.gsc-trailing-more-results:link{color:#0000CC}.gs-webResult:not(.gs-no-results-result):not(.gs-error-result) .gs-snippet,.gs-fileFormatType{color:#000000}.gs-webResult div.gs-visibleUrl{color:#008000}.gs-webResult div.gs-visibleUrl-short{color:#008000}.gsc-cursor-box{border-color:#F5F5F5}.gsc-results .gsc-cursor-box .gsc-cursor-page{border-color:#CCCCCC;background-color:#FFFFFF;color:#CCCCCC}.gsc-results .gsc-cursor-box .gsc-cursor-current-page{border-color:#CCCCCC;background-color:#FFFFFF;color:#CCCCCC}.gsc-webResult.gsc-result.gsc-promotion{border-color:#336699;background-color:#FFFFFF}.gsc-completion-title{color:#0000CC}.gsc-completion-snippet{color:#000000}.gs-promotion a.gs-title:link,.gs-promotion a.gs-title:link *,.gs-promotion .gs-snippet a:link{color:#0000CC}.gs-promotion a.gs-title:visited,.gs-promotion a.gs-title:visited *,.gs-promotion .gs-snippet a:visited{color:#0000CC}.gs-promotion a.gs-title:hover,.gs-promotion a.gs-title:hover *,.gs-promotion .gs-snippet a:hover{color:#0000CC}.gs-promotion a.gs-title:active,.gs-promotion a.gs-title:active *,.gs-promotion .gs-snippet a:active{color:#0000CC}.gs-promotion .gs-snippet,.gs-promotion .gs-title .gs-promotion-title-right,.gs-promotion .gs-title .gs-promotion-title-right *{color:#000000}.gs-promotion .gs-visibleUrl,.gs-promotion .gs-visibleUrl-short{color:#008000}.gcsc-find-more-on-google{color:#0000CC}.gcsc-find-more-on-google-magnifier{fill:#0000CC}</style><style type="text/css">.gscb_a{display:inline-block;font:27px/13px arial,sans-serif}.gsst_a .gscb_a{color:#a1b9ed;cursor:pointer}.gsst_a:hover .gscb_a,.gsst_a:focus .gscb_a{color:#36c}.gsst_a{display:inline-block}.gsst_a{cursor:pointer;padding:0 4px}.gsst_a:hover{text-decoration:none!important}.gsst_b{font-size:16px;padding:0 2px;position:relative;user-select:none;-webkit-user-select:none;white-space:nowrap}.gsst_e{vertical-align:middle;opacity:0.55;}.gsst_a:hover .gsst_e,.gsst_a:focus .gsst_e{opacity:0.72;}.gsst_a:active .gsst_e{opacity:1;}.gsst_f{background:white;text-align:left}.gsst_g{background-color:white;border:1px solid #ccc;border-top-color:#d9d9d9;box-shadow:0 2px 4px rgba(0,0,0,0.2);-webkit-box-shadow:0 2px 4px rgba(0,0,0,0.2);margin:-1px -3px;padding:0 6px}.gsst_h{background-color:white;height:1px;margin-bottom:-1px;position:relative;top:-1px}.gsib_a{width:100%;padding:4px 6px 0}.gsib_a,.gsib_b{vertical-align:top}.gssb_c{border:0;position:absolute;z-index:989}.gssb_e{border:1px solid #ccc;border-top-color:#d9d9d9;box-shadow:0 2px 4px rgba(0,0,0,0.2);-webkit-box-shadow:0 2px 4px rgba(0,0,0,0.2);cursor:default}.gssb_f{visibility:hidden;white-space:nowrap}.gssb_k{border:0;display:block;position:absolute;top:0;z-index:988}.gsdd_a{border:none!important}.gsq_a{padding:0}.gssb_a{padding:0 7px}.gssb_a,.gssb_a td{white-space:nowrap;overflow:hidden;line-height:22px}#gssb_b{font-size:11px;color:#36c;text-decoration:none}#gssb_b:hover{font-size:11px;color:#36c;text-decoration:underline}.gssb_g{text-align:center;padding:8px 0 7px;position:relative}.gssb_h{font-size:15px;height:28px;margin:0.2em;-webkit-appearance:button}.gssb_i{background:#eee}.gss_ifl{visibility:hidden;padding-left:5px}.gssb_i .gss_ifl{visibility:visible}a.gssb_j{font-size:13px;color:#36c;text-decoration:none;line-height:100%}a.gssb_j:hover{text-decoration:underline}.gssb_l{height:1px;background-color:#e5e5e5}.gssb_m{color:#000;background:#fff}.gssb_a{padding:0 9px}.gsib_a{padding:5px 9px 4px 9px}.gscb_a{line-height:27px}.gssb_e{border:0}.gssb_l{margin:5px 0}input.gsc-input::-webkit-input-placeholder{font-size:14px}input.gsc-input:-moz-placeholder{font-size:14px}input.gsc-input::-moz-placeholder{font-size:14px}input.gsc-input:-ms-input-placeholder{font-size:14px}input.gsc-input:focus::-webkit-input-placeholder{color:transparent}input.gsc-input:focus:-moz-placeholder{color:transparent}input.gsc-input:focus::-moz-placeholder{color:transparent}input.gsc-input:focus:-ms-input-placeholder{color:transparent}.gssb_c .gsc-completion-container{position:static}.gssb_c{z-index:5000}.gsc-completion-container table{background:transparent;font-size:inherit;font-family:inherit}.gssb_c > tbody > tr,.gssb_c > tbody > tr > td,.gssb_d,.gssb_d > tbody > tr,.gssb_d > tbody > tr > td,.gssb_e,.gssb_e > tbody > tr,.gssb_e > tbody > tr > td{padding:0;margin:0;border:0}.gssb_a table,.gssb_a table tr,.gssb_a table tr td{padding:0;margin:0;border:0}</style></head><body style=""><div class="navbar navbar-default navbar-fixed-top" role="navigation"><div class="container"><div class="navbar-header"> <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar"></span> <span class="icon-bar"></span> <span class="icon-bar"></span> </button> <a class="navbar-brand" href="http://www.iqtree.org/">IQ-TREE</a></div><div class="navbar-collapse collapse navbar-right"><ul class="nav navbar-nav"><li><a data-toggle="collapse" data-target=".navbar-collapse" href="http://www.iqtree.org/#download">Download</a></li><li><a href="http://www.iqtree.org/news/">News</a></li><li class="dropdown"> <a class="dropdown-toggle" data-toggle="dropdown" href="http://www.iqtree.org/doc/Advanced-Tutorial#">Web server <span class="caret"></span></a><ul class="dropdown-menu"><li><a href="http://iqtree.cibiv.univie.ac.at/" target="_blank">CIBIV, Austria</a></li><li><a href="https://www.hiv.lanl.gov/content/sequence/IQTREE/iqtree.html" target="_blank">Los Alamos Lab, USA</a></li><li><a href="https://www.phylo.org/" target="_blank">CIPRES Gateway, USA</a></li></ul></li><li><a href="http://www.iqtree.org/doc/">Docs</a></li><li><a href="http://www.iqtree.org/workshop/">Workshop</a></li><li><a href="http://www.iqtree.org/about/">About</a></li></ul></div></div></div><div id="blue"><div class="container"><div class="row"><div class="col-sm-6"><h3> Tutorial</h3></div><div class="col-sm-6"> <script type="text/javascript" async="" src="./Advanced Tutorial_files/f.txt"></script><script> (function() { var cx = '003669294240048101214:pace-yz0kou'; var gcse = document.createElement('script'); gcse.type = 'text/javascript'; gcse.async = true; gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') + '//cse.google.com/cse.js?cx=' + cx; var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(gcse, s); })(); </script> <div id="___gcse_0"><div class="gsc-control-cse gsc-control-cse-en"><div class="gsc-control-wrapper-cse" dir="ltr"><form class="gsc-search-box gsc-search-box-tools" accept-charset="utf-8"><table cellspacing="0" cellpadding="0" class="gsc-search-box"><tbody><tr><td class="gsc-input"><div class="gsc-input-box" id="gsc-iw-id1"><table cellspacing="0" cellpadding="0" id="gs_id50" class="gstl_50 gsc-input" style="width: 100%; padding: 0px;"><tbody><tr><td id="gs_tti50" class="gsib_a"><input autocomplete="off" type="text" size="10" class="gsc-input" name="search" title="search" id="gsc-i-id1" dir="ltr" spellcheck="false" style="width: 100%; padding: 0px; border: none; margin: 0px; height: auto; background: url(&quot;https://www.google.com/cse/static/images/1x/en/branding.png&quot;) left center no-repeat rgb(255, 255, 255); outline: none;"></td><td class="gsib_b"><div class="gsst_b" id="gs_st50" dir="ltr"><a class="gsst_a" href="javascript:void(0)" style="display: none;" title="Clear search box" role="button"><span class="gscb_a" id="gs_cb50" aria-hidden="true">×</span></a></div></td></tr></tbody></table></div></td><td class="gsc-search-button"><button class="gsc-search-button gsc-search-button-v2"><svg width="13" height="13" viewBox="0 0 13 13"><title>search</title><path d="m4.8495 7.8226c0.82666 0 1.5262-0.29146 2.0985-0.87438 0.57232-0.58292 0.86378-1.2877 0.87438-2.1144 0.010599-0.82666-0.28086-1.5262-0.87438-2.0985-0.59352-0.57232-1.293-0.86378-2.0985-0.87438-0.8055-0.010599-1.5103 0.28086-2.1144 0.87438-0.60414 0.59352-0.8956 1.293-0.87438 2.0985 0.021197 0.8055 0.31266 1.5103 0.87438 2.1144 0.56172 0.60414 1.2665 0.8956 2.1144 0.87438zm4.4695 0.2115 3.681 3.6819-1.259 1.284-3.6817-3.7 0.0019784-0.69479-0.090043-0.098846c-0.87973 0.76087-1.92 1.1413-3.1207 1.1413-1.3553 0-2.5025-0.46363-3.4417-1.3909s-1.4088-2.0686-1.4088-3.4239c0-1.3553 0.4696-2.4966 1.4088-3.4239 0.9392-0.92727 2.0864-1.3969 3.4417-1.4088 1.3553-0.011889 2.4906 0.45771 3.406 1.4088 0.9154 0.95107 1.379 2.0924 1.3909 3.4239 0 1.2126-0.38043 2.2588-1.1413 3.1385l0.098834 0.090049z"></path></svg></button></td><td class="gsc-clear-button"><div class="gsc-clear-button" title="clear results">&nbsp;</div></td></tr></tbody></table></form><div class="gsc-results-wrapper-overlay"><div class="gsc-results-close-btn" tabindex="0"></div><div class="gsc-positioningWrapper"><div class="gsc-tabsAreaInvisible"><div aria-label="refinement" role="tab" class="gsc-tabHeader gsc-inline-block gsc-tabhActive">Custom Search</div><span class="gs-spacer"> </span></div></div><div class="gsc-positioningWrapper"><div class="gsc-tabsAreaInvisible"></div></div><div class="gsc-above-wrapper-area-invisible"><table cellspacing="0" cellpadding="0" class="gsc-above-wrapper-area-container"><tbody><tr><td class="gsc-result-info-container"><div class="gsc-result-info-invisible"></div></td><td class="gsc-orderby-container"><div class="gsc-orderby-invisible"><div class="gsc-orderby-label gsc-inline-block">Sort by:</div><div class="gsc-option-menu-container gsc-inline-block"><div class="gsc-selected-option-container gsc-inline-block"><div class="gsc-selected-option">Relevance</div><div class="gsc-option-selector"></div></div><div class="gsc-option-menu-invisible"><div class="gsc-option-menu-item gsc-option-menu-item-highlighted"><div class="gsc-option">Relevance</div></div><div class="gsc-option-menu-item"><div class="gsc-option">Date</div></div></div></div></div></td></tr></tbody></table></div><div class="gsc-adBlockInvisible"></div><div class="gsc-wrapper"><div class="gsc-adBlockInvisible"></div><div class="gsc-resultsbox-invisible"><div class="gsc-resultsRoot gsc-tabData gsc-tabdActive"><div><div class="gsc-expansionArea"></div></div></div></div></div></div><div class="gsc-modal-background-image" tabindex="0"></div></div></div></div></div></div></div></div><div class="container mtb"><div class="row"><div class="col-md-9"><p><em>Last update: May 19, 2021, Contributors: Jana, Minh Bui</em></p><h1 id="advanced-tutorial">Advanced tutorial</h1><p>Recommended for experienced users to explore more features.</p><p>To get started, please read the <a href="http://www.iqtree.org/doc/Tutorial">Beginner’s Tutorial</a> first if not done so yet.</p><h2 id="partitioned-analysis-for-multi-gene-alignments">Partitioned analysis for multi-gene alignments</h2><div class="hline"></div><p>If you used partition model in a publication please cite:</p><blockquote><p><strong>O. Chernomor, A. von Haeseler, and B.Q. Minh</strong> (2016) Terrace aware data structure for phylogenomic inference from supermatrices. <em>Syst. Biol.</em>, 65:997-1008. <a href="https://doi.org/10.1093/sysbio/syw037">https://doi.org/10.1093/sysbio/syw037</a></p></blockquote><p>In the partition model, you can specify a substitution model for each gene/character set. IQ-TREE will then estimate the model parameters separately for every partition. Moreover, IQ-TREE provides edge-linked or edge-unlinked branch lengths between partitions:</p><ul><li><code>-q partition_file</code>: all partitions share the same set of branch lengths (like <code>-q</code> option of RAxML).</li><li><code>-p partition_file</code> (<code>-spp</code> in version 1.x): like above but allowing each partition to have its own evolution rate.</li><li><code>-Q partition_file</code> (<code>-sp</code> in version 1.x): each partition has its own set of branch lengths (like combination of <code>-q</code> and <code>-M</code> options in RAxML) to account for, e.g. <em>heterotachy</em> (<a href="http://mbe.oxfordjournals.org/content/19/1/1.full">Lopez et al., 2002</a>).</li></ul><blockquote><p><strong>NOTE</strong>: <code>-p</code> is recommended for typical analysis. <code>-q</code> is unrealistic and <code>-Q</code> is very parameter-rich. One can also perform all three analyses and compare e.g. the BIC scores to determine the best-fit partition model.</p></blockquote><p>IQ-TREE supports RAxML-style and NEXUS partition input file. The RAxML-style partition file may look like:</p><pre><code>DNA, part1 = 1-100
DNA, part2 = 101-384
</code></pre><p>If your partition file is called <code>example.partitions</code>, the partition analysis can be run with:</p><pre><code>iqtree -s example.phy -p example.partitions -m GTR+I+G
# for version 1.x change -p to -spp
</code></pre><p>Note that using RAxML-style partition file, all partitions will use the same rate heterogeneity model given in <code>-m</code> option (<code>+I+G</code> in this example). If you want to specify, say, <code>+G</code> for the first partition and <code>+I+G</code> for the second partition, then you need to create the more flexible NEXUS partition file. This file contains a <code>SETS</code> block with <code>CharSet</code> and <code>CharPartition</code> commands to specify individual genes and the partition, respectively. For example:</p><pre><code>#nexus
begin sets;
    charset part1 = 1-100;
    charset part2 = 101-384;
    charpartition mine = HKY+G:part1, GTR+I+G:part2;
end;
</code></pre><p>If your NEXUS file is called <code>example.nex</code>, then you can use the option <code>-p</code> to input the file as following:</p><pre><code>iqtree -s example.phy -p example.nex
# for version 1.x change -p to -spp
</code></pre><p>Here, IQ-TREE partitions the alignment <code>example.phy</code> into 2 sub-alignments named <code>part1</code> and <code>part2</code> containing sites (columns) 1-100 and 101-384, respectively. Moreover, IQ-TREE applies the subtitution models <code>HKY+G</code> and <code>GTR+I+G</code> to <code>part1</code> and <code>part2</code>, respectively. Substitution model parameters and trees with branch lengths can be found in the result file <code>example.nex.iqtree</code>.</p><p>Moreover, the <code>CharSet</code> command allows to specify non-consecutive sites with e.g.:</p><pre><code>charset part1 = 1-100 200-384;
</code></pre><p>That means, <code>part1</code> contains sites 1-100 and 200-384 of the alignment. Another example is:</p><pre><code>charset part1 = 1-100\3;
</code></pre><p>for extracting sites 1,4,7,…,100 from the alignment. This is useful for getting codon positions from the protein-coding alignment.</p><h2 id="partitioned-analysis-with-mixed-data">Partitioned analysis with mixed data</h2><div class="hline"></div><p>IQ-TREE also allows combining sub-alignments from different alignment files, which is helpful if you want to combine mixed data (e.g. DNA and protein) in a single analysis. Here is an example for mixing DNA, protein and codon models:</p><pre><code>#nexus
begin sets;
    charset part1 = dna.phy: 1-100 201-300;
    charset part2 = dna.phy: 101-200;
    charset part3 = prot.phy: 1-400;
    charset part4 = prot.phy: 401-600;
    charset part5 = codon.phy:CODON, *;
    charpartition mine = HKY:part1, GTR+G:part2, LG+G:part3, WAG+I+G:part4, GY:part5;
end;
</code></pre><p>Here, <code>part1</code> and <code>part2</code> contain sub-alignments from alignment file <code>dna.phy</code>, whereas <code>part3</code> and <code>part4</code> are loaded from alignment file <code>prot.phy</code> and <code>part5</code> from <code>codon.phy</code>. The <code>:</code> is needed to separate the alignment file name and site specification. Note that, for convenience <code>*</code> in <code>part5</code> specification means that <code>part5</code> corresponds to the entire alignment <code>codon.phy</code>.</p><blockquote class="tip"><p><strong>TIP</strong>: For <code>part5</code> the <code>CODON</code> keyword is specified so that IQ-TREE will apply a codon model. Moreover, this implicitly assumes the standard genetic code. If you want to use another genetic code, append <code>CODON</code> with the <a href="http://www.iqtree.org/doc/Substitution-Models#codon-models">code ID described here</a></p></blockquote><p>Because the alignment file names are now specified in this NEXUS file, you can omit the <code>-s</code> option:</p><pre><code>iqtree -p example.nex
# for version 1.x change -p to -spp
</code></pre><p>Note that <code>aln.phy</code> and <code>prot.phy</code> does not need to contain the same set of sequences. For instance, if some sequence occurs in <code>aln.phy</code> but not in <code>prot.phy</code>, IQ-TREE will treat the corresponding parts of sequence in <code>prot.phy</code> as missing data. For your convenience IQ-TREE writes the concatenated alignment into the file <code>example.nex.conaln</code>.</p><h2 id="choosing-the-right-partitioning-scheme">Choosing the right partitioning scheme</h2><div class="hline"></div><p>ModelFinder implements a greedy strategy (<a href="https://doi.org/10.1093/molbev/mss020">Lanfear et al., 2012</a>) that starts with the full partition model and subsequentially merges two genes until the model fit does not increase any further:</p><pre><code>iqtree -s example.phy -p example.nex -m MFP+MERGE
# for version 1.x change -p to -spp
</code></pre><p>Note that this command considers the FreeRate heterogeneity model (see <a href="http://www.iqtree.org/doc/Tutorial#choosing-the-right-substitution-model">model selection tutorial</a>). If you want to resemble PartitionFinder by just considering the invariable site and Gamma rate heterogeneity (thus saving computation times), then run:</p><pre><code>iqtree -s example.phy -p example.nex -m TESTMERGE
# for version 1.x change -p to -spp
</code></pre><p>After ModelFinder found the best partition, IQ-TREE will immediately start the tree reconstruction under the best-fit partition model. Sometimes you only want to find the best-fit partition model without doing tree reconstruction, then run:</p><pre><code>iqtree -s example.phy -p example.nex -m MF+MERGE
# for version 1.x change -p to -spp
</code></pre><p>To resemble PartitionFinder and save time:</p><pre><code>iqtree -s example.phy -p example.nex -m TESTMERGEONLY
# for version 1.x change -p to -spp
</code></pre><p>To reduce the computational burden IQ-TREE implements the <em>relaxed hierarchical clustering algorithm</em> (<a href="https://doi.org/10.1186/1471-2148-14-82">Lanfear et al., 2014</a>), which is invoked via <code>-rcluster</code> option:</p><pre><code>iqtree -s example.phy -p example.nex -m MF+MERGE -rcluster 10
# for version 1.x change -p to -spp
</code></pre><p>to only examine the top 10% partition merging schemes (similar to the <code>--rcluster-percent 10</code> option in PartitionFinder).</p><h2 id="ultrafast-bootstrapping-with-partition-model">Ultrafast bootstrapping with partition model</h2><div class="hline"></div><p>IQ-TREE can perform the ultrafast bootstrap with partition models by e.g.,</p><pre><code>iqtree -s example.phy -p example.nex -B 1000
# for version 1.x change -p to -spp and -B to -bb
</code></pre><p>Here, IQ-TREE will resample the sites <em>within</em> partitions (i.e., the bootstrap replicates are generated per partition separately and then concatenated together). The same holds true if you do the standard nonparametric bootstrap.</p><p>IQ-TREE supports the partition-resampling strategy as suggested by (<a href="https://doi.org/10.1073/pnas.051611498">Nei et al., 2001</a>):</p><pre><code>iqtree -s example.phy -p example.nex -B 1000 --sampling GENE
# for version 1.x change -p to -spp and -B to -bb and --sampling to -bsam
</code></pre><p>to resample partitions instead of sites. Moreover, IQ-TREE allows an even more complicated strategy: resampling partitions and then sites within resampled partitions (<a href="https://doi.org/10.1002/jez.b.21026">Gadagkar et al., 2005</a>; <a href="https://doi.org/10.1073/pnas.0408313102">Seo et al., 2005</a>). This may help to reduce false positives (i.e. wrong branch receiving 100% support):</p><pre><code>iqtree -s example.phy -p example.nex -B 1000 --sampling GENESITE
# for version 1.x change -p to -spp and -B to -bb and --sampling to -bsam
</code></pre><h2 id="constrained-tree-search">Constrained tree search</h2><div class="hline"></div><p>IQ-TREE supports constrained tree search via <code>-g</code> option, so that the resulting tree must obey a constraint tree topology. The constraint tree can be multifurcating and need not to contain all species. To illustrate, let’s return to the <a href="http://www.iqtree.org/doc/Tutorial#first-running-example">first running example</a>, where we want to force Human grouping with Seal whereas Cow with Whale. If you use the following constraint tree (NEWICK format):</p><pre><code>((Human,Seal),(Cow,Whale));
</code></pre><p>Save this to a file <code>example.constr0</code> and run:</p><pre><code>iqtree -s example.phy -m TIM2+I+G -g example.constr0 --prefix example.constr0
# for version 1.x change --prefix to -pre
</code></pre><p>(We use a prefix in order not to overwrite output files of the previous run). The resulting part of the tree extracted from <code>example.constr0.iqtree</code> looks exactly like a normal unconstrained tree search:</p><pre><code>        +--------------Human
     +--|
     |  |  +------Seal
     |  +--|
     |     |  +-----Cow
     |     +--|
     |        +-------Whale
+----|
|    |         +---Mouse
|    +---------|
|              +------Rat
</code></pre><p>This is the correct behavior: although Human and Seal are not monophyletic, this tree indeed satisfies the constraint, because the induced subtree separates (Human,Seal) from (Cow,Whale). This comes from the fact that the tree is <em>unrooted</em>. If you want them to be sister groups, then you need to include <em>outgroup</em> taxa into the constraint tree. For example:</p><pre><code>((Human,Seal),(Cow,Whale),Mouse);
</code></pre><p>Save this to <code>example.constr1</code> and run:</p><pre><code>iqtree -s example.phy -m TIM2+I+G -g example.constr1 --prefix example.constr1
# for version 1.x change --prefix to -pre
</code></pre><p>The resulting part of the tree is then:</p><pre><code>           +---------------Human
        +--|
        |  +------Seal
     +--|
     |  |  +-----Cow
     |  +--|
     |     +-------Whale
+----|
|    |         +---Mouse
|    +---------|
|              +------Rat
</code></pre><p>which shows the desired effect.</p><blockquote><p><strong>NOTE</strong>: While this option helps to enforce the tree based on prior knowledge, it is advised to always perform tree topology tests to make sure that the resulting constrained tree is NOT significantly worse than an unconstrained tree! See <a href="http://www.iqtree.org/doc/Advanced-Tutorial#tree-topology-tests">tree topology tests</a> and <a href="http://www.iqtree.org/doc/Advanced-Tutorial#testing-constrained-tree">testing constrained tree</a> below for a guide how to check this.</p></blockquote><h2 id="tree-topology-tests">Tree topology tests</h2><div class="hline"></div><p>IQ-TREE can compute log-likelihoods of a set of trees passed via the <code>-z</code> option:</p><pre><code>iqtree -s example.phy -z example.treels -m GTR+G
</code></pre><p>assuming that <code>example.treels</code> is an existing file containing a set of trees in NEWICK format. IQ-TREE first reconstructs an ML tree. Then, it will compute the log-likelihood of the trees in <code>example.treels</code> based on the estimated parameters done for the ML tree. <code>example.phy.iqtree</code> will have a section called <code>USER TREES</code> that lists the tree IDs and the corresponding log-likelihoods. The trees with optimized branch lengths can be found in <code>example.phy.treels.trees</code> If you only want to evaluate the trees without reconstructing the ML tree, you can run:</p><pre><code>iqtree -s example.phy -z example.treels -n 0
</code></pre><p>Here, the number of search iterations is set to 0 (<code>-n 0</code>), such that model parameters are quickly estimated from an initial parsimony tree, which is normally accurate enough for our purpose. If you, however, prefer to estimate model parameters based on a tree (e.g. reconstructed previously), use <code>-te &lt;treefile&gt;</code> option.</p><p>IQ-TREE also supports several tree topology tests using the RELL approximation (<a href="https://doi.org/10.1007/BF02109483">Kishino et al., 1990</a>). This includes bootstrap proportion (BP), Kishino-Hasegawa test (<a href="https://doi.org/10.1007/BF02100115">Kishino and Hasegawa, 1989</a>), Shimodaira-Hasegawa test (<a href="https://doi.org/10.1093/oxfordjournals.molbev.a026201">Shimodaira and Hasegawa, 1999</a>), expected likelihood weights (<a href="https://doi.org/10.1098/rspb.2001.1862">Strimmer and Rambaut, 2002</a>):</p><pre><code>iqtree -s example.phy -z example.treels -n 0 -zb 1000
</code></pre><p>Here, <code>-zb</code> specifies the number of RELL replicates, where 1000 is the recommended minimum number. The <code>USER TREES</code> section of <code>example.phy.iqtree</code> will list the results of BP, KH, SH, and ELW methods.</p><p>If you also want to perform the weighted KH and weighted SH tests, simply add <code>-zw</code> option:</p><pre><code>iqtree -s example.phy -z example.treels -n 0 -zb 1000 -zw
</code></pre><p>Starting with version 1.4.0 IQ-TREE supports approximately unbiased (AU) test (<a href="https://doi.org/10.1080/10635150290069913">Shimodaira, 2002</a>) via <code>-au</code> option:</p><pre><code>iqtree -s example.phy -z example.treels -n 0 -zb 1000 -zw -au
</code></pre><p>This will perform all above tests plus the AU test.</p><p>Finally, note that IQ-TREE will automatically detect duplicated tree topologies and omit them during the evaluation.</p><blockquote class="tip"><p><strong>HINTS</strong>:</p><ul><li><p>The KH, SH and AU tests return p-values, thus a tree is rejected if its p-value &lt; 0.05 (marked with a <code>-</code> sign).</p></li><li><p>bp-RELL and c-ELW return posterior weights which <em>are not p-value</em>. The weights sum up to 1 across the trees tested.</p></li><li><p>The KH test (<a href="https://doi.org/10.1007/BF02100115">Kishino and Hasegawa, 1989</a>) was designed to test 2 trees and thus has no correction for multiple testing. The SH test (<a href="https://doi.org/10.1093/oxfordjournals.molbev.a026201">Shimodaira and Hasegawa, 1999</a>) fixes this problem.</p></li><li><p>However, the SH test becomes too conservative (i.e., rejecting fewer trees than expected) when testing many trees. The AU test (<a href="https://doi.org/10.1080/10635150290069913">Shimodaira, 2002</a>) fixes this problem and is thus recommended as replacement for both KH and SH tests.</p></li></ul></blockquote><h2 id="testing-constrained-tree">Testing constrained tree</h2><div class="hline"></div><p>We now illustrate an example to use the AU test (see above) to test trees from unconstrained versus constrained search, which is helpful to know if a constrained search is sensible or not. Thus:</p><ol><li><p>Perform an unconstrained search:</p><pre><code> iqtree -s example.phy -m TIM2+I+G --prefix example.unconstr
 # for version 1.x change --prefix to -pre
</code></pre></li><li><p>Perform a constrained search, where <code>example.constr1</code> file contains: <code>((Human,Seal),(Cow,Whale),Mouse);</code>:</p><pre><code> iqtree -s example.phy -m TIM2+I+G -g example.constr1 --prefix example.constr1
 # for version 1.x change --prefix to -pre
</code></pre></li><li><p>Perform another constrained search, where <code>example.constr2</code> file contains <code>((Human,Cow,Whale),Seal,Mouse);</code>:</p><pre><code> iqtree -s example.phy -m TIM2+I+G -g example.constr2 --prefix example.constr2
 # for version 1.x change --prefix to -pre
</code></pre></li><li><p>Perform the last constrained search, where <code>example.constr3</code> file contains <code>((Human,Mouse),(Cow,Rat),Opossum);</code>:</p><pre><code> iqtree -s example.phy -m TIM2+I+G -g example.constr3 --prefix example.constr3
 # for version 1.x change --prefix to -pre
</code></pre></li><li><p>Concatenate all trees into a file:</p><pre><code> # for Linux or macOS
 cat example.unconstr.treefile example.constr1.treefile example.constr2.treefile example.constr3.treefile &gt; example.treels
    
 # for Windows
 type example.unconstr.treefile example.constr1.treefile example.constr2.treefile example.constr3.treefile &gt; example.treels
</code></pre></li><li><p>Test the set of trees:</p><pre><code> iqtree -s example.phy -m TIM2+I+G -z example.treels -n 0 -zb 1000 -au
</code></pre></li></ol><p>Now look at the resulting <code>.iqtree</code> file:</p><pre><code>USER TREES
----------

See example.phy.trees for trees with branch lengths.

Tree      logL    deltaL  bp-RELL    p-KH     p-SH    c-ELW     p-AU
-------------------------------------------------------------------------
  1   -21152.617   0.000  0.7110 + 0.7400 + 1.0000 + 0.6954 + 0.7939 + 
  2   -21156.802   4.185  0.2220 + 0.2600 + 0.5910 + 0.2288 + 0.3079 + 
  3   -21158.579   5.962  0.0670 + 0.1330 + 0.5130 + 0.0758 + 0.1452 + 
  4   -21339.596 186.980  0.0000 - 0.0000 - 0.0000 - 0.0000 - 0.0000 - 

deltaL  : logL difference from the maximal logl in the set.
bp-RELL : bootstrap proportion using RELL method (Kishino et al. 1990).
p-KH    : p-value of one sided Kishino-Hasegawa test (1989).
p-SH    : p-value of Shimodaira-Hasegawa test (2000).
c-ELW   : Expected Likelihood Weight (Strimmer &amp; Rambaut 2002).
p-AU    : p-value of approximately unbiased (AU) test (Shimodaira, 2002).

Plus signs denote the 95% confidence sets.
Minus signs denote significant exclusion.
All tests performed 1000 resamplings using the RELL method.
</code></pre><p>One sees that the AU test does not reject the first 3 trees (denoted by <code>+</code> sign below the <code>p-AU</code> column), whereas the last tree is significantly excluded (<code>-</code> sign). All other tests also agree with this. Therefore, groupings of <code>(Human,Mouse)</code> and <code>(Cow,Rat)</code> do not make sense. Whereas the phylogenetic position of <code>Seal</code> based on 3 first trees is still undecidable. This is in agreement with the SH-aLRT and ultrafast bootstrap supports <a href="http://www.iqtree.org/doc/Tutorial#assessing-branch-supports-with-single-branch-tests">done in the Tutorial</a>.</p><h2 id="consensus-construction-and-bootstrap-value-assignment">Consensus construction and bootstrap value assignment</h2><div class="hline"></div><p>IQ-TREE can construct an extended majority-rule consensus tree from a set of trees written in NEWICK or NEXUS format (e.g., produced by MrBayes):</p><pre><code>iqtree -con mytrees
</code></pre><p>To build a majority-rule consensus tree, simply set the minimum support threshold to 0.5:</p><pre><code>iqtree -con mytrees -minsup 0.5
</code></pre><p>If you want to specify a burn-in (the number of beginning trees to ignore from the trees file), use <code>-bi</code> option:</p><pre><code>iqtree -con mytrees -minsup 0.5 -bi 100
</code></pre><p>to skip the first 100 trees in the file.</p><p>IQ-TREE can also compute a consensus network and print it into a NEXUS file by:</p><pre><code>iqtree -net mytrees
</code></pre><p>Finally, a useful feature is to read in an input tree and a set of trees, then IQ-TREE can assign the support value onto the input tree (number of times each branch in the input tree occurs in the set of trees). This option is useful if you want to compute the support values for an ML tree based on alternative topologies.</p><pre><code>iqtree -sup input_tree set_of_trees
</code></pre><h2 id="user-defined-substitution-models">User-defined substitution models</h2><div class="hline"></div><p>Users can specify any DNA model using a 6-letter code that defines which rates should be equal. For example, <code>010010</code> corresponds to the HKY model and <code>012345</code> to the GTR model. In fact, IQ-TREE uses this specification internally to simplify the coding. The 6-letter code is specified via the <code>-m</code> option, e.g.:</p><pre><code>iqtree -s example.phy -m 010010+G
</code></pre><p>Moreover, with the <code>-m</code> option one can input a file which contains the 6 rates (A-C, A-G, A-T, C-G, C-T, G-T) and 4 base frequencies (A, C, G, T). For example:</p><pre><code>iqtree -s example.phy -m mymodel+G
</code></pre><p>where <code>mymodel</code> is a file containing the 10 entries described above, in the correct order. The entries can be seperated by either empty space(s) or newline character. One can even specify the rates within <code>-m</code> option by e.g.:</p><pre><code>iqtree -s example.phy -m 'TN{2.0,3.0}+G8{0.5}+I{0.15}'
</code></pre><p>That means, we use Tamura-Nei model with fixed transition-transversion rate ratio of 2.0 and purine/pyrimidine rate ratio of 3.0. Moreover, we use 8-category Gamma-distributed site rates with the shape parameter (alpha) equal to 0.5 and a proportion of invariable sites p-inv=0.15.</p><p>By default IQ-TREE computes empirical state frequencies from the alignment by counting, but one can also estimate the frequencies by maximum-likelihood with <code>+Fo</code> in the model name:</p><pre><code>iqtree -s example.phy -m GTR+G+Fo
</code></pre><p>For amino-acid alignments, IQ-TREE use the empirical frequencies specified in the model. If you want frequencies as counted from the alignment, use <code>+F</code>, for example:</p><pre><code>iqtree -s myprotein_alignment -m WAG+G+F
</code></pre><p>Note that all model specifications above can be used in the partition model NEXUS file.</p><h2 id="inferring-site-specific-rates">Inferring site-specific rates</h2><div class="hline"></div><p>IQ-TREE allows to infer site-specific evolutionary rates if a <a href="http://www.iqtree.org/doc/Substitution-Models#rate-heterogeneity-across-sites">site-rate heterogeneity model such as Gamma or FreeRate</a> is the best model. Here, IQ-TREE will estimate model parameters and then apply an empirical Bayesian approach to assign site-rates as the mean over rate categories, weighted by the posterior probability of the site falling into each category. This approach is provided in IQ-TREE because such empirical Bayesian approach was shown to be most accurate (<a href="https://doi.org/10.1093/molbev/msh194">Mayrose et al., 2004</a>). An example run:</p><pre><code>iqtree -s example.phy --rate
# for version 1.x change --rate to -wsr
</code></pre><p>IQ-TREE will write an output file <code>example.phy.rate</code> that looks like:</p><pre><code>Site    Rate    Category        Categorized_rate
1       0.26625 2       0.24393
2       0.99345 3       0.81124
3       2.69275 4       2.91367
4       0.25822 2       0.24393
5       0.25822 2       0.24393
6       0.42589 2       0.24393
7       0.30194 2       0.24393
8       0.72790 3       0.81124
9       0.25822 2       0.24393
10      0.09177 1       0.03116
</code></pre><p>The 1st column is site index of the alignment (starting from 1), the 2nd column <code>Rate</code> shows the mean site-specific rate as explained above, and the 3rd and 4th columns show the category index and rate of the Gamma rate category with the highest probability for this site (1 for slow and 4 for fast rate).</p><p>The above run will perform a full tree search. To speed up you can use <code>-n 0</code> to only use a parsimony tree for site rate estimates. Or if you have already infered an ML tree, you can specify it to improve the rate estimate:</p><pre><code>iqtree -s example.phy -t ml.treefile -n 0 --rate 
# for version 1.x change --rate to -wsr
</code></pre><p>where <code>-t</code> is the option to input a fixed tree topology and <code>ml.treefile</code> is the ML tree reconstructed previously.</p><p>If you already know the best-fit model for the alignment, you can use specify it via <code>-m</code> option to omit model selection and hence speed it up:</p><pre><code>iqtree -s example.phy -m GTR+R10 -n 0 --rate 
# for version 1.x change --rate to -wsr
</code></pre><p>Finally, IQ-TREE 2 allows to estimate rates by maximum likelihood via <code>--mlrate</code> option:</p><pre><code>iqtree -s example.phy -n 0 --mlrate 
</code></pre><p>This will print an output file <code>example.phy.mlrate</code> that looks like:</p><pre><code># Site-specific subtitution rates determined by maximum likelihood
# This file can be read in MS Excel or in R with command:
#   tab=read.table('example.phy.mlrate',header=TRUE)
# Columns are tab-separated with following meaning:
#   Site:   Alignment site ID
#   Rate:   Site rate estimated by maximum likelihood
Site    Rate
1       2.51550
2       12.89129
3       34.31350
4       2.44313
5       2.44313
6       4.41889
7       2.69577
8       9.27503
9       2.44313
10      0.00001
</code></pre><h2 id="where-to-go-from-here">Where to go from here?</h2><div class="hline"></div><p>See <a href="http://www.iqtree.org/doc/Command-Reference">Command Reference</a> for a complete list of all options available in IQ-TREE.</p><div class="spacing"></div></div><div class="col-md-3"><div class="well well-sm"> <a href="http://www.iqtree.org/doc/iqtree-doc.pdf"><i class="fa fa-file-pdf-o fa-lg" aria-hidden="true"></i> Download Tutorials and Manual in one PDF</a></div><div class="well well-sm" id="sidebar"><h4>Documentation</h4><p> <a href="http://www.iqtree.org/doc/Home"> <i class="fa fa-info-circle"></i> Introduction </a></p><p> <a href="http://www.iqtree.org/doc/Quickstart"> <i class="fa fa-info-circle"></i> Getting Started </a></p><p> <a href="http://www.iqtree.org/doc/Web-Server-Tutorial"> <i class="fa fa-info-circle"></i> Web Server Tutorial </a></p><p> <a href="http://www.iqtree.org/doc/Tutorial"> <i class="fa fa-info-circle"></i> Beginner's Tutorial </a></p><p> <i class="fa fa-info-circle"></i> Advanced Tutorial</p><ul><li><a href="http://www.iqtree.org/doc/Advanced-Tutorial#partitioned-analysis-for-multi-gene-alignments">Partitioned analysis</a></li><li><a href="http://www.iqtree.org/doc/Advanced-Tutorial#partitioned-analysis-with-mixed-data">Partitioning with mixed data</a></li><li><a href="http://www.iqtree.org/doc/Advanced-Tutorial#choosing-the-right-partitioning-scheme">Partition scheme selection</a></li><li><a href="http://www.iqtree.org/doc/Advanced-Tutorial#ultrafast-bootstrapping-with-partition-model">Bootstrapping partition model</a></li><li><a href="http://www.iqtree.org/doc/Advanced-Tutorial#constrained-tree-search">Constrained tree search</a></li><li><a href="http://www.iqtree.org/doc/Advanced-Tutorial#tree-topology-tests">Tree topology tests</a></li><li><a href="http://www.iqtree.org/doc/Advanced-Tutorial#testing-constrained-tree">Testing constrained tree</a></li><li><a href="http://www.iqtree.org/doc/Advanced-Tutorial#consensus-construction-and-bootstrap-value-assignment">Consensus construction and bootstrap value assignment</a></li><li><a href="http://www.iqtree.org/doc/Advanced-Tutorial#user-defined-substitution-models">User-defined models</a></li><li><a href="http://www.iqtree.org/doc/Advanced-Tutorial#inferring-site-specific-rates">Inferring site-specific rates</a></li></ul><p> <a href="http://www.iqtree.org/doc/Assessing-Phylogenetic-Assumptions"> <i class="fa fa-info-circle"></i> Assessing Phylogenetic Assumptions </a></p><p> <a href="http://www.iqtree.org/doc/Concordance-Factor"> <i class="fa fa-info-circle"></i> Concordance Factor </a></p><p> <a href="http://www.iqtree.org/doc/Dating"> <i class="fa fa-info-circle"></i> Phylogenetic Dating </a></p><p> <a href="http://www.iqtree.org/doc/Rootstrap"> <i class="fa fa-info-circle"></i> Rooting phylogenies </a></p><p> <a href="http://www.iqtree.org/doc/Frequently-Asked-Questions"> <i class="fa fa-question-circle"></i> Frequently Asked Questions </a></p><p> <a href="http://www.iqtree.org/doc/Substitution-Models"> <i class="fa fa-book"></i> Substitution Models </a></p><p> <a href="http://www.iqtree.org/doc/Complex-Models"> <i class="fa fa-book"></i> Complex Models </a></p><p> <a href="http://www.iqtree.org/doc/Polymorphism-Aware-Models"> <i class="fa fa-book"></i> Polymorphism-Aware Models </a></p><p> <a href="http://www.iqtree.org/doc/Command-Reference"> <i class="fa fa-book"></i> Command Reference </a></p><p> <a href="http://www.iqtree.org/doc/Compilation-Guide"> <i class="fa fa-book"></i> Compilation Guide </a></p><p> <a href="http://www.iqtree.org/doc/Developer-Guide"> <i class="fa fa-codepen"></i> Developer Guide </a></p><p> <a href="http://www.iqtree.org/doc/Analyzing-Big-Data"> <i class="fa fa-info-circle"></i> Analyzing Big Data </a></p></div></div></div></div><div id="footerwrap"><div class="container"><div class="row"><div class="col-sm-4"><h4>Software</h4><p> <a href="http://www.iqtree.org/release/">IQ-TREE release notes</a><br> <a href="https://github.com/Cibiv/IQ-TREE">GitHub repository</a><br> <a href="http://www.iqtree.org/about/">Development team</a><br> <a href="http://www.somsubhra.com/github-release-stats/?username=Cibiv&amp;repository=IQ-TREE">Download statistics</a><br></p></div><div class="col-sm-4"><h4>User support</h4><p> <a href="http://www.iqtree.org/doc/Frequently-Asked-Questions">Frequently asked questions</a><br> <a href="http://www.iqtree.org/doc/">User documentation</a><br> <a href="https://groups.google.com/d/forum/iqtree">IQ-TREE Google group</a><br> <a href="http://iqtree.cibiv.univie.ac.at/" target="_blank">IQ-TREE online web service</a><br></p></div><div class="col-sm-4"><h4>Links</h4><p> <a href="https://cs.anu.edu.au/">Research School of Computer Science, ANU</a><br> <a href="http://www.cibiv.at/">Center for Integrative Bioinformatics Vienna</a><br> <a href="http://validator.w3.org/check/referer">Link Checker</a><br></p></div></div></div></div><div class="alert alert-dismissible text-center cookiealert" role="alert"><div class="cookiealert-container"> We use cookies to ensure you get the best experience on our website <button type="button" class="btn btn-primary btn-sm acceptcookies" aria-label="Close"> I agree </button></div></div><script src="./Advanced Tutorial_files/jquery.min.js.download"></script> <script src="./Advanced Tutorial_files/bootstrap.min.js.download"></script> <script src="./Advanced Tutorial_files/retina-1.1.0.js.download"></script> <script src="./Advanced Tutorial_files/jquery.hoverdir.js.download"></script> <script src="./Advanced Tutorial_files/jquery.isotope.min.js.download"></script> <script src="./Advanced Tutorial_files/cookiealert-standalone.js.download"></script> <script> (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){ (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o), m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m) })(window,document,'script','//www.google-analytics.com/analytics.js','ga'); ga('create', 'UA-38693379-1', 'auto'); ga('send', 'pageview'); </script> <script> // Portfolio (function($) { "use strict"; var $container = $('.portfolio'), $items = $container.find('.portfolio-item'), portfolioLayout = 'fitRows'; if( $container.hasClass('portfolio-centered') ) { portfolioLayout = 'masonry'; } $container.isotope({ filter: '*', animationEngine: 'best-available', layoutMode: portfolioLayout, animationOptions: { duration: 750, easing: 'linear', queue: false }, masonry: { } }, refreshWaypoints()); function refreshWaypoints() { setTimeout(function() { }, 1000); } $('nav.portfolio-filter ul a').on('click', function() { var selector = $(this).attr('data-filter'); $container.isotope({ filter: selector }, refreshWaypoints()); $('nav.portfolio-filter ul a').removeClass('active'); $(this).addClass('active'); return false; }); function getColumnNumber() { var winWidth = $(window).width(), columnNumber = 1; if (winWidth > 1200) { columnNumber = 5; } else if (winWidth > 950) { columnNumber = 4; } else if (winWidth > 600) { columnNumber = 3; } else if (winWidth > 400) { columnNumber = 2; } else if (winWidth > 250) { columnNumber = 1; } return columnNumber; } function setColumns() { var winWidth = $(window).width(), columnNumber = getColumnNumber(), itemWidth = Math.floor(winWidth / columnNumber); $container.find('.portfolio-item').each(function() { $(this).css( { width : itemWidth + 'px' }); }); } function setPortfolio() { setColumns(); $container.isotope('reLayout'); } $container.imagesLoaded(function () { setPortfolio(); }); $(window).on('resize', function () { setPortfolio(); }); })(jQuery); </script>
<table cellspacing="0" cellpadding="0" class="gstl_50 gssb_c" style="width: 423px; display: none; top: 94px; position: absolute; left: 710px;"><tbody><tr><td class="gssb_f"></td><td class="gssb_e" style="width: 100%;"></td></tr></tbody></table></body></html>