<!DOCTYPE html>
<!-- saved from url=(0034)http://www.iqtree.org/doc/Tutorial -->
<html lang="en" class="csstransforms csstransforms3d csstransitions"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1"><meta name="description" content=""><meta name="author" content=""><title>Beginner's Tutorial</title><link href="./Beginner&#39;s Tutorial_files/bootstrap.min.css" rel="stylesheet"><link href="./Beginner&#39;s Tutorial_files/style.css" rel="stylesheet"><link href="./Beginner&#39;s Tutorial_files/font-awesome.min.css" rel="stylesheet"><link rel="stylesheet" href="./Beginner&#39;s Tutorial_files/cookiealert.css"><script async="" src="./Beginner&#39;s Tutorial_files/async-ads.js.download"></script><script async="" src="./Beginner&#39;s Tutorial_files/analytics.js.download"></script><script src="./Beginner&#39;s Tutorial_files/cse_element__en.js.download" type="text/javascript"></script><link type="text/css" rel="stylesheet" href="./Beginner&#39;s Tutorial_files/default+en.css"><link type="text/css" rel="stylesheet" href="./Beginner&#39;s Tutorial_files/default.css"><style type="text/css">.gsc-control-cse{font-family:times new roman, arial, serif}.gsc-control-cse .gsc-table-result{font-family:times new roman, arial, serif}.gsc-refinementsGradient{background:linear-gradient(to left,rgba(255,255,255,1),rgba(255,255,255,0))}.gsc-control-cse{border-color:#337AB7;background-color:#337AB7}input.gsc-input,.gsc-input-box,.gsc-input-box-hover,.gsc-input-box-focus{border-color:#FFFFFF}.gsc-search-button-v2,.gsc-search-button-v2:hover,.gsc-search-button-v2:focus{border-color:#4CAE4C;background-color:#5CB85C;background-image:none;filter:none}.gsc-search-button-v2 svg{fill:#FFFFFF}.gsc-tabHeader.gsc-tabhActive,.gsc-refinementHeader.gsc-refinementhActive{color:#CCCCCC;border-color:#CCCCCC;background-color:#FFFFFF}.gsc-tabHeader.gsc-tabhInactive,.gsc-refinementHeader.gsc-refinementhInactive{color:#CCCCCC;border-color:#CCCCCC;background-color:#FFFFFF}.gsc-webResult.gsc-result,.gsc-results .gsc-imageResult{border-color:#F5F5F5;background-color:#F5F5F5}.gsc-webResult.gsc-result:hover{border-color:#F5F5F5;background-color:#F5F5F5}.gs-webResult.gs-result a.gs-title:link,.gs-webResult.gs-result a.gs-title:link b,.gs-imageResult a.gs-title:link,.gs-imageResult a.gs-title:link b{color:#0000CC}.gs-webResult.gs-result a.gs-title:visited,.gs-webResult.gs-result a.gs-title:visited b,.gs-imageResult a.gs-title:visited,.gs-imageResult a.gs-title:visited b{color:#0000CC}.gs-webResult.gs-result a.gs-title:hover,.gs-webResult.gs-result a.gs-title:hover b,.gs-imageResult a.gs-title:hover,.gs-imageResult a.gs-title:hover b{color:#0000CC}.gs-webResult.gs-result a.gs-title:active,.gs-webResult.gs-result a.gs-title:active b,.gs-imageResult a.gs-title:active,.gs-imageResult a.gs-title:active b{color:#0000CC}.gsc-cursor-page{color:#0000CC}a.gsc-trailing-more-results:link{color:#0000CC}.gs-webResult:not(.gs-no-results-result):not(.gs-error-result) .gs-snippet,.gs-fileFormatType{color:#000000}.gs-webResult div.gs-visibleUrl{color:#008000}.gs-webResult div.gs-visibleUrl-short{color:#008000}.gsc-cursor-box{border-color:#F5F5F5}.gsc-results .gsc-cursor-box .gsc-cursor-page{border-color:#CCCCCC;background-color:#FFFFFF;color:#CCCCCC}.gsc-results .gsc-cursor-box .gsc-cursor-current-page{border-color:#CCCCCC;background-color:#FFFFFF;color:#CCCCCC}.gsc-webResult.gsc-result.gsc-promotion{border-color:#336699;background-color:#FFFFFF}.gsc-completion-title{color:#0000CC}.gsc-completion-snippet{color:#000000}.gs-promotion a.gs-title:link,.gs-promotion a.gs-title:link *,.gs-promotion .gs-snippet a:link{color:#0000CC}.gs-promotion a.gs-title:visited,.gs-promotion a.gs-title:visited *,.gs-promotion .gs-snippet a:visited{color:#0000CC}.gs-promotion a.gs-title:hover,.gs-promotion a.gs-title:hover *,.gs-promotion .gs-snippet a:hover{color:#0000CC}.gs-promotion a.gs-title:active,.gs-promotion a.gs-title:active *,.gs-promotion .gs-snippet a:active{color:#0000CC}.gs-promotion .gs-snippet,.gs-promotion .gs-title .gs-promotion-title-right,.gs-promotion .gs-title .gs-promotion-title-right *{color:#000000}.gs-promotion .gs-visibleUrl,.gs-promotion .gs-visibleUrl-short{color:#008000}.gcsc-find-more-on-google{color:#0000CC}.gcsc-find-more-on-google-magnifier{fill:#0000CC}</style><style type="text/css">.gscb_a{display:inline-block;font:27px/13px arial,sans-serif}.gsst_a .gscb_a{color:#a1b9ed;cursor:pointer}.gsst_a:hover .gscb_a,.gsst_a:focus .gscb_a{color:#36c}.gsst_a{display:inline-block}.gsst_a{cursor:pointer;padding:0 4px}.gsst_a:hover{text-decoration:none!important}.gsst_b{font-size:16px;padding:0 2px;position:relative;user-select:none;-webkit-user-select:none;white-space:nowrap}.gsst_e{vertical-align:middle;opacity:0.55;}.gsst_a:hover .gsst_e,.gsst_a:focus .gsst_e{opacity:0.72;}.gsst_a:active .gsst_e{opacity:1;}.gsst_f{background:white;text-align:left}.gsst_g{background-color:white;border:1px solid #ccc;border-top-color:#d9d9d9;box-shadow:0 2px 4px rgba(0,0,0,0.2);-webkit-box-shadow:0 2px 4px rgba(0,0,0,0.2);margin:-1px -3px;padding:0 6px}.gsst_h{background-color:white;height:1px;margin-bottom:-1px;position:relative;top:-1px}.gsib_a{width:100%;padding:4px 6px 0}.gsib_a,.gsib_b{vertical-align:top}.gssb_c{border:0;position:absolute;z-index:989}.gssb_e{border:1px solid #ccc;border-top-color:#d9d9d9;box-shadow:0 2px 4px rgba(0,0,0,0.2);-webkit-box-shadow:0 2px 4px rgba(0,0,0,0.2);cursor:default}.gssb_f{visibility:hidden;white-space:nowrap}.gssb_k{border:0;display:block;position:absolute;top:0;z-index:988}.gsdd_a{border:none!important}.gsq_a{padding:0}.gssb_a{padding:0 7px}.gssb_a,.gssb_a td{white-space:nowrap;overflow:hidden;line-height:22px}#gssb_b{font-size:11px;color:#36c;text-decoration:none}#gssb_b:hover{font-size:11px;color:#36c;text-decoration:underline}.gssb_g{text-align:center;padding:8px 0 7px;position:relative}.gssb_h{font-size:15px;height:28px;margin:0.2em;-webkit-appearance:button}.gssb_i{background:#eee}.gss_ifl{visibility:hidden;padding-left:5px}.gssb_i .gss_ifl{visibility:visible}a.gssb_j{font-size:13px;color:#36c;text-decoration:none;line-height:100%}a.gssb_j:hover{text-decoration:underline}.gssb_l{height:1px;background-color:#e5e5e5}.gssb_m{color:#000;background:#fff}.gssb_a{padding:0 9px}.gsib_a{padding:5px 9px 4px 9px}.gscb_a{line-height:27px}.gssb_e{border:0}.gssb_l{margin:5px 0}input.gsc-input::-webkit-input-placeholder{font-size:14px}input.gsc-input:-moz-placeholder{font-size:14px}input.gsc-input::-moz-placeholder{font-size:14px}input.gsc-input:-ms-input-placeholder{font-size:14px}input.gsc-input:focus::-webkit-input-placeholder{color:transparent}input.gsc-input:focus:-moz-placeholder{color:transparent}input.gsc-input:focus::-moz-placeholder{color:transparent}input.gsc-input:focus:-ms-input-placeholder{color:transparent}.gssb_c .gsc-completion-container{position:static}.gssb_c{z-index:5000}.gsc-completion-container table{background:transparent;font-size:inherit;font-family:inherit}.gssb_c > tbody > tr,.gssb_c > tbody > tr > td,.gssb_d,.gssb_d > tbody > tr,.gssb_d > tbody > tr > td,.gssb_e,.gssb_e > tbody > tr,.gssb_e > tbody > tr > td{padding:0;margin:0;border:0}.gssb_a table,.gssb_a table tr,.gssb_a table tr td{padding:0;margin:0;border:0}</style></head><body style=""><div class="navbar navbar-default navbar-fixed-top" role="navigation"><div class="container"><div class="navbar-header"> <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar"></span> <span class="icon-bar"></span> <span class="icon-bar"></span> </button> <a class="navbar-brand" href="http://www.iqtree.org/">IQ-TREE</a></div><div class="navbar-collapse collapse navbar-right"><ul class="nav navbar-nav"><li><a data-toggle="collapse" data-target=".navbar-collapse" href="http://www.iqtree.org/#download">Download</a></li><li><a href="http://www.iqtree.org/news/">News</a></li><li class="dropdown"> <a class="dropdown-toggle" data-toggle="dropdown" href="http://www.iqtree.org/doc/Tutorial#">Web server <span class="caret"></span></a><ul class="dropdown-menu"><li><a href="http://iqtree.cibiv.univie.ac.at/" target="_blank">CIBIV, Austria</a></li><li><a href="https://www.hiv.lanl.gov/content/sequence/IQTREE/iqtree.html" target="_blank">Los Alamos Lab, USA</a></li><li><a href="https://www.phylo.org/" target="_blank">CIPRES Gateway, USA</a></li></ul></li><li><a href="http://www.iqtree.org/doc/">Docs</a></li><li><a href="http://www.iqtree.org/workshop/">Workshop</a></li><li><a href="http://www.iqtree.org/about/">About</a></li></ul></div></div></div><div id="blue"><div class="container"><div class="row"><div class="col-sm-6"><h3> Tutorial</h3></div><div class="col-sm-6"> <script type="text/javascript" async="" src="./Beginner&#39;s Tutorial_files/f.txt"></script><script> (function() { var cx = '003669294240048101214:pace-yz0kou'; var gcse = document.createElement('script'); gcse.type = 'text/javascript'; gcse.async = true; gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') + '//cse.google.com/cse.js?cx=' + cx; var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(gcse, s); })(); </script> <div id="___gcse_0"><div class="gsc-control-cse gsc-control-cse-en"><div class="gsc-control-wrapper-cse" dir="ltr"><form class="gsc-search-box gsc-search-box-tools" accept-charset="utf-8"><table cellspacing="0" cellpadding="0" class="gsc-search-box"><tbody><tr><td class="gsc-input"><div class="gsc-input-box" id="gsc-iw-id1"><table cellspacing="0" cellpadding="0" id="gs_id50" class="gstl_50 gsc-input" style="width: 100%; padding: 0px;"><tbody><tr><td id="gs_tti50" class="gsib_a"><input autocomplete="off" type="text" size="10" class="gsc-input" name="search" title="search" id="gsc-i-id1" dir="ltr" spellcheck="false" style="width: 100%; padding: 0px; border: none; margin: 0px; height: auto; background: url(&quot;https://www.google.com/cse/static/images/1x/en/branding.png&quot;) left center no-repeat rgb(255, 255, 255); outline: none;"></td><td class="gsib_b"><div class="gsst_b" id="gs_st50" dir="ltr"><a class="gsst_a" href="javascript:void(0)" style="display: none;" title="Clear search box" role="button"><span class="gscb_a" id="gs_cb50" aria-hidden="true">×</span></a></div></td></tr></tbody></table></div></td><td class="gsc-search-button"><button class="gsc-search-button gsc-search-button-v2"><svg width="13" height="13" viewBox="0 0 13 13"><title>search</title><path d="m4.8495 7.8226c0.82666 0 1.5262-0.29146 2.0985-0.87438 0.57232-0.58292 0.86378-1.2877 0.87438-2.1144 0.010599-0.82666-0.28086-1.5262-0.87438-2.0985-0.59352-0.57232-1.293-0.86378-2.0985-0.87438-0.8055-0.010599-1.5103 0.28086-2.1144 0.87438-0.60414 0.59352-0.8956 1.293-0.87438 2.0985 0.021197 0.8055 0.31266 1.5103 0.87438 2.1144 0.56172 0.60414 1.2665 0.8956 2.1144 0.87438zm4.4695 0.2115 3.681 3.6819-1.259 1.284-3.6817-3.7 0.0019784-0.69479-0.090043-0.098846c-0.87973 0.76087-1.92 1.1413-3.1207 1.1413-1.3553 0-2.5025-0.46363-3.4417-1.3909s-1.4088-2.0686-1.4088-3.4239c0-1.3553 0.4696-2.4966 1.4088-3.4239 0.9392-0.92727 2.0864-1.3969 3.4417-1.4088 1.3553-0.011889 2.4906 0.45771 3.406 1.4088 0.9154 0.95107 1.379 2.0924 1.3909 3.4239 0 1.2126-0.38043 2.2588-1.1413 3.1385l0.098834 0.090049z"></path></svg></button></td><td class="gsc-clear-button"><div class="gsc-clear-button" title="clear results">&nbsp;</div></td></tr></tbody></table></form><div class="gsc-results-wrapper-overlay"><div class="gsc-results-close-btn" tabindex="0"></div><div class="gsc-positioningWrapper"><div class="gsc-tabsAreaInvisible"><div aria-label="refinement" role="tab" class="gsc-tabHeader gsc-inline-block gsc-tabhActive">Custom Search</div><span class="gs-spacer"> </span></div></div><div class="gsc-positioningWrapper"><div class="gsc-tabsAreaInvisible"></div></div><div class="gsc-above-wrapper-area-invisible"><table cellspacing="0" cellpadding="0" class="gsc-above-wrapper-area-container"><tbody><tr><td class="gsc-result-info-container"><div class="gsc-result-info-invisible"></div></td><td class="gsc-orderby-container"><div class="gsc-orderby-invisible"><div class="gsc-orderby-label gsc-inline-block">Sort by:</div><div class="gsc-option-menu-container gsc-inline-block"><div class="gsc-selected-option-container gsc-inline-block"><div class="gsc-selected-option">Relevance</div><div class="gsc-option-selector"></div></div><div class="gsc-option-menu-invisible"><div class="gsc-option-menu-item gsc-option-menu-item-highlighted"><div class="gsc-option">Relevance</div></div><div class="gsc-option-menu-item"><div class="gsc-option">Date</div></div></div></div></div></td></tr></tbody></table></div><div class="gsc-adBlockInvisible"></div><div class="gsc-wrapper"><div class="gsc-adBlockInvisible"></div><div class="gsc-resultsbox-invisible"><div class="gsc-resultsRoot gsc-tabData gsc-tabdActive"><div><div class="gsc-expansionArea"></div></div></div></div></div></div><div class="gsc-modal-background-image" tabindex="0"></div></div></div></div></div></div></div></div><div class="container mtb"><div class="row"><div class="col-md-9"><p><em>Last update: Dec 1, 2019, Contributors: Diep Thi Hoang, Heiko Schmidt, Jana, Minh Bui</em></p><h1 id="beginners-tutorial">Beginner’s tutorial</h1><p>This tutorial gives a beginner’s guide.</p><p>Please first <a href="http://www.iqtree.org/#download">download</a> and <a href="http://www.iqtree.org/doc/Quickstart">install</a> the binary for your platform. For the next steps, the folder containing your <code>iqtree</code> executable should be added to your PATH enviroment variable so that IQ-TREE can be invoked by simply entering <code>iqtree</code> at the command-line. Alternatively, you can also copy <code>iqtree</code> binary into your system search.</p><blockquote class="tip"><p><strong>TIP</strong>: For quick overview of all supported options in IQ-TREE, run the command <code>iqtree -h</code>.</p></blockquote><h2 id="input-data">Input data</h2><div class="hline"></div><p>IQ-TREE takes as input a <em>multiple sequence alignment</em> and will reconstruct an evolutionary tree that is best explained by the input data. If you have raw (unaligned) sequences, you need to first run an alignment program like <a href="http://mafft.cbrc.jp/alignment/software/">MAFFT</a> or <a href="http://www.clustal.org/">ClustalW</a> to align the sequences, before feeding them into IQ-TREE.</p><p>The input alignment can be in various common formats. For example the <a href="http://evolution.genetics.washington.edu/phylip/doc/sequence.html">PHYLIP format</a> which may look like:</p><pre><code>7 28
Frog       AAATTTGGTCCTGTGATTCAGCAGTGAT
Turtle     CTTCCACACCCCAGGACTCAGCAGTGAT
Bird       CTACCACACCCCAGGACTCAGCAGTAAT
Human      CTACCACACCCCAGGAAACAGCAGTGAT
Cow        CTACCACACCCCAGGAAACAGCAGTGAC
Whale      CTACCACGCCCCAGGACACAGCAGTGAT
Mouse      CTACCACACCCCAGGACTCAGCAGTGAT
</code></pre><p>This tiny alignment contains 7 DNA sequences from several animals with the sequence length of 28 nucleotides. IQ-TREE also supports other file formats such as FASTA, NEXUS, CLUSTALW. The FASTA file for the above example may look like this:</p><pre><code>&gt;Frog       
AAATTTGGTCCTGTGATTCAGCAGTGAT
&gt;Turtle     
CTTCCACACCCCAGGACTCAGCAGTGAT
&gt;Bird       
CTACCACACCCCAGGACTCAGCAGTAAT
&gt;Human      
CTACCACACCCCAGGAAACAGCAGTGAT
&gt;Cow        
CTACCACACCCCAGGAAACAGCAGTGAC
&gt;Whale      
CTACCACGCCCCAGGACACAGCAGTGAT
&gt;Mouse      
CTACCACACCCCAGGACTCAGCAGTGAT
</code></pre><blockquote class="tip"><p><strong>TIP</strong>: From version 2 you can input a directory of alignment files. IQ-TREE 2 will load and concatenate all alignments within the directory, eliminating the need for users to manually perform this step.</p></blockquote><h2 id="first-running-example">First running example</h2><div class="hline"></div><p>From the download there is an example alignment called <code>example.phy</code> in PHYLIP format. This example contains parts of the mitochondrial DNA sequences of several animals (Source: <a href="http://www.kuleuven.be/aidslab/phylogenybook/home.html">Phylogenetic Handbook</a>).</p><p>You can now start to reconstruct a maximum-likelihood tree from this alignment by entering (assuming that you are now in the same folder with <code>example.phy</code>):</p><pre><code>iqtree -s example.phy
</code></pre><p><code>-s</code> is the option to specify the name of the alignment file that is always required by IQ-TREE to work. At the end of the run IQ-TREE will write several output files including:</p><ul><li><code>example.phy.iqtree</code>: the main report file that is self-readable. You should look at this file to see the computational results. It also contains a textual representation of the final tree (see below).</li><li><code>example.phy.treefile</code>: the ML tree in NEWICK format, which can be visualized by any supported tree viewer programs like FigTree or iTOL.</li><li><code>example.phy.log</code>: log file of the entire run (also printed on the screen). To report bugs, please send this log file and the original alignment file to the authors.</li></ul><p>For this example data the resulting maximum-likelihood tree may look like this (extracted from <code>.iqtree</code> file):</p><pre><code>NOTE: Tree is UNROOTED although outgroup taxon 'LngfishAu' is drawn at root

+--------------LngfishAu
|
|        +--------------LngfishSA
+--------|
|        +--------------LngfishAf
|
|      +-------------------Frog
+------|
       |               +-----------------Turtle
       |         +-----|
       |         |     |      +-----------------------Sphenodon
       |         |     |   +--|
       |         |     |   |  +--------------------------Lizard
       |         |     +---|
       |         |         |      +---------------------Crocodile
       |         |         +------|
       |         |                +------------------Bird
       +---------|
                 |                  +----------------Human
                 |               +--|
                 |               |  |  +--------Seal
                 |               |  +--|
                 |               |     |   +-------Cow
                 |               |     +---|
                 |               |         +---------Whale
                 |          +----|
                 |          |    |         +------Mouse
                 |          |    +---------|
                 |          |              +--------Rat
                 +----------|
                            |   +----------------Platypus
                            +---|
                                +-------------Opossum
</code></pre><p>This makes sense as the mammals (<code>Human</code> to <code>Opossum</code>) form a clade, whereas the reptiles (<code>Turtle</code> to <code>Crocodile</code>) and <code>Bird</code> form a separate sister clade. Here the tree is drawn at the <em>outgroup</em> Lungfish which is more accient than other species in this example. However, please note that IQ-TREE always produces an <strong>unrooted tree</strong> as it knows nothing about this biological background; IQ-TREE simply draws the tree this way as <code>LngfishAu</code> is the first sequence occuring in the alignment.</p><p>During the example run above, IQ-TREE periodically wrote to disk a checkpoint file <code>example.phy.ckp.gz</code> (gzip-compressed to save space). This checkpoint file is used to resume an interrupted run, which is handy if you have a very large data sets or time limit on a cluster system. If the run did not finish, invoking IQ-TREE again with the very same command line will recover the analysis from the last stopped point, thus saving all computation time done before.</p><p>If the run successfully completed, running again will issue an error message:</p><pre><code>ERROR: Checkpoint (example.phy.ckp.gz) indicates that a previous run successfully finished
Use `-redo` option if you really want to redo the analysis and overwrite all output files.
</code></pre><p>This prevents lost of data if you accidentally re-run IQ-TREE. However, if you really want to re-run the analysis and overwrite all previous output files, use <code>-redo</code> option:</p><pre><code>iqtree -s example.phy -redo
</code></pre><p>Finally, the default prefix of all output files is the alignment file name. You can<br> change the prefix with:</p><pre><code>iqtree -s example.phy --prefix myprefix
# for version 1.x change --prefix to -pre
</code></pre><p>This prevents output files being overwritten when you perform multiple analyses on the same alignment within the same folder.</p><h2 id="choosing-the-right-substitution-model">Choosing the right substitution model</h2><div class="hline"></div><p>NOTE: If you use model selection please cite the following paper:</p><blockquote><p><strong>S. Kalyaanamoorthy, B.Q. Minh, T.K.F. Wong, A. von Haeseler, and L.S. Jermiin</strong> (2017) ModelFinder: fast model selection for accurate phylogenetic estimates. <em>Nat. Methods</em>, 14:587–589. DOI: <a href="https://doi.org/10.1038/nmeth.4285">10.1038/nmeth.4285</a></p></blockquote><p>IQ-TREE supports a wide range of <a href="http://www.iqtree.org/doc/Substitution-Models">substitution models</a> for DNA, protein, codon, binary and morphological alignments. If you do not know which model is appropriate for your data, you can use ModelFinder to determine the best-fit model:</p><pre><code>iqtree -s example.phy -m MFP
# change -m MFP to -m TEST to resemble jModelTest/ProtTest
</code></pre><p><code>-m</code> is the option to specify the model name to use during the analysis. The special <code>MFP</code> key word stands for <em>ModelFinder Plus</em>, which tells IQ-TREE to perform ModelFinder and the remaining analysis using the selected model. ModelFinder computes the log-likelihoods of an initial parsimony tree for many different models and the <em>Akaike information criterion</em> (AIC), <em>corrected Akaike information criterion</em> (AICc), and the <em>Bayesian information criterion</em> (BIC). Then ModelFinder chooses the model that minimizes the BIC score (you can also change to AIC or AICc by adding the option <code>-AIC</code> or <code>-AICc</code>, respectively).</p><blockquote class="tip"><p><strong>TIP</strong>: Starting with version 1.5.4, <code>-m MFP</code> is the default behavior. Thus, this run is equivalent to <code>iqtree -s example.phy</code>.</p></blockquote><p>Here, IQ-TREE will write an additional file:</p><ul><li><code>example.phy.model</code>: log-likelihoods for all models tested. It serves as a checkpoint file to recover an interrupted model selection.</li></ul><p>If you now look at <code>example.phy.iqtree</code> you will see that IQ-TREE selected <code>TIM2+I+G4</code> as the best-fit model for this example data. Thus, for additional analyses you do not have to perform the model test again and can use the selected model:</p><pre><code>iqtree -s example.phy -m TIM2+I+G
</code></pre><p>Sometimes you only want to find the best-fit model without doing tree reconstruction, then run:</p><pre><code>iqtree -s example.phy -m MF
# change -m MF to -m TESTONLY to resemble jModelTest/ProtTest
</code></pre><p>By default, the maximum number of categories is limitted to 10 due to computational reasons. If your sequence alignment is long enough, then you can increase this upper limit with the <code>cmax</code> option:</p><pre><code>iqtree -s example.phy -m MF -cmax 15
</code></pre><p>will test <code>+R2</code> to <code>+R15</code> instead of at most <code>+R10</code>.</p><p>To reduce computational burden, one can use the option <code>-mset</code> to restrict the testing procedure to a subset of base models instead of testing the entire set of all available models. For example, <code>-mset WAG,LG</code> will test only models like <code>WAG+...</code> or <code>LG+...</code>. Another useful option in this respect is <code>-msub</code> for AA data sets. With <code>-msub nuclear</code> only general AA models are included, whereas with <code>-msub viral</code> only AA models for viruses are included.</p><p>If you have enough computational resource, you can perform a thorough and more accurate analysis that invokes a full tree search for each model considered via the <code>-mtree</code> option:</p><pre><code>iqtree -s example.phy -m MF -mtree
</code></pre><h2 id="using-codon-models">Using codon models</h2><div class="hline"></div><p>IQ-TREE supports a number of <a href="http://www.iqtree.org/doc/Substitution-Models#codon-models">codon models</a>. You need to input a protein-coding DNA alignment and specify codon data by option <code>-st CODON</code> (Otherwise, IQ-TREE applies DNA model because it detects that your alignment has DNA sequences):</p><pre><code>iqtree -s coding_gene.phy -st CODON 
</code></pre><p>If your alignment length is not divisible by 3, IQ-TREE will stop with an error message. IQ-TREE will group sites 1,2,3 into codon site 1; sites 4,5,6 to codon site 2; etc. Moreover, any codon, which has at least one gap/unknown/ambiguous nucleotide, will be treated as unknown codon character.</p><p>Note that the above command assumes the standard genetic code. If your sequences follow ‘The Invertebrate Mitochondrial Code’ (see <a href="http://www.iqtree.org/doc/Substitution-Models#codon-models">the full list of supported genetic code here</a>), then run:</p><pre><code>iqtree -s coding_gene.phy -st CODON5 
</code></pre><h2 id="binary-morphological-and-snp-data">Binary, morphological and SNP data</h2><div class="hline"></div><p>Binary alignments contain sequences with characters 0 and 1, which can be in any common formats supported by IQ-TREE, for example, in PHYLIP format:</p><pre><code>4 6
S1   010101
S2   110011
S3   0--100
S4   10--10
</code></pre><p>Morphological alignments have an extended characeter alphabet of 0-9 and A-Z (for states 10-31). For example (PHYLIP format):</p><pre><code>4 10
S1   0123401234
S2   03---20432
S3   3202-04--0
S4   4230120340
</code></pre><p>IQ-TREE will automatically determine the sequence type and the alphabet size. To run IQ-TREE on such alignments:</p><pre><code>iqtree -s morphology.phy
</code></pre><p>or</p><pre><code>iqtree -s morphology.phy -st MORPH
</code></pre><p>IQ-TREE implements to two morphological ML models: <a href="http://www.iqtree.org/doc/Substitution-Models#binary-and-morphological-models">MK and ORDERED</a>. Morphological data typically do not have constant (uninformative) sites. In such cases, you should apply <a href="http://www.iqtree.org/doc/Substitution-Models#ascertainment-bias-correction">ascertainment bias correction</a> model by e.g.:</p><pre><code>iqtree -s morphology.phy -st MORPH -m MK+ASC
</code></pre><p>You can again select the best-fit binary/morphological model:</p><pre><code>iqtree -s morphology.phy -st MORPH
</code></pre><p>For SNP data (DNA) that typically do not contain constant sites, you can explicitly tell the model to include ascertainment bias correction:</p><pre><code>iqtree -s SNP_data.phy -m GTR+ASC
</code></pre><p>You can explicitly tell model testing to only include <code>+ASC</code> model with:</p><pre><code>iqtree -s SNP_data.phy -m MFP+ASC
</code></pre><h2 id="assessing-branch-supports-with-ultrafast-bootstrap-approximation">Assessing branch supports with ultrafast bootstrap approximation</h2><div class="hline"></div><p>To overcome the computational burden required by the nonparametric bootstrap, IQ-TREE introduces an ultrafast bootstrap approximation (UFBoot) (<a href="https://doi.org/10.1093/molbev/mst024">Minh et al., 2013</a>; <a href="https://doi.org/10.1093/molbev/msx281">Hoang et al., 2018</a>) that is orders of magnitude faster than the standard procedure and provides relatively unbiased branch support values. Citation for UFBoot:</p><blockquote><p><strong>D.T. Hoang, O. Chernomor, A. von Haeseler, B.Q. Minh, and L.S. Vinh</strong> (2018) UFBoot2: Improving the ultrafast bootstrap approximation. <em>Mol. Biol. Evol.</em>, 35:518–522. <a href="https://doi.org/10.1093/molbev/msx281">https://doi.org/10.1093/molbev/msx281</a></p></blockquote><p>To run UFBoot:</p><pre><code>iqtree -s example.phy -m TIM2+I+G -B 1000
# for version 1.x change -B to -bb
</code></pre><p><code>-B</code> specifies the number of bootstrap replicates where 1000 is the minimum number recommended. The section <code>MAXIMUM LIKELIHOOD TREE</code> in <code>example.phy.iqtree</code> shows a textual representation of the maximum likelihood tree with branch support values in percentage. The NEWICK format of the tree is printed to the file <code>example.phy.treefile</code>. In addition, IQ-TREE writes the following files:</p><ul><li><code>example.phy.contree</code>: the consensus tree with assigned branch supports where branch lengths are optimized on the original alignment.</li><li><code>example.phy.splits.nex</code>: support values in percentage for all splits (bipartitions), computed as the occurence frequencies in the bootstrap trees. This file can be viewed with the program <a href="http://www.splitstree.org/">SplitsTree</a> to explore the conflicting signals in the data. So it is more informative than consensus tree, e.g. you can see how highly supported the second best conflicting split is, which had no chance to enter the consensus tree.</li><li><code>example.phy.splits</code> (if using <code>-wsplits</code> option): This file contains the same information as <code>example.phy.splits.nex</code> but in star-dot format.</li></ul><blockquote><p><strong>NOTE</strong>: UFBoot support values have a different interpretation to the standard bootstrap. Refer to <a href="http://www.iqtree.org/doc/Frequently-Asked-Questions#how-do-i-interpret-ultrafast-bootstrap-ufboot-support-values">FAQ: UFBoot support values interpretation</a> for more information.</p></blockquote><h2 id="reducing-impact-of-severe-model-violations-with-ufboot">Reducing impact of severe model violations with UFBoot</h2><div class="hline"></div><p>Starting with IQ-TREE version 1.6 we provide a new option <code>-bnni</code> to reduce the risk of overestimating branch supports with UFBoot due to severe model violations. With this option UFBoot will further optimize each bootstrap tree using a hill-climbing nearest neighbor interchange (NNI) search based directly on the corresponding bootstrap alignment.</p><p>Thus, if severe model violations are present in the data set at hand, users are advised to append <code>-bnni</code> to the regular UFBoot command:</p><pre><code>iqtree -s example.phy -m TIM2+I+G -B 1000 -bnni
# for version 1.x change -B to -bb
</code></pre><h2 id="assessing-branch-supports-with--standard-nonparametric-bootstrap">Assessing branch supports with standard nonparametric bootstrap</h2><div class="hline"></div><p>The standard nonparametric bootstrap is invoked by the <code>-b</code> option:</p><pre><code>iqtree -s example.phy -m TIM2+I+G -b 100
</code></pre><p><code>-b</code> specifies the number of bootstrap replicates where 100 is the minimum recommended number. The output files are similar to those produced by the UFBoot procedure.</p><h2 id="assessing-branch-supports-with-single-branch-tests">Assessing branch supports with single branch tests</h2><div class="hline"></div><p>IQ-TREE provides an implementation of the SH-like approximate likelihood ratio test (<a href="https://doi.org/10.1093/sysbio/syq010">Guindon et al., 2010</a>). To perform this test, run:</p><pre><code>iqtree -s example.phy -m TIM2+I+G -alrt 1000
</code></pre><p><code>-alrt</code> specifies the number of bootstrap replicates for SH-aLRT where 1000 is the minimum number recommended.</p><p>IQ-TREE also supports other tests such as the aBayes test (<a href="https://doi.org/10.1093/sysbio/syr041">Anisimova et al., 2011</a>) and the local bootstrap test (<a href="http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.476.8552">Adachi and Hasegawa, 1996b</a>). See <a href="http://www.iqtree.org/doc/Command-Reference#single-branch-tests">single branch tests</a> for more details.</p><p>You can also perform both SH-aLRT and the ultrafast bootstrap within one single run:</p><pre><code>iqtree -s example.phy -m TIM2+I+G -alrt 1000 -B 1000
# for version 1.x change -B to -bb
</code></pre><p>The branches of the resulting <code>.treefile</code> will be assigned with both SH-aLRT and UFBoot support values, which are readable by any tree viewer program like FigTree, Dendroscope or ETE. You can also look at the textual tree figure in <code>.iqtree</code> file:</p><pre><code>NOTE: Tree is UNROOTED although outgroup taxon 'LngfishAu' is drawn at root
Numbers in parentheses are SH-aLRT support (%) / ultrafast bootstrap support (%)

+-------------LngfishAu
|
|       +--------------LngfishSA
+-------| (100/100)
|       +------------LngfishAf
|
|      +--------------------Frog
+------| (99.8/100)
       |                     +-----------------Turtle
       |                  +--| (85/72)
       |                  |  |    +------------------------Crocodile
       |                  |  +----| (96.5/97)
       |                  |       +------------------Bird
       |               +--| (39/51)
       |               |  +---------------------------Sphenodon
       |         +-----| (98.2/99)
       |         |     +-------------------------------Lizard
       +---------| (100/100)
                 |                   +--------------Human
                 |                +--| (92.3/93)
                 |                |  |  +------Seal
                 |                |  +--| (68.3/75)
                 |                |     |  +-----Cow
                 |                |     +--| (99.7/100)
                 |                |        +-------Whale
                 |           +----| (99.1/100)
                 |           |    |         +---Mouse
                 |           |    +---------| (100/100)
                 |           |              +------Rat
                 +-----------| (100/100)
                             |  +--------------Platypus
                             +--| (93/98)
                                +-----------Opossum
</code></pre><p>From this figure, the branching patterns within reptiles are poorly supported (e.g. <code>Sphenodon</code> with SH-aLRT: 39%, UFBoot: 51% and <code>Turtle</code> with SH-aLRT: 85%, UFBoot: 72%) as well as the phylogenetic position of <code>Seal</code> within mammals (SH-aLRT: 68.3%, UFBoot: 75%). Other branches appear to be well supported.</p><h2 id="utilizing-multi-core-cpus">Utilizing multi-core CPUs</h2><div class="hline"></div><p>IQ-TREE can utilize multiple CPU cores to speed up the analysis. A complement option <code>-T</code> (or <code>-nt</code> for version 1.x) allows specifying the number of CPU cores to use. For example:</p><pre><code>iqtree -s example.phy -m TIM2+I+G -T 2
# for version 1.x change -T to -nt
</code></pre><p>Here, IQ-TREE will use 2 CPU cores to perform the analysis.</p><p>Note that the parallel efficiency is only good for long alignments. A good practice is to use <code>-T AUTO</code> to determine the best number of cores:</p><pre><code>iqtree -s example.phy -m TIM2+I+G -T AUTO
# for version 1.x change -T to -nt
</code></pre><p>Then while running IQ-TREE may print something like this on to the screen:</p><pre><code>Measuring multi-threading efficiency up to 8 CPU cores
Threads: 1 / Time: 8.001 sec / Speedup: 1.000 / Efficiency: 100% / LogL: -22217
Threads: 2 / Time: 4.346 sec / Speedup: 1.841 / Efficiency: 92% / LogL: -22217
Threads: 3 / Time: 3.381 sec / Speedup: 2.367 / Efficiency: 79% / LogL: -22217
Threads: 4 / Time: 4.385 sec / Speedup: 1.825 / Efficiency: 46% / LogL: -22217
BEST NUMBER OF THREADS: 3
</code></pre><p>Therefore, I would only use 3 cores for this example data. For later analysis with your same data set, you can stick to the determined number.</p><p>Depending on the compute system it might be required to set an upper limit of CPU cores that can automatically be assigned. Use the <code>-ntmax</code> option to do so. For instance</p><pre><code>iqtree -s example.phy -m TIM2+I+G -T AUTO -ntmax 8
# for version 1.x change -T to -nt
</code></pre><p>does the same as above, but only allows to use up to 8 CPU cores. By default all cores of the current machine would be used as maximum.</p><h2 id="where-to-go-from-here">Where to go from here?</h2><div class="hline"></div><p>Once confident enough you can go on with a <strong><a href="http://www.iqtree.org/doc/Advanced-Tutorial">more advanced tutorial</a></strong>, which covers topics like phylogenomic (multi-gene) analyses using partition models or mixture models.</p><div class="spacing"></div></div><div class="col-md-3"><div class="well well-sm"> <a href="http://www.iqtree.org/doc/iqtree-doc.pdf"><i class="fa fa-file-pdf-o fa-lg" aria-hidden="true"></i> Download Tutorials and Manual in one PDF</a></div><div class="well well-sm" id="sidebar"><h4>Documentation</h4><p> <a href="http://www.iqtree.org/doc/Home"> <i class="fa fa-info-circle"></i> Introduction </a></p><p> <a href="http://www.iqtree.org/doc/Quickstart"> <i class="fa fa-info-circle"></i> Getting Started </a></p><p> <a href="http://www.iqtree.org/doc/Web-Server-Tutorial"> <i class="fa fa-info-circle"></i> Web Server Tutorial </a></p><p> <i class="fa fa-info-circle"></i> Beginner's Tutorial</p><ul><li><a href="http://www.iqtree.org/doc/Tutorial#input-data">Input data</a></li><li><a href="http://www.iqtree.org/doc/Tutorial#first-running-example">First example</a></li><li><a href="http://www.iqtree.org/doc/Tutorial#choosing-the-right-substitution-model">Model selection</a></li><li><a href="http://www.iqtree.org/doc/Tutorial#using-codon-models">Using codon models</a></li><li><a href="http://www.iqtree.org/doc/Tutorial#binary-morphological-and-snp-data">Binary, Morphological, SNPs</a></li><li><a href="http://www.iqtree.org/doc/Tutorial#assessing-branch-supports-with-ultrafast-bootstrap-approximation">Ultrafast bootstrap</a></li><li><a href="http://www.iqtree.org/doc/Tutorial#reducing-impact-of-severe-model-violations-with-ufboot">Reducing impact of severe model violations with UFBoot</a></li><li><a href="http://www.iqtree.org/doc/Tutorial#assessing-branch-supports-with--standard-nonparametric-bootstrap">Nonparametric bootstrap</a></li><li><a href="http://www.iqtree.org/doc/Tutorial#assessing-branch-supports-with-single-branch-tests">Single branch tests</a></li><li><a href="http://www.iqtree.org/doc/Tutorial#utilizing-multi-core-cpus">Utilizing multi-core CPUs</a></li></ul><p> <a href="http://www.iqtree.org/doc/Advanced-Tutorial"> <i class="fa fa-info-circle"></i> Advanced Tutorial </a></p><p> <a href="http://www.iqtree.org/doc/Assessing-Phylogenetic-Assumptions"> <i class="fa fa-info-circle"></i> Assessing Phylogenetic Assumptions </a></p><p> <a href="http://www.iqtree.org/doc/Concordance-Factor"> <i class="fa fa-info-circle"></i> Concordance Factor </a></p><p> <a href="http://www.iqtree.org/doc/Dating"> <i class="fa fa-info-circle"></i> Phylogenetic Dating </a></p><p> <a href="http://www.iqtree.org/doc/Rootstrap"> <i class="fa fa-info-circle"></i> Rooting phylogenies </a></p><p> <a href="http://www.iqtree.org/doc/Frequently-Asked-Questions"> <i class="fa fa-question-circle"></i> Frequently Asked Questions </a></p><p> <a href="http://www.iqtree.org/doc/Substitution-Models"> <i class="fa fa-book"></i> Substitution Models </a></p><p> <a href="http://www.iqtree.org/doc/Complex-Models"> <i class="fa fa-book"></i> Complex Models </a></p><p> <a href="http://www.iqtree.org/doc/Polymorphism-Aware-Models"> <i class="fa fa-book"></i> Polymorphism-Aware Models </a></p><p> <a href="http://www.iqtree.org/doc/Command-Reference"> <i class="fa fa-book"></i> Command Reference </a></p><p> <a href="http://www.iqtree.org/doc/Compilation-Guide"> <i class="fa fa-book"></i> Compilation Guide </a></p><p> <a href="http://www.iqtree.org/doc/Developer-Guide"> <i class="fa fa-codepen"></i> Developer Guide </a></p><p> <a href="http://www.iqtree.org/doc/Analyzing-Big-Data"> <i class="fa fa-info-circle"></i> Analyzing Big Data </a></p></div></div></div></div><div id="footerwrap"><div class="container"><div class="row"><div class="col-sm-4"><h4>Software</h4><p> <a href="http://www.iqtree.org/release/">IQ-TREE release notes</a><br> <a href="https://github.com/Cibiv/IQ-TREE">GitHub repository</a><br> <a href="http://www.iqtree.org/about/">Development team</a><br> <a href="http://www.somsubhra.com/github-release-stats/?username=Cibiv&amp;repository=IQ-TREE">Download statistics</a><br></p></div><div class="col-sm-4"><h4>User support</h4><p> <a href="http://www.iqtree.org/doc/Frequently-Asked-Questions">Frequently asked questions</a><br> <a href="http://www.iqtree.org/doc/">User documentation</a><br> <a href="https://groups.google.com/d/forum/iqtree">IQ-TREE Google group</a><br> <a href="http://iqtree.cibiv.univie.ac.at/" target="_blank">IQ-TREE online web service</a><br></p></div><div class="col-sm-4"><h4>Links</h4><p> <a href="https://cs.anu.edu.au/">Research School of Computer Science, ANU</a><br> <a href="http://www.cibiv.at/">Center for Integrative Bioinformatics Vienna</a><br> <a href="http://validator.w3.org/check/referer">Link Checker</a><br></p></div></div></div></div><div class="alert alert-dismissible text-center cookiealert" role="alert"><div class="cookiealert-container"> We use cookies to ensure you get the best experience on our website <button type="button" class="btn btn-primary btn-sm acceptcookies" aria-label="Close"> I agree </button></div></div><script src="./Beginner&#39;s Tutorial_files/jquery.min.js.download"></script> <script src="./Beginner&#39;s Tutorial_files/bootstrap.min.js.download"></script> <script src="./Beginner&#39;s Tutorial_files/retina-1.1.0.js.download"></script> <script src="./Beginner&#39;s Tutorial_files/jquery.hoverdir.js.download"></script> <script src="./Beginner&#39;s Tutorial_files/jquery.isotope.min.js.download"></script> <script src="./Beginner&#39;s Tutorial_files/cookiealert-standalone.js.download"></script> <script> (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){ (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o), m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m) })(window,document,'script','//www.google-analytics.com/analytics.js','ga'); ga('create', 'UA-38693379-1', 'auto'); ga('send', 'pageview'); </script> <script> // Portfolio (function($) { "use strict"; var $container = $('.portfolio'), $items = $container.find('.portfolio-item'), portfolioLayout = 'fitRows'; if( $container.hasClass('portfolio-centered') ) { portfolioLayout = 'masonry'; } $container.isotope({ filter: '*', animationEngine: 'best-available', layoutMode: portfolioLayout, animationOptions: { duration: 750, easing: 'linear', queue: false }, masonry: { } }, refreshWaypoints()); function refreshWaypoints() { setTimeout(function() { }, 1000); } $('nav.portfolio-filter ul a').on('click', function() { var selector = $(this).attr('data-filter'); $container.isotope({ filter: selector }, refreshWaypoints()); $('nav.portfolio-filter ul a').removeClass('active'); $(this).addClass('active'); return false; }); function getColumnNumber() { var winWidth = $(window).width(), columnNumber = 1; if (winWidth > 1200) { columnNumber = 5; } else if (winWidth > 950) { columnNumber = 4; } else if (winWidth > 600) { columnNumber = 3; } else if (winWidth > 400) { columnNumber = 2; } else if (winWidth > 250) { columnNumber = 1; } return columnNumber; } function setColumns() { var winWidth = $(window).width(), columnNumber = getColumnNumber(), itemWidth = Math.floor(winWidth / columnNumber); $container.find('.portfolio-item').each(function() { $(this).css( { width : itemWidth + 'px' }); }); } function setPortfolio() { setColumns(); $container.isotope('reLayout'); } $container.imagesLoaded(function () { setPortfolio(); }); $(window).on('resize', function () { setPortfolio(); }); })(jQuery); </script>
<table cellspacing="0" cellpadding="0" class="gstl_50 gssb_c" style="width: 423px; display: none; top: 94px; position: absolute; left: 710px;"><tbody><tr><td class="gssb_f"></td><td class="gssb_e" style="width: 100%;"></td></tr></tbody></table></body></html>