<!DOCTYPE html>
<!-- saved from url=(0043)http://www.iqtree.org/doc/Command-Reference -->
<html lang="en" class="csstransforms csstransforms3d csstransitions"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1"><meta name="description" content=""><meta name="author" content=""><title>Command Reference</title><link href="./Command Reference_files/bootstrap.min.css" rel="stylesheet"><link href="./Command Reference_files/style.css" rel="stylesheet"><link href="./Command Reference_files/font-awesome.min.css" rel="stylesheet"><link rel="stylesheet" href="./Command Reference_files/cookiealert.css"><script async="" src="./Command Reference_files/async-ads.js.download"></script><script async="" src="./Command Reference_files/analytics.js.download"></script><script src="./Command Reference_files/cse_element__en.js.download" type="text/javascript"></script><link type="text/css" rel="stylesheet" href="./Command Reference_files/default+en.css"><link type="text/css" rel="stylesheet" href="./Command Reference_files/default.css"><style type="text/css">.gsc-control-cse{font-family:times new roman, arial, serif}.gsc-control-cse .gsc-table-result{font-family:times new roman, arial, serif}.gsc-refinementsGradient{background:linear-gradient(to left,rgba(255,255,255,1),rgba(255,255,255,0))}.gsc-control-cse{border-color:#337AB7;background-color:#337AB7}input.gsc-input,.gsc-input-box,.gsc-input-box-hover,.gsc-input-box-focus{border-color:#FFFFFF}.gsc-search-button-v2,.gsc-search-button-v2:hover,.gsc-search-button-v2:focus{border-color:#4CAE4C;background-color:#5CB85C;background-image:none;filter:none}.gsc-search-button-v2 svg{fill:#FFFFFF}.gsc-tabHeader.gsc-tabhActive,.gsc-refinementHeader.gsc-refinementhActive{color:#CCCCCC;border-color:#CCCCCC;background-color:#FFFFFF}.gsc-tabHeader.gsc-tabhInactive,.gsc-refinementHeader.gsc-refinementhInactive{color:#CCCCCC;border-color:#CCCCCC;background-color:#FFFFFF}.gsc-webResult.gsc-result,.gsc-results .gsc-imageResult{border-color:#F5F5F5;background-color:#F5F5F5}.gsc-webResult.gsc-result:hover{border-color:#F5F5F5;background-color:#F5F5F5}.gs-webResult.gs-result a.gs-title:link,.gs-webResult.gs-result a.gs-title:link b,.gs-imageResult a.gs-title:link,.gs-imageResult a.gs-title:link b{color:#0000CC}.gs-webResult.gs-result a.gs-title:visited,.gs-webResult.gs-result a.gs-title:visited b,.gs-imageResult a.gs-title:visited,.gs-imageResult a.gs-title:visited b{color:#0000CC}.gs-webResult.gs-result a.gs-title:hover,.gs-webResult.gs-result a.gs-title:hover b,.gs-imageResult a.gs-title:hover,.gs-imageResult a.gs-title:hover b{color:#0000CC}.gs-webResult.gs-result a.gs-title:active,.gs-webResult.gs-result a.gs-title:active b,.gs-imageResult a.gs-title:active,.gs-imageResult a.gs-title:active b{color:#0000CC}.gsc-cursor-page{color:#0000CC}a.gsc-trailing-more-results:link{color:#0000CC}.gs-webResult:not(.gs-no-results-result):not(.gs-error-result) .gs-snippet,.gs-fileFormatType{color:#000000}.gs-webResult div.gs-visibleUrl{color:#008000}.gs-webResult div.gs-visibleUrl-short{color:#008000}.gsc-cursor-box{border-color:#F5F5F5}.gsc-results .gsc-cursor-box .gsc-cursor-page{border-color:#CCCCCC;background-color:#FFFFFF;color:#CCCCCC}.gsc-results .gsc-cursor-box .gsc-cursor-current-page{border-color:#CCCCCC;background-color:#FFFFFF;color:#CCCCCC}.gsc-webResult.gsc-result.gsc-promotion{border-color:#336699;background-color:#FFFFFF}.gsc-completion-title{color:#0000CC}.gsc-completion-snippet{color:#000000}.gs-promotion a.gs-title:link,.gs-promotion a.gs-title:link *,.gs-promotion .gs-snippet a:link{color:#0000CC}.gs-promotion a.gs-title:visited,.gs-promotion a.gs-title:visited *,.gs-promotion .gs-snippet a:visited{color:#0000CC}.gs-promotion a.gs-title:hover,.gs-promotion a.gs-title:hover *,.gs-promotion .gs-snippet a:hover{color:#0000CC}.gs-promotion a.gs-title:active,.gs-promotion a.gs-title:active *,.gs-promotion .gs-snippet a:active{color:#0000CC}.gs-promotion .gs-snippet,.gs-promotion .gs-title .gs-promotion-title-right,.gs-promotion .gs-title .gs-promotion-title-right *{color:#000000}.gs-promotion .gs-visibleUrl,.gs-promotion .gs-visibleUrl-short{color:#008000}.gcsc-find-more-on-google{color:#0000CC}.gcsc-find-more-on-google-magnifier{fill:#0000CC}</style><style type="text/css">.gscb_a{display:inline-block;font:27px/13px arial,sans-serif}.gsst_a .gscb_a{color:#a1b9ed;cursor:pointer}.gsst_a:hover .gscb_a,.gsst_a:focus .gscb_a{color:#36c}.gsst_a{display:inline-block}.gsst_a{cursor:pointer;padding:0 4px}.gsst_a:hover{text-decoration:none!important}.gsst_b{font-size:16px;padding:0 2px;position:relative;user-select:none;-webkit-user-select:none;white-space:nowrap}.gsst_e{vertical-align:middle;opacity:0.55;}.gsst_a:hover .gsst_e,.gsst_a:focus .gsst_e{opacity:0.72;}.gsst_a:active .gsst_e{opacity:1;}.gsst_f{background:white;text-align:left}.gsst_g{background-color:white;border:1px solid #ccc;border-top-color:#d9d9d9;box-shadow:0 2px 4px rgba(0,0,0,0.2);-webkit-box-shadow:0 2px 4px rgba(0,0,0,0.2);margin:-1px -3px;padding:0 6px}.gsst_h{background-color:white;height:1px;margin-bottom:-1px;position:relative;top:-1px}.gsib_a{width:100%;padding:4px 6px 0}.gsib_a,.gsib_b{vertical-align:top}.gssb_c{border:0;position:absolute;z-index:989}.gssb_e{border:1px solid #ccc;border-top-color:#d9d9d9;box-shadow:0 2px 4px rgba(0,0,0,0.2);-webkit-box-shadow:0 2px 4px rgba(0,0,0,0.2);cursor:default}.gssb_f{visibility:hidden;white-space:nowrap}.gssb_k{border:0;display:block;position:absolute;top:0;z-index:988}.gsdd_a{border:none!important}.gsq_a{padding:0}.gssb_a{padding:0 7px}.gssb_a,.gssb_a td{white-space:nowrap;overflow:hidden;line-height:22px}#gssb_b{font-size:11px;color:#36c;text-decoration:none}#gssb_b:hover{font-size:11px;color:#36c;text-decoration:underline}.gssb_g{text-align:center;padding:8px 0 7px;position:relative}.gssb_h{font-size:15px;height:28px;margin:0.2em;-webkit-appearance:button}.gssb_i{background:#eee}.gss_ifl{visibility:hidden;padding-left:5px}.gssb_i .gss_ifl{visibility:visible}a.gssb_j{font-size:13px;color:#36c;text-decoration:none;line-height:100%}a.gssb_j:hover{text-decoration:underline}.gssb_l{height:1px;background-color:#e5e5e5}.gssb_m{color:#000;background:#fff}.gssb_a{padding:0 9px}.gsib_a{padding:5px 9px 4px 9px}.gscb_a{line-height:27px}.gssb_e{border:0}.gssb_l{margin:5px 0}input.gsc-input::-webkit-input-placeholder{font-size:14px}input.gsc-input:-moz-placeholder{font-size:14px}input.gsc-input::-moz-placeholder{font-size:14px}input.gsc-input:-ms-input-placeholder{font-size:14px}input.gsc-input:focus::-webkit-input-placeholder{color:transparent}input.gsc-input:focus:-moz-placeholder{color:transparent}input.gsc-input:focus::-moz-placeholder{color:transparent}input.gsc-input:focus:-ms-input-placeholder{color:transparent}.gssb_c .gsc-completion-container{position:static}.gssb_c{z-index:5000}.gsc-completion-container table{background:transparent;font-size:inherit;font-family:inherit}.gssb_c > tbody > tr,.gssb_c > tbody > tr > td,.gssb_d,.gssb_d > tbody > tr,.gssb_d > tbody > tr > td,.gssb_e,.gssb_e > tbody > tr,.gssb_e > tbody > tr > td{padding:0;margin:0;border:0}.gssb_a table,.gssb_a table tr,.gssb_a table tr td{padding:0;margin:0;border:0}</style></head><body style=""><div class="navbar navbar-default navbar-fixed-top" role="navigation"><div class="container"><div class="navbar-header"> <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar"></span> <span class="icon-bar"></span> <span class="icon-bar"></span> </button> <a class="navbar-brand" href="http://www.iqtree.org/">IQ-TREE</a></div><div class="navbar-collapse collapse navbar-right"><ul class="nav navbar-nav"><li><a data-toggle="collapse" data-target=".navbar-collapse" href="http://www.iqtree.org/#download">Download</a></li><li><a href="http://www.iqtree.org/news/">News</a></li><li class="dropdown"> <a class="dropdown-toggle" data-toggle="dropdown" href="http://www.iqtree.org/doc/Command-Reference#">Web server <span class="caret"></span></a><ul class="dropdown-menu"><li><a href="http://iqtree.cibiv.univie.ac.at/" target="_blank">CIBIV, Austria</a></li><li><a href="https://www.hiv.lanl.gov/content/sequence/IQTREE/iqtree.html" target="_blank">Los Alamos Lab, USA</a></li><li><a href="https://www.phylo.org/" target="_blank">CIPRES Gateway, USA</a></li></ul></li><li><a href="http://www.iqtree.org/doc/">Docs</a></li><li><a href="http://www.iqtree.org/workshop/">Workshop</a></li><li><a href="http://www.iqtree.org/about/">About</a></li></ul></div></div></div><div id="blue"><div class="container"><div class="row"><div class="col-sm-6"><h3> Manual</h3></div><div class="col-sm-6"> <script type="text/javascript" async="" src="./Command Reference_files/f.txt"></script><script> (function() { var cx = '003669294240048101214:pace-yz0kou'; var gcse = document.createElement('script'); gcse.type = 'text/javascript'; gcse.async = true; gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') + '//cse.google.com/cse.js?cx=' + cx; var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(gcse, s); })(); </script> <div id="___gcse_0"><div class="gsc-control-cse gsc-control-cse-en"><div class="gsc-control-wrapper-cse" dir="ltr"><form class="gsc-search-box gsc-search-box-tools" accept-charset="utf-8"><table cellspacing="0" cellpadding="0" class="gsc-search-box"><tbody><tr><td class="gsc-input"><div class="gsc-input-box" id="gsc-iw-id1"><table cellspacing="0" cellpadding="0" id="gs_id50" class="gstl_50 gsc-input" style="width: 100%; padding: 0px;"><tbody><tr><td id="gs_tti50" class="gsib_a"><input autocomplete="off" type="text" size="10" class="gsc-input" name="search" title="search" id="gsc-i-id1" dir="ltr" spellcheck="false" style="width: 100%; padding: 0px; border: none; margin: 0px; height: auto; background: url(&quot;https://www.google.com/cse/static/images/1x/en/branding.png&quot;) left center no-repeat rgb(255, 255, 255); outline: none;"></td><td class="gsib_b"><div class="gsst_b" id="gs_st50" dir="ltr"><a class="gsst_a" href="javascript:void(0)" style="display: none;" title="Clear search box" role="button"><span class="gscb_a" id="gs_cb50" aria-hidden="true">×</span></a></div></td></tr></tbody></table></div></td><td class="gsc-search-button"><button class="gsc-search-button gsc-search-button-v2"><svg width="13" height="13" viewBox="0 0 13 13"><title>search</title><path d="m4.8495 7.8226c0.82666 0 1.5262-0.29146 2.0985-0.87438 0.57232-0.58292 0.86378-1.2877 0.87438-2.1144 0.010599-0.82666-0.28086-1.5262-0.87438-2.0985-0.59352-0.57232-1.293-0.86378-2.0985-0.87438-0.8055-0.010599-1.5103 0.28086-2.1144 0.87438-0.60414 0.59352-0.8956 1.293-0.87438 2.0985 0.021197 0.8055 0.31266 1.5103 0.87438 2.1144 0.56172 0.60414 1.2665 0.8956 2.1144 0.87438zm4.4695 0.2115 3.681 3.6819-1.259 1.284-3.6817-3.7 0.0019784-0.69479-0.090043-0.098846c-0.87973 0.76087-1.92 1.1413-3.1207 1.1413-1.3553 0-2.5025-0.46363-3.4417-1.3909s-1.4088-2.0686-1.4088-3.4239c0-1.3553 0.4696-2.4966 1.4088-3.4239 0.9392-0.92727 2.0864-1.3969 3.4417-1.4088 1.3553-0.011889 2.4906 0.45771 3.406 1.4088 0.9154 0.95107 1.379 2.0924 1.3909 3.4239 0 1.2126-0.38043 2.2588-1.1413 3.1385l0.098834 0.090049z"></path></svg></button></td><td class="gsc-clear-button"><div class="gsc-clear-button" title="clear results">&nbsp;</div></td></tr></tbody></table></form><div class="gsc-results-wrapper-overlay"><div class="gsc-results-close-btn" tabindex="0"></div><div class="gsc-positioningWrapper"><div class="gsc-tabsAreaInvisible"><div aria-label="refinement" role="tab" class="gsc-tabHeader gsc-inline-block gsc-tabhActive">Custom Search</div><span class="gs-spacer"> </span></div></div><div class="gsc-positioningWrapper"><div class="gsc-tabsAreaInvisible"></div></div><div class="gsc-above-wrapper-area-invisible"><table cellspacing="0" cellpadding="0" class="gsc-above-wrapper-area-container"><tbody><tr><td class="gsc-result-info-container"><div class="gsc-result-info-invisible"></div></td><td class="gsc-orderby-container"><div class="gsc-orderby-invisible"><div class="gsc-orderby-label gsc-inline-block">Sort by:</div><div class="gsc-option-menu-container gsc-inline-block"><div class="gsc-selected-option-container gsc-inline-block"><div class="gsc-selected-option">Relevance</div><div class="gsc-option-selector"></div></div><div class="gsc-option-menu-invisible"><div class="gsc-option-menu-item gsc-option-menu-item-highlighted"><div class="gsc-option">Relevance</div></div><div class="gsc-option-menu-item"><div class="gsc-option">Date</div></div></div></div></div></td></tr></tbody></table></div><div class="gsc-adBlockInvisible"></div><div class="gsc-wrapper"><div class="gsc-adBlockInvisible"></div><div class="gsc-resultsbox-invisible"><div class="gsc-resultsRoot gsc-tabData gsc-tabdActive"><div><div class="gsc-expansionArea"></div></div></div></div></div></div><div class="gsc-modal-background-image" tabindex="0"></div></div></div></div></div></div></div></div><div class="container mtb"><div class="row"><div class="col-md-9"><p><em>Last update: May 19, 2021, Contributors: Diep Thi Hoang, Dominik Schrempf, Heiko Schmidt, Jana Trifinopoulos, Minh Bui</em></p><h1 id="command-reference">Command reference</h1><p>Commprehensive documentation of command-line options.</p><p>IQ-TREE is invoked from the command-line with e.g.:</p><pre><code>iqtree -s &lt;alignment&gt; [OPTIONS]
</code></pre><p>assuming that IQ-TREE can be run by simply entering <code>iqtree</code>. If not, please change <code>iqtree</code> to the actual path of the executable or read the <a href="http://www.iqtree.org/doc/Quickstart">Quick start guide</a>.</p><h2 id="general-options">General options</h2><div class="hline"></div><p>General options are mainly intended for specifying input and output files:</p><table><thead><tr><th>Option</th><th>Usage and meaning</th></tr></thead><tbody><tr><td><code>-h</code> or <code>-?</code></td><td>Print help usage.</td></tr><tr><td><code>-s</code></td><td>Specify input alignment file in PHYLIP, FASTA, NEXUS, CLUSTAL or MSF format.</td></tr><tr><td><code>-st</code></td><td>Specify sequence type as either of <code>DNA</code>, <code>AA</code>, <code>BIN</code>, <code>MORPH</code>, <code>CODON</code> or <code>NT2AA</code> for DNA, amino-acid, binary, morphological, codon or DNA-to-AA-translated sequences. This is only necessary if IQ-TREE did not detect the sequence type correctly. Note that <code>-st CODON</code> is always necessary when using codon models (otherwise, IQ-TREE applies DNA models) and you also need to specify a <a href="http://www.iqtree.org/doc/Substitution-Models#codon-models">genetic code like this</a> if differed from the standard genetic code. <code>-st NT2AA</code> tells IQ-TREE to translate protein-coding DNA into AA sequences and then subsequent analysis will work on the AA sequences. You can also use a genetic code like <code>-st NT2AA5</code> for the Invertebrate Mitochondrial Code (see <a href="http://www.iqtree.org/doc/Substitution-Models#codon-models">genetic code table</a>).</td></tr><tr><td><code>-t</code></td><td>Specify a file containing starting tree for tree search. The special option <code>-t BIONJ</code> starts tree search from BIONJ tree and <code>-t RANDOM</code> starts tree search from completely random tree. <em>DEFAULT: 100 parsimony trees + BIONJ tree</em></td></tr><tr><td><code>-te</code></td><td>Like <code>-t</code> but fixing user tree. That means, no tree search is performed and IQ-TREE computes the log-likelihood of the fixed user tree.</td></tr><tr><td><code>-o</code></td><td>Specify an outgroup taxon name to root the tree. The output tree in <code>.treefile</code> will be rooted accordingly. <em>DEFAULT: first taxon in alignment</em></td></tr><tr><td><code>-pre</code></td><td>Specify a prefix for all output files. <em>DEFAULT: either alignment file name (<code>-s</code>) or partition file name (<code>-q</code>, <code>-spp</code> or <code>-sp</code>)</em></td></tr><tr><td><code>-nt</code></td><td>Specify the number of CPU cores for the multicore version. A special option <code>-nt AUTO</code> will tell IQ-TREE to automatically determine the best number of cores given the current data and computer.</td></tr><tr><td><code>-ntmax</code></td><td>Specify the maximal number of CPU cores <code>-nt AUTO</code> is allowed to allocate <em>DEFAULT: #CPU cores on the current machine</em></td></tr><tr><td><code>-seed</code></td><td>Specify a random number seed to reproduce a previous run. This is normally used for debugging purposes. <em>DEFAULT: based on current machine clock</em></td></tr><tr><td><code>-v</code></td><td>Turn on verbose mode for printing more messages to screen. This is normally used for debugging purposes. <em>DEFAULT: OFF</em></td></tr><tr><td><code>-quiet</code></td><td>Silent mode, suppress printing to the screen. Note that <code>.log</code> file is still written.</td></tr><tr><td><code>-keep-ident</code></td><td>Keep identical sequences in the alignment. Bu default: IQ-TREE will remove them during the analysis and add them in the end.</td></tr><tr><td><code>-safe</code></td><td>Turn on safe numerical mode to avoid numerical underflow for large data sets with many sequences (typically in the order of thousands). This mode is automatically turned on when having more than 2000 sequences.</td></tr><tr><td><code>-mem</code></td><td>Specify maximal RAM usage, for example, <code>-mem 64G</code> to use at most 64 GB of RAM. By default, IQ-TREE will try to not to exceed the computer RAM size.</td></tr></tbody></table><h3 id="example-usages">Example usages:</h3><ul><li><p>Reconstruct a maximum-likelihood tree given a sequence alignment file <code>example.phy</code>:</p><pre><code>  iqtree -s example.phy
</code></pre></li><li><p>Reconstruct a maximum-likelihood tree using at most 8 GB RAM and automatically detect the number of cores to use:</p><pre><code>  # For version &lt;= 1.5.X
  iqtree-omp -s example.phy -nt AUTO -mem 8G
    
  # For version &gt;= 1.6.0, change iqtree-omp to just iqtree
  iqtree -s example.phy -nt AUTO -mem 8G
</code></pre></li><li><p>Like above but write all output to <code>myrun.*</code> files:</p><pre><code>  # For version &lt;= 1.5.X
  iqtree-omp -s example.phy -nt AUTO -mem 8G -pre myrun

  # For version &lt;= 1.6.0
  iqtree -s example.phy -nt AUTO -mem 8G -pre myrun
</code></pre></li></ul><h2 id="checkpointing-to-resume-stopped-run">Checkpointing to resume stopped run</h2><div class="hline"></div><p>Starting with version 1.4.0 IQ-TREE supports checkpointing: If an IQ-TREE run was interrupted for some reason, rerunning it with the same command line and input data will automatically resume the analysis from the last stopped point. The previous log file will then be appended. If a run successfully finished, IQ-TREE will deny to rerun and issue an error message. A few options that control checkpointing behavior:</p><table><thead><tr><th>Option</th><th>Usage and meaning</th></tr></thead><tbody><tr><td><code>-redo</code></td><td>Redo the entire analysis no matter if it was stopped or successful. WARNING: This option will overwrite all existing output files.</td></tr><tr><td><code>-cptime</code></td><td>Specify the minimum checkpoint time interval in seconds (default: 20s)</td></tr></tbody></table><blockquote><p><strong>NOTE</strong>: IQ-TREE writes a checkpoint file with name suffix <code>.ckp.gz</code> in gzip format. Do not modify this file. If you delete this file, all checkpointing information will be lost!</p></blockquote><h2 id="likelihood-mapping-analysis">Likelihood mapping analysis</h2><div class="hline"></div><p>Starting with version 1.4.0, IQ-TREE implements the likelihood mapping approach (<a href="http://www.pnas.org/content/94/13/6815.long">Strimmer and von Haeseler, 1997</a>) to assess the phylogenetic information of an input alignment. The detailed results will be printed to <code>.iqtree</code> report file. The likelihood mapping plots will be printed to <code>.lmap.svg</code> and <code>.lmap.eps</code> files.</p><p>Compared with the original implementation in TREE-PUZZLE, IQ-TREE is much faster and supports many more substitution models (including partition and mixture models).</p><table><thead><tr><th>Option</th><th>Usage and meaning</th></tr></thead><tbody><tr><td><code>-lmap</code></td><td>Specify the number of quartets to be randomly drawn. If you specify <code>-lmap ALL</code>, all unique quartets will be drawn, instead.</td></tr><tr><td><code>-lmclust</code></td><td>Specify a NEXUS file containing taxon clusters (see below for example) for quartet mapping analysis.</td></tr><tr><td><code>-n 0</code></td><td>Skip subsequent tree search, useful when you only want to assess the phylogenetic information of the alignment.</td></tr><tr><td><code>-wql</code></td><td>Write quartet log-likelihoods into <code>.lmap.quartetlh</code> file (typically not needed).</td></tr></tbody></table><blockquote class="tip"><p><strong>TIP</strong>: The number of quartets specified via <code>-lmap</code> is recommended to be at least 25 times the number of sequences in the alignment, such that each sequence is covered ~100 times in the set of quartets drawn.</p></blockquote><p>An example NEXUS cluster file (where A, B, C, etc. are sequence names):</p><pre><code>#NEXUS
begin sets;
    taxset Cluster1 = A B C;
    taxset Cluster2 = D E;
    taxset Cluster3 = F G H I;
    taxset Cluster4 = J;
    taxset IGNORED = X;
end;
</code></pre><p>Here, <code>Cluster1</code> to <code>Cluster4</code> are four user-defined clusters of sequences. Note that users can give any names to the clusters instead of <code>Cluster1</code>, etc. If a cluster is named <code>ignored</code> or <code>IGNORED</code> the sequences included will be ignored in the likelihood mapping analysis.</p><p>If you provide a cluster file it has to contain between 1 and 4 clusters (plus an optional <code>IGNORED</code> or <code>ignored</code> cluster), which will tell IQ-TREE to perform an unclustered (default without a cluster file) or a clustered analysis with 2, 3 or 4 groups, respectively.</p><p>The names given to the clusters in the cluster file will be used to label the corners of the triangle diagrams.</p><h3 id="example-usages-1">Example usages:</h3><ul><li><p>Perform solely likelihood mapping analysis (ignoring tree search) with 2000 quartets for an alignment <code>example.phy</code> with model being automatically selected:</p><pre><code>  iqtree -s example.phy -lmap 2000 -n 0 -m TEST
</code></pre></li></ul><p>You can now view the likelihood mapping plot file <code>example.phy.lmap.svg</code>, which looks like this:</p><p><img src="./Command Reference_files/example.phy.lmap.pdf" alt="Likelihood mapping plot."></p><p>It shows phylogenetic information of the alignment <code>example.phy</code>. On the top: distribution of quartets depicted by dots on the likelihood mapping plot. On the left: the three areas show support for one of the different groupings like (a,b)-(c,d). On the right: quartets falling into the three corners are informative. Those in three rectangles are partly informative and those in the center are uninformative. A good data set should have high number of informative quartets and low number of uninformative quartets. The meanings can also be found in the <code>LIKELIHOOD MAPPING STATISTICS</code> section of the report file <code>example.phy.iqtree</code>:</p><pre><code>LIKELIHOOD MAPPING STATISTICS
-----------------------------

           (a,b)-(c,d)                              (a,b)-(c,d)      
                /\                                      /\           
               /  \                                    /  \          
              /    \                                  /  1 \         
             /  a1  \                                / \  / \        
            /\      /\                              /   \/   \       
           /  \    /  \                            /    /\    \      
          /    \  /    \                          / 6  /  \  4 \     
         /      \/      \                        /\   /  7 \   /\    
        /        |       \                      /  \ /______\ /  \   
       /   a3    |    a2  \                    / 3  |    5   |  2 \  
      /__________|_________\                  /_____|________|_____\ 
(a,d)-(b,c)            (a,c)-(b,d)      (a,d)-(b,c)            (a,c)-(b,d) 

Division of the likelihood mapping plots into 3 or 7 areas.
On the left the areas show support for one of the different groupings
like (a,b|c,d).
On the right the right quartets falling into the areas 1, 2 and 3 are
informative. Those in the rectangles 4, 5 and 6 are partly informative
and those in the center (7) are not informative.
.....
</code></pre><h2 id="automatic-model-selection">Automatic model selection</h2><div class="hline"></div><p>The default model (e.g., <code>HKY+F</code> for DNA, <code>LG</code> for protein data) may not fit well to the data. Therefore, IQ-TREE allows to automatically determine the best-fit model via a series of <code>-m TEST...</code> option:</p><table><thead><tr><th>Option</th><th>Usage and meaning</th></tr></thead><tbody><tr><td><code>-m TESTONLY</code></td><td>Perform standard model selection like jModelTest (for DNA) and ProtTest (for protein). Moreover, IQ-TREE also works for codon, binary and morphogical data. If a partition file is specified, IQ-TREE will find the best-fit model for each partition.</td></tr><tr><td><code>-m TEST</code></td><td>Like <code>-m TESTONLY</code> but immediately followed by tree reconstruction using the best-fit model found. So this performs both model selection and tree inference within a single run.</td></tr><tr><td><code>-m TESTNEWONLY</code> or <code>-m MF</code></td><td>Perform an extended model selection that additionally includes FreeRate model compared with <code>-m TESTONLY</code>. <em>Recommended as replacement for <code>-m TESTONLY</code></em>. Note that <code>LG4X</code> is a FreeRate model, but by default is not included because it is also a protein mixture model. To include it, use <code>-madd</code> option (see table below).</td></tr><tr><td><code>-m TESTNEW</code> or <code>-m MFP</code></td><td>Like <code>-m MF</code> but immediately followed by tree reconstruction using the best-fit model found.</td></tr></tbody></table><blockquote class="tip"><p><strong>TIP</strong>: During model section, IQ-TREE will write a model checkpoint file (suffix <code>.model</code> in version &lt;= 1.5.X or <code>.model.gz</code> in version &gt;= 1.6.X) that stores information of all models tested so far. Thus, if IQ-TREE is interrupted for whatever reason, restarting the run will load this file to reuse the computation.</p></blockquote><p>IQ-TREE version 1.6 or later allows to additionally test <a href="http://www.iqtree.org/doc/Substitution-Models#lie-markov-models">Lie Markov DNA models</a> by adding the following keyword to <code>-m</code> option:</p><table><thead><tr><th>Option</th><th>Usage and meaning</th></tr></thead><tbody><tr><td><code>+LM</code></td><td>Additionally consider all Lie Markov models</td></tr><tr><td><code>+LMRY</code></td><td>Additionally consider all Lie Markov models with RY symmetry</td></tr><tr><td><code>+LMWS</code></td><td>Additionally consider all Lie Markov models with WS symmetry</td></tr><tr><td><code>+LMMK</code></td><td>Additionally consider all Lie Markov models with MK symmetry</td></tr><tr><td><code>+LMSS</code></td><td>Additionally consider all strand-symmetric Lie Markov models</td></tr></tbody></table><p>When <a href="http://www.iqtree.org/doc/Command-Reference#partition-model-options">a partition file is specified</a> then you can append <code>MERGE</code> keyword into <code>-m</code> option to find the best-fit partitioning scheme like PartitionFinder (<a href="https://doi.org/10.1093/molbev/mss020">Lanfear et al., 2012</a>). More specifically,</p><table><thead><tr><th>Option</th><th>Usage and meaning</th></tr></thead><tbody><tr><td><code>-m TESTMERGEONLY</code></td><td>Select best-fit partitioning scheme by possibly merging partitions to reduce over-parameterization and increase model fit. It implements the greedy algorithm of PartitionFinder.</td></tr><tr><td><code>-m TESTMERGE</code></td><td>Like <code>-m TESTMERGEONLY</code> but immediately followed by tree reconstruction using the best partitioning scheme found.</td></tr><tr><td><code>-m TESTNEWMERGEONLY</code> or <code>-m MF+MERGE</code></td><td>Like <code>-m TESTMERGEONLY</code> but additionally includes FreeRate model.</td></tr><tr><td><code>-m TESTNEWMERGE</code> or <code>-m MFP+MERGE</code></td><td>Like <code>-m MF+MERGE</code> but immediately followed by tree reconstruction using the best partitioning scheme found.</td></tr><tr><td><code>-rcluster</code></td><td>Specify the percentage for the relaxed clustering algorithm (<a href="https://doi.org/10.1186/1471-2148-14-82">Lanfear et al., 2014</a>) to speed up the computation instead of the default slow greedy algorithm. This is similar to <code>--rcluster-percent</code> option of PartitionFinder. For example, with <code>-rcluster 10</code> only the top 10% partition schemes are considered to save computations.</td></tr><tr><td><code>-rclusterf</code></td><td>Similar to <code>-rcluster</code> but using the <strong>fast</strong> relaxed clustering algorithm (<a href="https://doi.org/10.1093/molbev/msw260">Lanfear et al., 2017</a>) of PartitionFinder2. Introduced in version 1.6.</td></tr><tr><td><code>-rcluster-max</code></td><td>Specify the absolute maximum number of partition pairs in the paritition merging phase. Default: the larger of 1000 and 10 times the number of partitions. This option is similar to <code>--rcluster-max</code> option of PartitionFinder2.</td></tr></tbody></table><blockquote><p><strong>WARNING</strong>: For versions &lt;= 1.5.X, all commands with <code>-m ...MERGE...</code> will always perform an edge-unlinked partition scheme finding even if <code>-spp</code> option is used. Only in the next phase of tree reconstruction, then an edge-linked partition model is used. However, for versions 1.6.X onwards, the edge-linked partition finding is performed by <code>-spp</code> option.</p></blockquote><p>Several parameters can be set to e.g. reduce computations:</p><table><thead><tr><th>Option</th><th>Usage and meaning</th></tr></thead><tbody><tr><td><code>-mset</code></td><td>Specify the name of a program (<code>raxml</code>, <code>phyml</code> or <code>mrbayes</code>) to restrict to only those models supported by the specified program. Alternatively, one can specify a comma-separated list of base models. For example, <code>-mset WAG,LG,JTT</code> will restrict model selection to WAG, LG, and JTT instead of all 18 AA models to save computations.</td></tr><tr><td><code>-msub</code></td><td>Specify either <code>nuclear</code>, <code>mitochondrial</code>, <code>chloroplast</code> or <code>viral</code> to restrict to those AA models designed for specified source.</td></tr><tr><td><code>-mfreq</code></td><td>Specify a comma-separated list of frequency types for model selection. <em>DEFAULT: <code>-mfreq FU,F</code> for protein models (FU = AA frequencies given by the protein matrix, F = empirical AA frequencies from the data), <code>-mfreq ,F1x4,F3x4,F</code> for codon models</em></td></tr><tr><td><code>-mrate</code></td><td>Specify a comma-separated list of rate heterogeneity types for model selection. <em>DEFAULT: <code>-mrate E,I,G,I+G</code> for standard procedure, <code>-mrate E,I,G,I+G,R</code> for new selection procedure</em>. (E means Equal/homogeneous rate model).</td></tr><tr><td><code>-cmin</code></td><td>Specify minimum number of categories for FreeRate model. <em>DEFAULT: 2</em></td></tr><tr><td><code>-cmax</code></td><td>Specify maximum number of categories for FreeRate model. It is recommended to increase if alignment is long enough. <em>DEFAULT: 10</em></td></tr><tr><td><code>-merit</code></td><td>Specify either <code>AIC</code>, <code>AICc</code> or <code>BIC</code> for the optimality criterion to apply for new procedure. <em>DEFAULT: all three criteria are considered</em></td></tr><tr><td><code>-mtree</code></td><td>Turn on full tree search for each model considered, to obtain more accurate result. Only recommended if enough computational resources are available. <em>DEFAULT: fixed starting tree</em></td></tr><tr><td><code>-mredo</code></td><td>Ignore model checkpoint file computed earlier. <em>DEFAULT: model checkpoint file (if exists) is loaded to reuse previous computations</em></td></tr><tr><td><code>-madd</code></td><td>Specify a comma-separated list of mixture models to additionally consider for model selection. For example, <code>-madd LG4M,LG4X</code> to additionally include these two <a href="http://www.iqtree.org/doc/Substitution-Models#protein-models">protein mixture models</a>.</td></tr><tr><td><code>-mdef</code></td><td>Specify a <a href="http://www.iqtree.org/doc/Complex-Models#nexus-model-file">NEXUS model file</a> to define new models.</td></tr></tbody></table><blockquote><p><strong>NOTE</strong>: Some of the above options require a comma-separated list, which should not contain any empty space!</p></blockquote><h3 id="example-usages-2">Example usages:</h3><ul><li><p>Select best-fit model for alignment <code>data.phy</code> based on Bayesian information criterion (BIC):</p><pre><code>  iqtree -s data.phy -m TESTONLY
</code></pre></li><li><p>Select best-fit model for a protein alignment <code>prot.phy</code> using the new testing procedure and only consider WAG, LG and JTT matrix to save time:</p><pre><code>  iqtree -s prot.phy -m MF -mset WAG,LG,JTT
</code></pre></li><li><p>Find the best partitioning scheme for alignment <code>data.phy</code> and partition file <code>partition.nex</code> with a relaxed clustering at 10% to save time:</p><pre><code>  iqtree -s data.phy -spp partition.nex -m TESTMERGEONLY -rcluster 10
</code></pre></li></ul><h2 id="specifying-substitution-models">Specifying substitution models</h2><div class="hline"></div><p><code>-m</code> is a powerful option to specify substitution models, state frequency and rate heterogeneity type. The general syntax is:</p><pre><code>-m MODEL+FreqType+RateType
</code></pre><p>where <code>MODEL</code> is a model name, <code>+FreqType</code> (optional) is the frequency type and <code>+RateType</code> (optional) is the rate heterogeneity type.</p><p>The following <code>MODEL</code>s are available:</p><table><thead><tr><th>DataType</th><th>Model names</th></tr></thead><tbody><tr><td>DNA</td><td>JC/JC69, F81, K2P/K80, HKY/HKY85, TN/TrN/TN93, TNe, K3P/K81, K81u, TPM2, TPM2u, TPM3, TPM3u, TIM, TIMe, TIM2, TIM2e, TIM3, TIM3e, TVM, TVMe, SYM, GTR and 6-digit specification. See <a href="http://www.iqtree.org/doc/Substitution-Models#dna-models">DNA models</a> for more details.</td></tr><tr><td>Protein</td><td>BLOSUM62, cpREV, Dayhoff, DCMut, FLU, HIVb, HIVw, JTT, JTTDCMut, LG, mtART, mtMAM, mtREV, mtZOA, mtMet, mtVer, mtInv, Poisson, PMB, rtREV, VT, WAG.</td></tr><tr><td>Protein</td><td>Mixture models: C10, …, C60 (CAT model) (<a href="https://doi.org/10.1093/molbev/msh112">Lartillot and Philippe, 2004</a>), EX2, EX3, EHO, UL2, UL3, EX_EHO, LG4M, LG4X, CF4. See <a href="http://www.iqtree.org/doc/Substitution-Models#protein-models">Protein models</a> for more details.</td></tr><tr><td>Codon</td><td>MG, MGK, MG1KTS, MG1KTV, MG2K, GY, GY1KTS, GY1KTV, GY2K, ECMK07/KOSI07, ECMrest, ECMS05/SCHN05 and combined empirical-mechanistic models. See <a href="http://www.iqtree.org/doc/Substitution-Models#codon-models">Codon models</a> for more details.</td></tr><tr><td>Binary</td><td>JC2, GTR2. See <a href="http://www.iqtree.org/doc/Substitution-Models#binary-and-morphological-models">Binary and morphological models</a> for more details.</td></tr><tr><td>Morphology</td><td>MK, ORDERED. See <a href="http://www.iqtree.org/doc/Substitution-Models#binary-and-morphological-models">Binary and morphological models</a> for more details.</td></tr></tbody></table><p>The following <code>FreqType</code>s are supported:</p><table><thead><tr><th>FreqType</th><th>Meaning</th></tr></thead><tbody><tr><td><code>+F</code></td><td>Empirical state frequency observed from the data.</td></tr><tr><td><code>+FO</code></td><td>State frequency optimized by maximum-likelihood from the data. Note that this is with letter-O and not digit-0.</td></tr><tr><td><code>+FQ</code></td><td>Equal state frequency.</td></tr><tr><td><code>+F1x4</code></td><td>See <a href="http://www.iqtree.org/doc/Substitution-Models#codon-frequencies">Codon frequencies</a>.</td></tr><tr><td><code>+F3x4</code></td><td>See <a href="http://www.iqtree.org/doc/Substitution-Models#codon-frequencies">Codon frequencies</a>.</td></tr></tbody></table><p>Further options:</p><table><thead><tr><th>Option</th><th>Usage and meaning</th></tr></thead><tbody><tr><td><code>-mwopt</code></td><td>Turn on optimizing weights of mixture models. Note that for models like <code>LG+C20+F+G</code> this mode is automatically turned on, but not for <code>LG+C20+G</code>.</td></tr></tbody></table><h3 id="example-usages-3">Example usages:</h3><ul><li><p>Infer an ML tree for a DNA alignment <code>dna.phy</code> under GTR+I+G model:</p><pre><code>  iqtree -s dna.phy -m GTR+I+G
</code></pre></li><li><p>Infer an ML tree for a protein alignment <code>prot.phy</code> under LG+F+G model:</p><pre><code>  iqtree -s prot.phy -m LG+F+G
</code></pre></li><li><p>Infer an ML tree for a protein alignment <code>prot.phy</code> under profile mixture model LG+C10+F+G:</p><pre><code>  iqtree -s prot.phy -m LG+C10+F+G
</code></pre></li></ul><h2 id="rate-heterogeneity">Rate heterogeneity</h2><div class="hline"></div><p>The following <code>RateType</code>s are supported:</p><table><thead><tr><th>RateType</th><th>Meaning</th></tr></thead><tbody><tr><td><code>+I</code></td><td>Allowing for a proportion of invariable sites.</td></tr><tr><td><code>+G</code></td><td>Discrete Gamma model (<a href="https://doi.org/10.1007/BF00160154">Yang, 1994</a>) with default 4 rate categories. The number of categories can be changed with e.g. <code>+G8</code>.</td></tr><tr><td><code>+I+G</code></td><td>Invariable site plus discrete Gamma model (<a href="http://mbe.oxfordjournals.org/content/12/4/546.abstract">Gu et al., 1995</a>).</td></tr><tr><td><code>+R</code></td><td>FreeRate model (<a href="http://www.genetics.org/content/139/2/993.abstract">Yang, 1995</a>; <a href="https://doi.org/10.1093/molbev/mss140">Soubrier et al., 2012</a>) that generalizes <code>+G</code> by relaxing the assumption of Gamma-distributed rates. The number of categories can be specified with e.g. <code>+R6</code>. <em>DEFAULT: 4 categories</em></td></tr><tr><td><code>+I+R</code></td><td>invariable site plus FreeRate model.</td></tr></tbody></table><p>See <a href="http://www.iqtree.org/doc/Substitution-Models#rate-heterogeneity-across-sites">Rate heterogeneity across sites</a> for more details.</p><p>Further options:</p><table><thead><tr><th>Option</th><th>Usage and meaning</th></tr></thead><tbody><tr><td><code>-a</code></td><td>Specify the Gamma shape parameter (default: estimate)</td></tr><tr><td><code>-gmedian</code></td><td>Perform the <em>median</em> approximation for Gamma rate heterogeneity instead of the default <em>mean</em> approximation (<a href="https://doi.org/10.1007/BF00160154">Yang, 1994</a>)</td></tr><tr><td><code>-i</code></td><td>Specify the proportion of invariable sites (default: estimate)</td></tr><tr><td><code>--opt-gamma-inv</code></td><td>Perform more thorough estimation for <code>+I+G</code> model parameters</td></tr><tr><td><code>-wsr</code></td><td>Write per-site rates to <code>.rate</code> file</td></tr></tbody></table><p>Optionally, one can specify <a href="http://www.iqtree.org/doc/Substitution-Models#ascertainment-bias-correction">Ascertainment bias correction</a> by appending <code>+ASC</code> to the model string. <a href="http://www.iqtree.org/doc/Complex-Models#mixture-models">Advanced mixture models</a> can also be specified via <code>MIX{...}</code> and <code>FMIX{...}</code> syntax. Option <code>-mwopt</code> can be used to turn on optimizing weights of mixture models.</p><h2 id="partition-model-options">Partition model options</h2><div class="hline"></div><p>Partition models are used for phylogenomic data with multiple genes. You first have to prepare <a href="http://www.iqtree.org/doc/Complex-Models#partition-file-format">a partition file in NEXUS or RAxML-style format</a>. Then use the following options to input the partition file:</p><table><thead><tr><th>Option</th><th>Usage and meaning</th></tr></thead><tbody><tr><td><code>-q</code></td><td>Specify partition file for edge-equal <a href="http://www.iqtree.org/doc/Complex-Models#partition-models">partition model</a>. That means, all partitions share the same set of branch lengths (like <code>-q</code> option of RAxML).</td></tr><tr><td><code>-spp</code></td><td>Like <code>-q</code> but allowing partitions to have different evolutionary speeds (<a href="http://www.iqtree.org/doc/Complex-Models#partition-models">edge-proportional partition model</a>).</td></tr><tr><td><code>-sp</code></td><td>Specify partition file for <a href="http://www.iqtree.org/doc/Complex-Models#partition-models">edge-unlinked partition model</a>. That means, each partition has its own set of branch lengths (like <code>-M</code> option of RAxML). This is the most parameter-rich partition model to accomodate <em>heterotachy</em>.</td></tr></tbody></table><h2 id="site-specific-frequency-model-options">Site-specific frequency model options</h2><div class="hline"></div><p>The site-specific frequency model is used to substantially reduce the time and memory requirement compared with full profile mixture models <code>C10</code> to <code>C60</code>. For full details see <a href="http://www.iqtree.org/doc/Complex-Models#site-specific-frequency-models">site-specific frequency model</a>. To use this model you have to specify a profile mixture model with e.g. <code>-m LG+C20+F+G</code> together with a guide tree or a site frequency file:</p><table><thead><tr><th>Option</th><th>Usage and meaning</th></tr></thead><tbody><tr><td><code>-ft</code></td><td>Specify a guide tree (in Newick format) to infer site frequency profiles.</td></tr><tr><td><code>-fs</code></td><td>Specify a site frequency file, e.g. the <code>.sitefreq</code> file obtained from <code>-ft</code> run. This will save memory used for the first phase of the analysis.</td></tr><tr><td><code>-fmax</code></td><td>Switch to posterior maximum mode for obtaining site-specific profiles. Default: posterior mean.</td></tr></tbody></table><p>With <code>-fs</code> option you can input a file containing your own site frequency profiles. The format of this file is that each line contains the site ID (starting from 1) and the state frequencies (20 for amino-acid) separated by white space. So it has as many lines as the number of sites in the alignment. The order of amino-acids is:</p><pre><code> A   R   N   D   C   Q   E   G   H   I   L   K   M   F   P   S   T   W   Y   V
</code></pre><h2 id="tree-search-parameters">Tree search parameters</h2><div class="hline"></div><p>The new IQ-TREE search algorithm (<a href="https://doi.org/10.1093/molbev/msu300">Nguyen et al., 2015</a>) has several parameters that can be changed with:</p><table><thead><tr><th>Option</th><th>Usage and meaning</th></tr></thead><tbody><tr><td><code>-allnni</code></td><td>Turn on more thorough and slower NNI search. It means that IQ-TREE will consider all possible NNIs instead of only those in the vicinity of previously applied NNIs. <em>DEFAULT: OFF</em></td></tr><tr><td><code>-djc</code></td><td>Avoid computing ML pairwise distances and BIONJ tree.</td></tr><tr><td><code>-fast</code></td><td>Turn on the fast tree search mode, where IQ-TREE will just construct two starting trees: maximum parsimony and BIONJ, which are then optimized by nearest neighbor interchange (NNI). Introduced in version 1.6.</td></tr><tr><td><code>-g</code></td><td>Specify a topological constraint tree file in NEWICK format. The constraint tree can be a multifurcating tree and need not to include all taxa.</td></tr><tr><td><code>-ninit</code></td><td>Specify number of initial parsimony trees. <em>DEFAULT: 100</em>. Here <a href="http://www.libpll.org/">the PLL library</a> (<a href="https://doi.org/10.1093/sysbio/syu084">Flouri et al., 2015</a>) is used, which implements the randomized stepwise addition and parsimony subtree pruning and regafting (SPR).</td></tr><tr><td><code>-n</code></td><td>Specify number of iterations to stop. This option overrides <code>-nstop</code> criterion.</td></tr><tr><td><code>-ntop</code></td><td>Specify number of top initial parsimony trees to optimize with ML nearest neighbor interchange (NNI) search to initialize the candidate set. <em>DEFAULT: 20</em></td></tr><tr><td><code>-nbest</code></td><td>Specify number of trees in the candidate set to maintain during ML tree search. <em>DEFAULT: 5</em></td></tr><tr><td><code>-nstop</code></td><td>Specify number of unsuccessful iterations to stop. <em>DEFAULT: 100</em></td></tr><tr><td><code>-pers</code></td><td>Specify perturbation strength (between 0 and 1) for randomized NNI. <em>DEFAULT: 0.5</em></td></tr><tr><td><code>-sprrad</code></td><td>Specify SPR radius for the initial parsimony tree search. <em>DEFAULT: 6</em></td></tr></tbody></table><blockquote><p><strong>NOTE</strong>: While the default parameters were empirically determined to work well under our extensive benchmark (<a href="https://doi.org/10.1093/molbev/msu300">Nguyen et al., 2015</a>), it might not hold true for all data sets. If in doubt that tree search is still stuck in local optima, one should repeat analysis with at least 10 IQ-TREE runs. Moreover, our experience showed that <code>-pers</code> and <code>-nstop</code> are the most relevant options to change in such case. For example, data sets with many short sequences should be analyzed with smaller perturbation strength (e.g. <code>-pers 0.2</code>) and larger number of stop iterations (e.g. <code>-nstop 500</code>).</p></blockquote><h3 id="example-usages-4">Example usages:</h3><ul><li><p>Infer an ML tree for an alignment <code>data.phy</code> with increased stopping iteration of 500 and reduced perturbation strength of 0.2:</p><pre><code>  iqtree -s data.phy -m TEST -nstop 500 -pers 0.2
</code></pre></li><li><p>Infer an ML tree for an alignment <code>data.phy</code> obeying a topological constraint tree <code>constraint.tree</code>:</p><pre><code>  iqtree -s data.phy -m TEST -g constraint.tree
</code></pre></li></ul><h2 id="ultrafast-bootstrap-parameters">Ultrafast bootstrap parameters</h2><div class="hline"></div><p>The ultrafast bootstrap (UFBoot) approximation (<a href="https://doi.org/10.1093/molbev/mst024">Minh et al., 2013</a>; <a href="https://doi.org/10.1093/molbev/msx281">Hoang et al., in press</a>) has several parameters that can be changed with:</p><table><thead><tr><th>Option</th><th>Usage and meaning</th></tr></thead><tbody><tr><td><code>-bb</code></td><td>Specify number of bootstrap replicates (&gt;=1000).</td></tr><tr><td><code>-bcor</code></td><td>Specify minimum correlation coefficient for UFBoot convergence criterion. <em>DEFAULT: 0.99</em></td></tr><tr><td><code>-beps</code></td><td>Specify a small epsilon to break tie in RELL evaluation for bootstrap trees. <em>DEFAULT: 0.5</em></td></tr><tr><td><code>-bnni</code></td><td>Perform an additional step to further optimize UFBoot trees by nearest neighbor interchange (NNI) based directly on bootstrap alignments. This option is recommended in the presence of <strong>severe model violations</strong>. It increases computing time by 2-fold but reduces the risk of overestimating branch supports due to severe model violations. Introduced in IQ-TREE 1.6.</td></tr><tr><td><code>-bsam</code></td><td>Specify the resampling strategies for partitioned analysis. By default, IQ-TREE resamples alignment sites within partitions. With <code>-bsam GENE</code> IQ-TREE will resample partitions. With <code>-bsam GENESITE</code> IQ-TREE will resample partitions and then resample sites within resampled partitions (<a href="https://doi.org/10.1002/jez.b.21026">Gadagkar et al., 2005</a>; <a href="https://doi.org/10.1073/pnas.0408313102">Seo et al., 2005</a>).</td></tr><tr><td><code>-nm</code></td><td>Specify maximum number of iterations to stop. <em>DEFAULT: 1000</em></td></tr><tr><td><code>-nstep</code></td><td>Specify iteration interval checking for UFBoot convergence. <em>DEFAULT: every 100 iterations</em></td></tr><tr><td><code>-wbt</code></td><td>Turn on writing bootstrap trees to <code>.ufboot</code> file. <em>DEFAULT: OFF</em></td></tr><tr><td><code>-wbtl</code></td><td>Like <code>-wbt</code> but bootstrap trees written with branch lengths. <em>DEFAULT: OFF</em></td></tr></tbody></table><h3 id="example-usages-5">Example usages:</h3><ul><li><p>Select best-fit model, infer an ML tree and perform ultrafast bootstrap with 1000 replicates:</p><pre><code>  iqtree -s data.phy -m TEST -bb 1000
</code></pre></li><li><p>Reconstruct ML and perform ultrafast bootstrap (5000 replicates) under a partition model file <code>partition.nex</code>:</p><pre><code>  iqtree -s data.phy -spp partition.nex -m TEST -bb 5000
</code></pre></li></ul><h2 id="nonparametric-bootstrap">Nonparametric bootstrap</h2><div class="hline"></div><p>The slow standard nonparametric bootstrap (<a href="https://doi.org/10.2307/2408678">Felsenstein, 1985</a>) can be run with:</p><table><thead><tr><th>Option</th><th>Usage and meaning</th></tr></thead><tbody><tr><td><code>-b</code></td><td>Specify number of bootstrap replicates (recommended &gt;=100). This will perform both bootstrap and analysis on original alignment and provide a consensus tree.</td></tr><tr><td><code>-bc</code></td><td>Like <code>-b</code> but omit analysis on original alignment.</td></tr><tr><td><code>-bo</code></td><td>Like <code>-b</code> but only perform bootstrap analysis (no analysis on original alignment and no consensus tree).</td></tr></tbody></table><h2 id="single-branch-tests">Single branch tests</h2><div class="hline"></div><p>The following single branch tests are faster than all bootstrap analysis and recommended for extremely large data sets (e.g., &gt;10,000 taxa):</p><table><thead><tr><th>Option</th><th>Usage and meaning</th></tr></thead><tbody><tr><td><code>-alrt</code></td><td>Specify number of replicates (&gt;=1000) to perform SH-like approximate likelihood ratio test (SH-aLRT) (<a href="https://doi.org/10.1093/sysbio/syq010">Guindon et al., 2010</a>). If number of replicates is set to 0 (<code>-alrt 0</code>), then the parametric aLRT test (<a href="https://doi.org/10.1080/10635150600755453">Anisimova and Gascuel 2006</a>) is performed, instead of SH-aLRT.</td></tr><tr><td><code>-abayes</code></td><td>Perform approximate Bayes test (<a href="https://doi.org/10.1093/sysbio/syr041">Anisimova et al., 2011</a>).</td></tr><tr><td><code>-lbp</code></td><td>Specify number of replicates (&gt;=1000) to perform fast local bootstrap probability method (<a href="http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.476.8552">Adachi and Hasegawa, 1996b</a>).</td></tr></tbody></table><blockquote class="tip"><p><strong>TIP</strong>: One can combine all these tests (also including UFBoot <code>-bb</code> option) within a single IQ-TREE run. Each branch in the resulting tree will be assigned several support values separated by slash (<code>/</code>), where the order of support values is stated in the <code>.iqtree</code> report file.</p></blockquote><h3 id="example-usages-6">Example usages:</h3><ul><li><p>Infer an ML tree and perform SH-aLRT test as well as ultrafast bootstrap with 1000 replicates:</p><pre><code>  iqtree -s data.phy -m TEST -alrt 1000 -bb 1000
</code></pre></li></ul><h2 id="ancestral-sequence-reconstruction">Ancestral sequence reconstruction</h2><div class="hline"></div><p>This feature is newly introduced in version 1.6. You can combine this feature with <code>-te</code> option to determine ancestral sequences along a user-defined tree (Otherwise, IQ-TREE computes ancestral sequences of the ML tree).</p><table><thead><tr><th>Option</th><th>Usage and meaning</th></tr></thead><tbody><tr><td><code>-asr</code></td><td>Write ancestral sequences (by empirical Bayesian method) for all nodes of the tree to <code>.state</code> file.</td></tr><tr><td><code>-asr-min</code></td><td>Specify the minimum threshold of posterior probability to determine the best ancestral state. Default: observed state frequency from the alignment.</td></tr><tr><td><code>-te</code></td><td>Specify a user-defined tree to determine ancestral sequences along this tree. You can assign each node of this tree with a node name, and IQ-TREE will report the ancestral sequences of the corresponding nodes. If nodes do not have names, IQ-TREE will automatically assign node namdes as Node1, Node2, etc.</td></tr></tbody></table><h3 id="example-usages-7">Example usages:</h3><pre><code>iqtree -s example.phy -m JC+G -asr
</code></pre><p>The first few lines of the output file <code>example.phy.state</code> may look like this:</p><pre><code># Ancestral state reconstruction for all nodes in example.phy.treefile
# This file can be read in MS Excel or in R with command:
#   tab=read.table('example.phy.state',header=TRUE)
# Columns are tab-separated with following meaning:
#   Node:  Node name in the tree
#   Site:  Alignment site ID
#   State: Most likely state assignment
#   p_X:   Posterior probability for state X (empirical Bayesian method)
Node    Site    State   p_A     p_C     p_G     p_T
Node2   1       C       0.00004 0.99992 0.00002 0.00002
Node2   2       A       0.92378 0.00578 0.00577 0.06468
Node2   3       A       0.95469 0.02634 0.00675 0.01222
Node2   4       C       0.00002 0.99992 0.00002 0.00004
...
</code></pre><h2 id="tree-topology-tests">Tree topology tests</h2><div class="hline"></div><p>IQ-TREE provides a number of tests for significant topological differences between trees. The AU test implementation in IQ-TREE is much more efficient than the original CONSEL by supporting SSE, AVX and multicore parallelization. Moreover, it is more appropriate than CONSEL for partition analysis by bootstrap resampling sites <em>within</em> partitions, whereas CONSEL is not partition-aware.</p><blockquote><p><strong>NOTE</strong>: There is a discrepancy between IQ-TREE and CONSEL for the AU test: IQ-TREE implements the least-square estimate for p-values whereas CONSEL provides the maximum-likelihood estimate (MLE) for p-values. Hence, the AU p-values might be slightly different. We plan to implement MLE for AU p-values in IQ-TREE soon.</p></blockquote><table><thead><tr><th>Option</th><th>Usage and meaning</th></tr></thead><tbody><tr><td><code>-z</code></td><td>Specify a file containing a set of trees. IQ-TREE will compute the log-likelihoods of all trees.</td></tr><tr><td><code>-zb</code></td><td>Specify the number of RELL (<a href="https://doi.org/10.1007/BF02109483">Kishino et al., 1990</a>) replicates (&gt;=1000) to perform several tree topology tests for all trees passed via <code>-z</code>. The tests include bootstrap proportion (BP), KH test (<a href="https://doi.org/10.1007/BF02100115">Kishino and Hasegawa, 1989</a>), SH test (<a href="https://doi.org/10.1093/oxfordjournals.molbev.a026201">Shimodaira and Hasegawa, 1999</a>) and expected likelihood weights (ELW) (<a href="https://doi.org/10.1098/rspb.2001.1862">Strimmer and Rambaut, 2002</a>).</td></tr><tr><td><code>-zw</code></td><td>Used together with <code>-zb</code> to additionally perform the weighted-KH and weighted-SH tests.</td></tr><tr><td><code>-au</code></td><td>Used together with <code>-zb</code> to additionally perform the approximately unbiased (AU) test (<a href="https://doi.org/10.1080/10635150290069913">Shimodaira, 2002</a>). Note that you have to specify the number of replicates for the AU test via <code>-zb</code>.</td></tr><tr><td><code>-n 0</code></td><td>Only estimate model parameters on an initial parsimony tree and ignore a full tree search to save time.</td></tr><tr><td><code>-te</code></td><td>Specify a fixed user tree to estimate model parameters. Thus it behaves like <code>-n 0</code> but uses a user-defined tree instead of parsimony tree.</td></tr></tbody></table><h3 id="example-usages-8">Example usages:</h3><ul><li><p>Given alignment <code>data.phy</code>, test a set of trees in <code>data.trees</code> using AU test with 10,000 replicates:</p><pre><code>  iqtree -s data.phy -m GTR+G -n 0 -z data.trees -zb 10000 -au 
</code></pre></li><li><p>Same above but for a partitioned data <code>partition.nex</code> and additionally performing weighted test:</p><pre><code>  iqtree -s data.phy -spp partition.nex -n 0 -z data.trees -zb 10000 -au -zw
</code></pre></li></ul><h2 id="constructing-consensus-tree">Constructing consensus tree</h2><div class="hline"></div><p>IQ-TREE provides a fast implementation of consensus tree construction for post analysis:</p><table><thead><tr><th>Option</th><th>Usage and meaning</th></tr></thead><tbody><tr><td><code>-t</code></td><td>Specify a file containing a set of trees.</td></tr><tr><td><code>-con</code></td><td>Compute consensus tree of the trees passed via <code>-t</code>. Resulting consensus tree is written to <code>.contree</code> file.</td></tr><tr><td><code>-net</code></td><td>Compute consensus network of the trees passed via <code>-t</code>. Resulting consensus network is written to <code>.nex</code> file.</td></tr><tr><td><code>-minsup</code></td><td>Specify a minimum threshold (between 0 and 1) to keep branches in the consensus tree. <code>-minsup 0.5</code> means to compute majority-rule consensus tree. <em>DEFAULT: 0 to compute extended majority-rule consensus.</em></td></tr><tr><td><code>-bi</code></td><td>Specify a burn-in, which is the number of beginning trees passed via <code>-t</code> to discard before consensus construction. This is useful e.g. when summarizing trees from MrBayes analysis.</td></tr><tr><td><code>-sup</code></td><td>Specify an input “target” tree file. That means, support values are first extracted from the trees passed via <code>-t</code>, and then mapped onto the target tree. Resulting tree with assigned support values is written to <code>.suptree</code> file. This option is useful to map and compare support values from different approaches onto a single tree.</td></tr><tr><td><code>-suptag</code></td><td>Specify name of a node in <code>-sup</code> target tree. The corresponding node of <code>.suptree</code> will then be assigned with IDs of trees where this node appears. Special option <code>-suptag ALL</code> will assign such IDs for all nodes of the target tree.</td></tr><tr><td><code>-scale</code></td><td>Set the scaling factor of support values for <code>-sup</code> option (default: 100, i.e. percentages)</td></tr><tr><td><code>-prec</code></td><td>Set the precision of support values for <code>-sup</code> option (default: 0)</td></tr></tbody></table><h2 id="computing-robinson-foulds-distance">Computing Robinson-Foulds distance</h2><div class="hline"></div><p>IQ-TREE provides a fast implementation of Robinson-Foulds (RF) distance computation between trees:</p><table><thead><tr><th>Option</th><th>Usage and meaning</th></tr></thead><tbody><tr><td><code>-t</code></td><td>Specify a file containing a set of trees.</td></tr><tr><td><code>-rf</code></td><td>Specify a second set of trees. IQ-TREE computes all pairwise RF distances between two tree sets passed via <code>-t</code> and <code>-rf</code></td></tr><tr><td><code>-rf_all</code></td><td>Compute all-to-all RF distances between all trees passed via <code>-t</code></td></tr><tr><td><code>-rf_adj</code></td><td>Compute RF distances between adjacent trees passed via <code>-t</code></td></tr></tbody></table><h3 id="example-usages-9">Example usages:</h3><ul><li><p>Compute the pairwise RF distances between 2 sets of trees:</p><pre><code>  iqtree -rf tree_set1 tree_set2
</code></pre></li><li><p>Compute the all-to-all RF distances within a set of trees:</p><pre><code>  iqtree -rf_all tree_set
</code></pre></li></ul><h2 id="generating-random-trees">Generating random trees</h2><div class="hline"></div><p>IQ-TREE provides several random tree generation models:</p><table><thead><tr><th>Option</th><th>Usage and meaning</th></tr></thead><tbody><tr><td><code>-r</code></td><td>Specify number of taxa. IQ-TREE will create a random tree under Yule-Harding model with specified number of taxa</td></tr><tr><td><code>-ru</code></td><td>Like <code>-r</code>, but a random tree is created under uniform model.</td></tr><tr><td><code>-rcat</code></td><td>Like <code>-r</code>, but a random caterpillar tree is created.</td></tr><tr><td><code>-rbal</code></td><td>Like <code>-r</code>, but a random balanced tree is created.</td></tr><tr><td><code>-rcsg</code></td><td>Like <code>-r</code>, bur a random circular split network is created.</td></tr><tr><td><code>-rlen</code></td><td>Specify three numbers: minimum, mean and maximum branch lengths of the random tree. <em>DEFAULT: <code>-rlen 0.001 0.1 0.999</code></em></td></tr></tbody></table><h3 id="example-usages-10">Example usages:</h3><ul><li><p>Generate a 100-taxon random tree into the file <code>100.tree</code> under the Yule Harding model:</p><pre><code>  iqtree -r 100 100.tree 
</code></pre></li><li><p>Generate 100-taxon random tree with mean branch lengths of 0.2 and branch lengths are between 0.05 and 0.3:</p><pre><code>  iqtree -r 100 -rlen 0.05 0.2 0.3 100.tree 
</code></pre></li><li><p>Generate a random tree under uniform model:</p><pre><code>  iqtree -ru 100 100.tree 
</code></pre></li><li><p>Generate a random tree where taxon labels follow an alignment:</p><pre><code>  iqtree -s example.phy -r 17 example.random.tree 
</code></pre></li></ul><p>Note that, you still need to specify the <code>-r</code> option being equal to the number of taxa in the alignment.</p><h2 id="miscellaneous-options">Miscellaneous options</h2><div class="hline"></div><table><thead><tr><th>Option</th><th>Usage and meaning</th></tr></thead><tbody><tr><td><code>-alninfo</code></td><td>Print alignment site statistics to <code>.alninfo</code> file.</td></tr><tr><td><code>-blfix</code></td><td>Fix branch lengths of tree passed via <code>-t</code> or <code>-te</code>. This is useful to evaluate the log-likelihood of an input tree with fixed tolopogy and branch lengths. <em>DEFAULT: OFF</em></td></tr><tr><td><code>-blmin</code></td><td>Specify minimum branch length. Default: the smaller of 0.000001 and 0.1/alignment_length.</td></tr><tr><td><code>-blmax</code></td><td>Specify the maximum branch length. Default: 10</td></tr><tr><td><code>-czb</code></td><td>Collapse near zero branches, so that the final tree may be multifurcating. This is useful for bootstrapping in the presence of polytomy to reduce bootstrap supports of short branches.</td></tr><tr><td><code>-me</code></td><td>Specify the log-likelihood epsilon for final model parameter estimation (Default: 0.01). With <code>-fast</code> option, the epsilon is raised to 0.05.</td></tr><tr><td><code>-wpl</code></td><td>Write partition log-likelihoods to <code>.partlh</code> file. Only effective with partition model.</td></tr><tr><td><code>-wspr</code></td><td>Write site posterior probabilities per rate category to <code>.siteprob</code> file.</td></tr><tr><td><code>-wspm</code></td><td>Write site posterior probabilities per mixture class to <code>.siteprob</code> file.</td></tr><tr><td><code>-wspmr</code></td><td>Write site posterior probabilities per mixture class and rate category to <code>.siteprob</code> file.</td></tr><tr><td><code>-wsl</code></td><td>Write site log-likelihoods to <code>.sitelh</code> file in <a href="http://www.tree-puzzle.de/">TREE-PUZZLE</a> format. Such file can then be passed on to <a href="http://www.sigmath.es.osaka-u.ac.jp/shimo-lab/prog/consel/">CONSEL</a> for further tree tests.</td></tr><tr><td><code>-wslr</code></td><td>Write site log-likelihoods per rate category to <code>.sitelh</code> file.</td></tr><tr><td><code>-wslm</code></td><td>Write site log-likelihoods per mixture class to <code>.sitelh</code> file.</td></tr><tr><td><code>-wslmr</code></td><td>Write site log-likelihoods per mixture class and rate category to <code>.sitelh</code> file.</td></tr><tr><td><code>-wt</code></td><td>Turn on writing all locally optimal trees into <code>.treels</code> file.</td></tr><tr><td><code>-fconst</code></td><td>Specify a list of comma-separated integer numbers. The number of entries should be equal to the number of states in the model (e.g. 4 for DNA and 20 for protein). IQ-TREE will then add a number of constant sites accordingly. For example, <code>-fconst 10,20,15,40</code> will add 10 constant sites of all A, 20 constant sites of all C, 15 constant sites of all G and 40 constant sites of all T into the alignment.</td></tr></tbody></table><h3 id="example-usages-11">Example usages:</h3><ul><li><p>Print alignment information about parsimony informative sites:</p><pre><code>  iqtree -s example.phy -m JC -n 0 -alninfo
</code></pre></li></ul><p>The first few lines of the output file <code>example.phy.alninfo</code> may look like this:</p><pre><code># Alignment site statistics
# This file can be read in MS Excel or in R with command:
#   tab=read.table('example.phy.alninfo',header=TRUE)
# Columns are tab-separated with following meaning:
#   Site:   Site ID
#   Stat:   Statistic, I=informative, C=constant, c=constant+ambiguous,
#           U=Uninformative but not constant, -=all-gaps
Site    Stat
1       U
2       I
3       I
4       U
5       U
</code></pre><ul><li><p>Print site log-likelihood and posterior probability for each Gamma rate category:</p><pre><code>  iqtree -s example.phy -m JC+G -wspr -wslr  -n 0
</code></pre></li></ul><p>The first few lines of the output file example.phy.siteprob (printed by <code>-wspr</code> option) may look like this:</p><pre><code>Site    p1      p2      p3      p4
1       0.180497        0.534405        0.281   0.00409816
2       4.73239e-05     0.0373409       0.557705        0.404907
3       1.23186e-08     0.000426294     0.0672021       0.932372
4       0.180497        0.534405        0.281   0.00409816
5       0.180497        0.534405        0.281   0.00409816
</code></pre><p>where <code>pX</code> is the probability of the site falling into rate category <code>X</code>.</p><p>The first few lines of the output file example.phy.sitelh (printed by <code>-wslr</code> option) may look like this:</p><pre><code># Site likelihood per rate/mixture category
# This file can be read in MS Excel or in R with command:
#   tab=read.table('example.phy.sitelh',header=TRUE,fill=TRUE)
# Columns are tab-separated with following meaning:
#   Site:   Alignment site ID
#   LnL:    Logarithm of site likelihood
#           Thus, sum of LnL is equal to tree log-likelihood
#   LnLW_k: Logarithm of (category-k site likelihood times category-k weight)
#           Thus, sum of exp(LnLW_k) is equal to exp(LnL)
Site    LnL     LnLW_1  LnLW_2  LnLW_3  LnLW_4
1       -7.0432 -8.7552 -7.6698 -8.3126 -12.5404
2       -17.5900        -27.5485        -20.8776        -18.1739        -18.4941
3       -20.3313        -38.5435        -28.0917        -23.0314        -20.4014
4       -7.0432 -8.7552 -7.6698 -8.3126 -12.5404
5       -7.0432 -8.7552 -7.6698 -8.3126 -12.5404
</code></pre><div class="spacing"></div></div><div class="col-md-3"><div class="well well-sm"> <a href="http://www.iqtree.org/doc/iqtree-doc.pdf"><i class="fa fa-file-pdf-o fa-lg" aria-hidden="true"></i> Download Tutorials and Manual in one PDF</a></div><div class="well well-sm" id="sidebar"><h4>Documentation</h4><p> <a href="http://www.iqtree.org/doc/Home"> <i class="fa fa-info-circle"></i> Introduction </a></p><p> <a href="http://www.iqtree.org/doc/Quickstart"> <i class="fa fa-info-circle"></i> Getting Started </a></p><p> <a href="http://www.iqtree.org/doc/Web-Server-Tutorial"> <i class="fa fa-info-circle"></i> Web Server Tutorial </a></p><p> <a href="http://www.iqtree.org/doc/Tutorial"> <i class="fa fa-info-circle"></i> Beginner's Tutorial </a></p><p> <a href="http://www.iqtree.org/doc/Advanced-Tutorial"> <i class="fa fa-info-circle"></i> Advanced Tutorial </a></p><p> <a href="http://www.iqtree.org/doc/Assessing-Phylogenetic-Assumptions"> <i class="fa fa-info-circle"></i> Assessing Phylogenetic Assumptions </a></p><p> <a href="http://www.iqtree.org/doc/Concordance-Factor"> <i class="fa fa-info-circle"></i> Concordance Factor </a></p><p> <a href="http://www.iqtree.org/doc/Dating"> <i class="fa fa-info-circle"></i> Phylogenetic Dating </a></p><p> <a href="http://www.iqtree.org/doc/Rootstrap"> <i class="fa fa-info-circle"></i> Rooting phylogenies </a></p><p> <a href="http://www.iqtree.org/doc/Estimating-amino-acid-substitution-models"> <i class="fa fa-info-circle"></i> Estimating amino acid substitution models </a></p><p> <a href="http://www.iqtree.org/doc/AliSim"> <i class="fa fa-info-circle"></i> Simulating sequence alignments </a></p><p> <a href="http://www.iqtree.org/doc/Frequently-Asked-Questions"> <i class="fa fa-question-circle"></i> Frequently Asked Questions </a></p><p> <a href="http://www.iqtree.org/doc/Polymorphism-Aware-Models"> <i class="fa fa-book"></i> Polymorphism-Aware Models </a></p><p> <i class="fa fa-book"></i> Command Reference</p><ul><li><a href="http://www.iqtree.org/doc/Command-Reference#general-options">General options</a></li><li><a href="http://www.iqtree.org/doc/Command-Reference#checkpointing-to-resume-stopped-run">Checkpointing to resume stopped run</a></li><li><a href="http://www.iqtree.org/doc/Command-Reference#likelihood-mapping-analysis">Likelihood mapping analysis</a></li><li><a href="http://www.iqtree.org/doc/Command-Reference#automatic-model-selection">Automatic model selection</a></li><li><a href="http://www.iqtree.org/doc/Command-Reference#specifying-substitution-models">Specifying substitution models</a></li><li><a href="http://www.iqtree.org/doc/Command-Reference#rate-heterogeneity">Rate heterogeneity</a></li><li><a href="http://www.iqtree.org/doc/Command-Reference#partition-model-options">Partition model options</a></li><li><a href="http://www.iqtree.org/doc/Command-Reference#site-specific-frequency-model-options">Site-specific frequency model options</a></li><li><a href="http://www.iqtree.org/doc/Command-Reference#tree-search-parameters">Tree search parameters</a></li><li><a href="http://www.iqtree.org/doc/Command-Reference#ultrafast-bootstrap-parameters">Ultrafast bootstrap parameters</a></li><li><a href="http://www.iqtree.org/doc/Command-Reference#nonparametric-bootstrap">Nonparametric bootstrap</a></li><li><a href="http://www.iqtree.org/doc/Command-Reference#single-branch-tests">Single branch tests</a></li><li><a href="http://www.iqtree.org/doc/Command-Reference#ancestral-sequence-reconstruction">Ancestral sequence reconstruction</a></li><li><a href="http://www.iqtree.org/doc/Command-Reference#tree-topology-tests">Tree topology tests</a></li><li><a href="http://www.iqtree.org/doc/Command-Reference#constructing-consensus-tree">Constructing consensus tree</a></li><li><a href="http://www.iqtree.org/doc/Command-Reference#computing-robinson-foulds-distance">Robinson-Foulds distance</a></li><li><a href="http://www.iqtree.org/doc/Command-Reference#generating-random-trees">Generating random trees</a></li><li><a href="http://www.iqtree.org/doc/Command-Reference#miscellaneous-options">Miscellaneous options</a></li></ul><p> <a href="http://www.iqtree.org/doc/Compilation-Guide"> <i class="fa fa-book"></i> Compilation Guide </a></p><p> <a href="http://www.iqtree.org/doc/Developer-Guide"> <i class="fa fa-codepen"></i> Developer Guide </a></p><p> <a href="http://www.iqtree.org/doc/Analyzing-Big-Data"> <i class="fa fa-info-circle"></i> Analyzing Big Data </a></p></div></div></div></div><div id="footerwrap"><div class="container"><div class="row"><div class="col-sm-4"><h4>Software</h4><p> <a href="http://www.iqtree.org/release/">IQ-TREE release notes</a><br> <a href="https://github.com/Cibiv/IQ-TREE">GitHub repository</a><br> <a href="http://www.iqtree.org/about/">Development team</a><br> <a href="http://www.somsubhra.com/github-release-stats/?username=Cibiv&amp;repository=IQ-TREE">Download statistics</a><br></p></div><div class="col-sm-4"><h4>User support</h4><p> <a href="http://www.iqtree.org/doc/Frequently-Asked-Questions">Frequently asked questions</a><br> <a href="http://www.iqtree.org/doc/">User documentation</a><br> <a href="https://groups.google.com/d/forum/iqtree">IQ-TREE Google group</a><br> <a href="http://iqtree.cibiv.univie.ac.at/" target="_blank">IQ-TREE online web service</a><br></p></div><div class="col-sm-4"><h4>Links</h4><p> <a href="https://cs.anu.edu.au/">Research School of Computer Science, ANU</a><br> <a href="http://www.cibiv.at/">Center for Integrative Bioinformatics Vienna</a><br> <a href="http://validator.w3.org/check/referer">Link Checker</a><br></p></div></div></div></div><div class="alert alert-dismissible text-center cookiealert show" role="alert"><div class="cookiealert-container"> We use cookies to ensure you get the best experience on our website <button type="button" class="btn btn-primary btn-sm acceptcookies" aria-label="Close"> I agree </button></div></div><script src="./Command Reference_files/jquery.min.js.download"></script> <script src="./Command Reference_files/bootstrap.min.js.download"></script> <script src="./Command Reference_files/retina-1.1.0.js.download"></script> <script src="./Command Reference_files/jquery.hoverdir.js.download"></script> <script src="./Command Reference_files/jquery.isotope.min.js.download"></script> <script src="./Command Reference_files/cookiealert-standalone.js.download"></script> <script> (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){ (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o), m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m) })(window,document,'script','//www.google-analytics.com/analytics.js','ga'); ga('create', 'UA-38693379-1', 'auto'); ga('send', 'pageview'); </script> <script> // Portfolio (function($) { "use strict"; var $container = $('.portfolio'), $items = $container.find('.portfolio-item'), portfolioLayout = 'fitRows'; if( $container.hasClass('portfolio-centered') ) { portfolioLayout = 'masonry'; } $container.isotope({ filter: '*', animationEngine: 'best-available', layoutMode: portfolioLayout, animationOptions: { duration: 750, easing: 'linear', queue: false }, masonry: { } }, refreshWaypoints()); function refreshWaypoints() { setTimeout(function() { }, 1000); } $('nav.portfolio-filter ul a').on('click', function() { var selector = $(this).attr('data-filter'); $container.isotope({ filter: selector }, refreshWaypoints()); $('nav.portfolio-filter ul a').removeClass('active'); $(this).addClass('active'); return false; }); function getColumnNumber() { var winWidth = $(window).width(), columnNumber = 1; if (winWidth > 1200) { columnNumber = 5; } else if (winWidth > 950) { columnNumber = 4; } else if (winWidth > 600) { columnNumber = 3; } else if (winWidth > 400) { columnNumber = 2; } else if (winWidth > 250) { columnNumber = 1; } return columnNumber; } function setColumns() { var winWidth = $(window).width(), columnNumber = getColumnNumber(), itemWidth = Math.floor(winWidth / columnNumber); $container.find('.portfolio-item').each(function() { $(this).css( { width : itemWidth + 'px' }); }); } function setPortfolio() { setColumns(); $container.isotope('reLayout'); } $container.imagesLoaded(function () { setPortfolio(); }); $(window).on('resize', function () { setPortfolio(); }); })(jQuery); </script>
<table cellspacing="0" cellpadding="0" class="gstl_50 gssb_c" style="width: 213px; display: none; top: 94px; position: absolute; left: 506px;"><tbody><tr><td class="gssb_f"></td><td class="gssb_e" style="width: 100%;"></td></tr></tbody></table></body></html>