<!DOCTYPE html>
<!-- saved from url=(0040)http://www.iqtree.org/doc/Complex-Models -->
<html lang="en" class="csstransforms csstransforms3d csstransitions"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1"><meta name="description" content=""><meta name="author" content=""><title>Complex Models</title><link href="./Complex Models_files/bootstrap.min.css" rel="stylesheet"><link href="./Complex Models_files/style.css" rel="stylesheet"><link href="./Complex Models_files/font-awesome.min.css" rel="stylesheet"><link rel="stylesheet" href="./Complex Models_files/cookiealert.css"><script async="" src="./Complex Models_files/async-ads.js.download"></script><script async="" src="./Complex Models_files/analytics.js.download"></script><script src="./Complex Models_files/cse_element__en.js.download" type="text/javascript"></script><link type="text/css" rel="stylesheet" href="./Complex Models_files/default+en.css"><link type="text/css" rel="stylesheet" href="./Complex Models_files/default.css"><style type="text/css">.gsc-control-cse{font-family:times new roman, arial, serif}.gsc-control-cse .gsc-table-result{font-family:times new roman, arial, serif}.gsc-refinementsGradient{background:linear-gradient(to left,rgba(255,255,255,1),rgba(255,255,255,0))}.gsc-control-cse{border-color:#337AB7;background-color:#337AB7}input.gsc-input,.gsc-input-box,.gsc-input-box-hover,.gsc-input-box-focus{border-color:#FFFFFF}.gsc-search-button-v2,.gsc-search-button-v2:hover,.gsc-search-button-v2:focus{border-color:#4CAE4C;background-color:#5CB85C;background-image:none;filter:none}.gsc-search-button-v2 svg{fill:#FFFFFF}.gsc-tabHeader.gsc-tabhActive,.gsc-refinementHeader.gsc-refinementhActive{color:#CCCCCC;border-color:#CCCCCC;background-color:#FFFFFF}.gsc-tabHeader.gsc-tabhInactive,.gsc-refinementHeader.gsc-refinementhInactive{color:#CCCCCC;border-color:#CCCCCC;background-color:#FFFFFF}.gsc-webResult.gsc-result,.gsc-results .gsc-imageResult{border-color:#F5F5F5;background-color:#F5F5F5}.gsc-webResult.gsc-result:hover{border-color:#F5F5F5;background-color:#F5F5F5}.gs-webResult.gs-result a.gs-title:link,.gs-webResult.gs-result a.gs-title:link b,.gs-imageResult a.gs-title:link,.gs-imageResult a.gs-title:link b{color:#0000CC}.gs-webResult.gs-result a.gs-title:visited,.gs-webResult.gs-result a.gs-title:visited b,.gs-imageResult a.gs-title:visited,.gs-imageResult a.gs-title:visited b{color:#0000CC}.gs-webResult.gs-result a.gs-title:hover,.gs-webResult.gs-result a.gs-title:hover b,.gs-imageResult a.gs-title:hover,.gs-imageResult a.gs-title:hover b{color:#0000CC}.gs-webResult.gs-result a.gs-title:active,.gs-webResult.gs-result a.gs-title:active b,.gs-imageResult a.gs-title:active,.gs-imageResult a.gs-title:active b{color:#0000CC}.gsc-cursor-page{color:#0000CC}a.gsc-trailing-more-results:link{color:#0000CC}.gs-webResult:not(.gs-no-results-result):not(.gs-error-result) .gs-snippet,.gs-fileFormatType{color:#000000}.gs-webResult div.gs-visibleUrl{color:#008000}.gs-webResult div.gs-visibleUrl-short{color:#008000}.gsc-cursor-box{border-color:#F5F5F5}.gsc-results .gsc-cursor-box .gsc-cursor-page{border-color:#CCCCCC;background-color:#FFFFFF;color:#CCCCCC}.gsc-results .gsc-cursor-box .gsc-cursor-current-page{border-color:#CCCCCC;background-color:#FFFFFF;color:#CCCCCC}.gsc-webResult.gsc-result.gsc-promotion{border-color:#336699;background-color:#FFFFFF}.gsc-completion-title{color:#0000CC}.gsc-completion-snippet{color:#000000}.gs-promotion a.gs-title:link,.gs-promotion a.gs-title:link *,.gs-promotion .gs-snippet a:link{color:#0000CC}.gs-promotion a.gs-title:visited,.gs-promotion a.gs-title:visited *,.gs-promotion .gs-snippet a:visited{color:#0000CC}.gs-promotion a.gs-title:hover,.gs-promotion a.gs-title:hover *,.gs-promotion .gs-snippet a:hover{color:#0000CC}.gs-promotion a.gs-title:active,.gs-promotion a.gs-title:active *,.gs-promotion .gs-snippet a:active{color:#0000CC}.gs-promotion .gs-snippet,.gs-promotion .gs-title .gs-promotion-title-right,.gs-promotion .gs-title .gs-promotion-title-right *{color:#000000}.gs-promotion .gs-visibleUrl,.gs-promotion .gs-visibleUrl-short{color:#008000}.gcsc-find-more-on-google{color:#0000CC}.gcsc-find-more-on-google-magnifier{fill:#0000CC}</style><style type="text/css">.gscb_a{display:inline-block;font:27px/13px arial,sans-serif}.gsst_a .gscb_a{color:#a1b9ed;cursor:pointer}.gsst_a:hover .gscb_a,.gsst_a:focus .gscb_a{color:#36c}.gsst_a{display:inline-block}.gsst_a{cursor:pointer;padding:0 4px}.gsst_a:hover{text-decoration:none!important}.gsst_b{font-size:16px;padding:0 2px;position:relative;user-select:none;-webkit-user-select:none;white-space:nowrap}.gsst_e{vertical-align:middle;opacity:0.55;}.gsst_a:hover .gsst_e,.gsst_a:focus .gsst_e{opacity:0.72;}.gsst_a:active .gsst_e{opacity:1;}.gsst_f{background:white;text-align:left}.gsst_g{background-color:white;border:1px solid #ccc;border-top-color:#d9d9d9;box-shadow:0 2px 4px rgba(0,0,0,0.2);-webkit-box-shadow:0 2px 4px rgba(0,0,0,0.2);margin:-1px -3px;padding:0 6px}.gsst_h{background-color:white;height:1px;margin-bottom:-1px;position:relative;top:-1px}.gsib_a{width:100%;padding:4px 6px 0}.gsib_a,.gsib_b{vertical-align:top}.gssb_c{border:0;position:absolute;z-index:989}.gssb_e{border:1px solid #ccc;border-top-color:#d9d9d9;box-shadow:0 2px 4px rgba(0,0,0,0.2);-webkit-box-shadow:0 2px 4px rgba(0,0,0,0.2);cursor:default}.gssb_f{visibility:hidden;white-space:nowrap}.gssb_k{border:0;display:block;position:absolute;top:0;z-index:988}.gsdd_a{border:none!important}.gsq_a{padding:0}.gssb_a{padding:0 7px}.gssb_a,.gssb_a td{white-space:nowrap;overflow:hidden;line-height:22px}#gssb_b{font-size:11px;color:#36c;text-decoration:none}#gssb_b:hover{font-size:11px;color:#36c;text-decoration:underline}.gssb_g{text-align:center;padding:8px 0 7px;position:relative}.gssb_h{font-size:15px;height:28px;margin:0.2em;-webkit-appearance:button}.gssb_i{background:#eee}.gss_ifl{visibility:hidden;padding-left:5px}.gssb_i .gss_ifl{visibility:visible}a.gssb_j{font-size:13px;color:#36c;text-decoration:none;line-height:100%}a.gssb_j:hover{text-decoration:underline}.gssb_l{height:1px;background-color:#e5e5e5}.gssb_m{color:#000;background:#fff}.gssb_a{padding:0 9px}.gsib_a{padding:5px 9px 4px 9px}.gscb_a{line-height:27px}.gssb_e{border:0}.gssb_l{margin:5px 0}input.gsc-input::-webkit-input-placeholder{font-size:14px}input.gsc-input:-moz-placeholder{font-size:14px}input.gsc-input::-moz-placeholder{font-size:14px}input.gsc-input:-ms-input-placeholder{font-size:14px}input.gsc-input:focus::-webkit-input-placeholder{color:transparent}input.gsc-input:focus:-moz-placeholder{color:transparent}input.gsc-input:focus::-moz-placeholder{color:transparent}input.gsc-input:focus:-ms-input-placeholder{color:transparent}.gssb_c .gsc-completion-container{position:static}.gssb_c{z-index:5000}.gsc-completion-container table{background:transparent;font-size:inherit;font-family:inherit}.gssb_c > tbody > tr,.gssb_c > tbody > tr > td,.gssb_d,.gssb_d > tbody > tr,.gssb_d > tbody > tr > td,.gssb_e,.gssb_e > tbody > tr,.gssb_e > tbody > tr > td{padding:0;margin:0;border:0}.gssb_a table,.gssb_a table tr,.gssb_a table tr td{padding:0;margin:0;border:0}</style></head><body style=""><div class="navbar navbar-default navbar-fixed-top" role="navigation"><div class="container"><div class="navbar-header"> <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar"></span> <span class="icon-bar"></span> <span class="icon-bar"></span> </button> <a class="navbar-brand" href="http://www.iqtree.org/">IQ-TREE</a></div><div class="navbar-collapse collapse navbar-right"><ul class="nav navbar-nav"><li><a data-toggle="collapse" data-target=".navbar-collapse" href="http://www.iqtree.org/#download">Download</a></li><li><a href="http://www.iqtree.org/news/">News</a></li><li class="dropdown"> <a class="dropdown-toggle" data-toggle="dropdown" href="http://www.iqtree.org/doc/Complex-Models#">Web server <span class="caret"></span></a><ul class="dropdown-menu"><li><a href="http://iqtree.cibiv.univie.ac.at/" target="_blank">CIBIV, Austria</a></li><li><a href="https://www.hiv.lanl.gov/content/sequence/IQTREE/iqtree.html" target="_blank">Los Alamos Lab, USA</a></li><li><a href="https://www.phylo.org/" target="_blank">CIPRES Gateway, USA</a></li></ul></li><li><a href="http://www.iqtree.org/doc/">Docs</a></li><li><a href="http://www.iqtree.org/workshop/">Workshop</a></li><li><a href="http://www.iqtree.org/about/">About</a></li></ul></div></div></div><div id="blue"><div class="container"><div class="row"><div class="col-sm-6"><h3> Manual</h3></div><div class="col-sm-6"> <script type="text/javascript" async="" src="./Complex Models_files/f.txt"></script><script> (function() { var cx = '003669294240048101214:pace-yz0kou'; var gcse = document.createElement('script'); gcse.type = 'text/javascript'; gcse.async = true; gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') + '//cse.google.com/cse.js?cx=' + cx; var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(gcse, s); })(); </script> <div id="___gcse_0"><div class="gsc-control-cse gsc-control-cse-en"><div class="gsc-control-wrapper-cse" dir="ltr"><form class="gsc-search-box gsc-search-box-tools" accept-charset="utf-8"><table cellspacing="0" cellpadding="0" class="gsc-search-box"><tbody><tr><td class="gsc-input"><div class="gsc-input-box" id="gsc-iw-id1"><table cellspacing="0" cellpadding="0" id="gs_id50" class="gstl_50 gsc-input" style="width: 100%; padding: 0px;"><tbody><tr><td id="gs_tti50" class="gsib_a"><input autocomplete="off" type="text" size="10" class="gsc-input" name="search" title="search" id="gsc-i-id1" dir="ltr" spellcheck="false" style="width: 100%; padding: 0px; border: none; margin: 0px; height: auto; background: url(&quot;https://www.google.com/cse/static/images/1x/en/branding.png&quot;) left center no-repeat rgb(255, 255, 255); outline: none;"></td><td class="gsib_b"><div class="gsst_b" id="gs_st50" dir="ltr"><a class="gsst_a" href="javascript:void(0)" style="display: none;" title="Clear search box" role="button"><span class="gscb_a" id="gs_cb50" aria-hidden="true">×</span></a></div></td></tr></tbody></table></div></td><td class="gsc-search-button"><button class="gsc-search-button gsc-search-button-v2"><svg width="13" height="13" viewBox="0 0 13 13"><title>search</title><path d="m4.8495 7.8226c0.82666 0 1.5262-0.29146 2.0985-0.87438 0.57232-0.58292 0.86378-1.2877 0.87438-2.1144 0.010599-0.82666-0.28086-1.5262-0.87438-2.0985-0.59352-0.57232-1.293-0.86378-2.0985-0.87438-0.8055-0.010599-1.5103 0.28086-2.1144 0.87438-0.60414 0.59352-0.8956 1.293-0.87438 2.0985 0.021197 0.8055 0.31266 1.5103 0.87438 2.1144 0.56172 0.60414 1.2665 0.8956 2.1144 0.87438zm4.4695 0.2115 3.681 3.6819-1.259 1.284-3.6817-3.7 0.0019784-0.69479-0.090043-0.098846c-0.87973 0.76087-1.92 1.1413-3.1207 1.1413-1.3553 0-2.5025-0.46363-3.4417-1.3909s-1.4088-2.0686-1.4088-3.4239c0-1.3553 0.4696-2.4966 1.4088-3.4239 0.9392-0.92727 2.0864-1.3969 3.4417-1.4088 1.3553-0.011889 2.4906 0.45771 3.406 1.4088 0.9154 0.95107 1.379 2.0924 1.3909 3.4239 0 1.2126-0.38043 2.2588-1.1413 3.1385l0.098834 0.090049z"></path></svg></button></td><td class="gsc-clear-button"><div class="gsc-clear-button" title="clear results">&nbsp;</div></td></tr></tbody></table></form><div class="gsc-results-wrapper-overlay"><div class="gsc-results-close-btn" tabindex="0"></div><div class="gsc-positioningWrapper"><div class="gsc-tabsAreaInvisible"><div aria-label="refinement" role="tab" class="gsc-tabHeader gsc-inline-block gsc-tabhActive">Custom Search</div><span class="gs-spacer"> </span></div></div><div class="gsc-positioningWrapper"><div class="gsc-tabsAreaInvisible"></div></div><div class="gsc-above-wrapper-area-invisible"><table cellspacing="0" cellpadding="0" class="gsc-above-wrapper-area-container"><tbody><tr><td class="gsc-result-info-container"><div class="gsc-result-info-invisible"></div></td><td class="gsc-orderby-container"><div class="gsc-orderby-invisible"><div class="gsc-orderby-label gsc-inline-block">Sort by:</div><div class="gsc-option-menu-container gsc-inline-block"><div class="gsc-selected-option-container gsc-inline-block"><div class="gsc-selected-option">Relevance</div><div class="gsc-option-selector"></div></div><div class="gsc-option-menu-invisible"><div class="gsc-option-menu-item gsc-option-menu-item-highlighted"><div class="gsc-option">Relevance</div></div><div class="gsc-option-menu-item"><div class="gsc-option">Date</div></div></div></div></div></td></tr></tbody></table></div><div class="gsc-adBlockInvisible"></div><div class="gsc-wrapper"><div class="gsc-adBlockInvisible"></div><div class="gsc-resultsbox-invisible"><div class="gsc-resultsRoot gsc-tabData gsc-tabdActive"><div><div class="gsc-expansionArea"></div></div></div></div></div></div><div class="gsc-modal-background-image" tabindex="0"></div></div></div></div></div></div></div></div><div class="container mtb"><div class="row"><div class="col-md-9"><p><em>Last update: Aug 8, 2019, Contributors: Dominik Schrempf, Jana Trifinopoulos, Minh Bui</em></p><h1 id="complex-models">Complex models</h1><p>Complex models such as partition and mixture models.</p><p>This document gives detailed descriptions of complex maximum-likelihood models available in IQ-TREE. It is assumed that you know the <a href="http://www.iqtree.org/doc/Substitution-Models">basic substitution models</a> already.</p><h2 id="partition-models">Partition models</h2><div class="hline"></div><p>Partition models are intended for phylogenomic (e.g., multi-gene) alignments, which allow each partition to have its own substitution models and evolutionary rates. IQ-TREE supports three types of partition models:</p><ol><li><em>Edge-equal</em> partition model with equal branch lengths: All partitions share the same set of branch lengths.</li><li><em>Edge-proportional</em> partition model with proportional branch lengths: Like above but each partition has its own partition specific rate, that rescales all its branch lengths. This model accomodates different evolutionary rates between partitions (e.g. between 1st, 2nd, and 3rd codon positions).</li><li><em>Edge-unlinked</em> partition model: Each partition has its own set of branch lengths. This is the most parameter-rich partition model, that accounts for e.g., <em>heterotachy</em> (<a href="http://mbe.oxfordjournals.org/content/19/1/1.full">Lopez et al., 2002</a>).</li></ol><blockquote class="tip"><p><strong>TIP</strong>: The edge-equal partition model is typically unrealistic as it does not account for different evolutionary speeds between partitions, whereas the edge-unlinked partition model can be overfitting if there are many short partitions. Therefore, the edge-proportional partition model is recommended for a typical analysis.</p></blockquote><h3 id="partition-file-format">Partition file format</h3><p>To apply partition models users must first prepare a partition file in RAxML-style or NEXUS format. The RAxML-style is defined by the RAxML software and may look like:</p><pre><code>DNA, part1 = 1-100
DNA, part2 = 101-384
</code></pre><p>This means two DNA partitions of an alignment, where one groups aligment sites <code>1-100</code> into <code>part1</code> and <code>101-384</code> into <code>part2</code>.</p><p>The NEXUS format is more complex but more powerful. For example, the above partition scheme may look like:</p><pre><code>#nexus
begin sets;
    charset part1 = 1-100;
    charset part2 = 101-384;
    charpartition mine = HKY+G:part1, GTR+I+G:part2;
end;
</code></pre><p>The first line contains the keyword <code>#nexus</code> to indicate a NEXUS file. It has a <code>sets</code> block, which contains two character sets (<code>charset</code> command) named <code>part1</code> and <code>part2</code>. Furthermore, with the <code>charpartition</code> command we set the model <code>HKY+G</code> for <code>part1</code> and <code>GTR+I+G</code> for <code>part2</code>. This is not possible with the RAxML-style format (i.e., one cannot specify <code>+G</code> rate model for one partition and <code>+I+G</code> rate model for the other partition).</p><blockquote class="tip"><p><strong>TIP</strong>: IQ-TREE fully supports mixed rate heterogeneity types types between partitions (see above example).</p></blockquote><p>One can also specify non-consecutive sites of a partition, e.g. under RAxML-style format:</p><pre><code>DNA, part1 = 1-100, 250-384
DNA, part2 = 101-249\3, 102-249\3
DNA, part3 = 103-249\3
</code></pre><p>or under NEXUS format:</p><pre><code>#nexus
begin sets;
    charset part1 = 1-100 250-384;
    charset part2 = 101-249\3 102-249\3;
    charset part3 = 103-249\3;
end;
</code></pre><p>This means, <code>part2</code> contains sites 101, 102, 104, 105, 107, …, 246, 248, 249; whereas <code>part3</code> contains sites 103, 106, …, 247. This is useful to specify partitions corresponding to 1st, 2nd and 3rd codon positions.</p><p>Moreover, the NEXUS file allows each partition to come from a separate alignment file (not possible under RAxML-style format) with e.g.:</p><pre><code>#nexus
begin sets;
    charset part1 = aln1.phy: 1-100\3 201-300;
    charset part2 = aln1.phy: 101-200;
    charset part3 = aln2.phy: *;
    charpartition mine = HKY:part1, GTR+G:part2, WAG+I+G:part3;
end;
</code></pre><p>Here, <code>part1</code> and <code>part2</code> correspond to sub-alignments of <code>aln1.phy</code> file and <code>part3</code> is the entire alignment file <code>aln2.phy</code>. Note that <code>aln2.phy</code> is a protein alignment in this example.</p><blockquote class="tip"><p><strong>TIP</strong>: IQ-TREE fully supports mixed data types between partitions.</p></blockquote><p>If you want to specify codon model for a partition, use the <code>CODON</code> keyword (otherwise, the partition may be detected as DNA):</p><pre><code>#nexus
begin sets;
    charset part1 = aln1.phy:CODON, 1-300;
    charset part2 = aln1.phy: 301-400;
    charset part3 = aln2.phy: *;
    charpartition mine = GY:part1, GTR+G:part2, WAG+I+G:part3;
end;
</code></pre><p>Note that this assumes <code>part1</code> has standard genetic code. If not, append <code>CODON</code> with <a href="http://www.iqtree.org/doc/Substitution-Models#codon-models">the right genetic code ID</a>.</p><h3 id="partitioned-analysis">Partitioned analysis</h3><p>Having prepared a partition file, one is ready to start a partitioned analysis with <code>-q</code> (edge-equal), <code>-spp</code> (edge-proportional) or <code>-sp</code> (edge-unlinked) option. See <a href="http://www.iqtree.org/doc/Advanced-Tutorial#partitioned-analysis-for-multi-gene-alignments">this tutorial</a> for more details.</p><h2 id="mixture-models">Mixture models</h2><div class="hline"></div><h3 id="what-is-the-difference-between-partition-and-mixture-models">What is the difference between partition and mixture models?</h3><p>Mixture models, like partition models, allow more than one substitution model along the sequences. However, while a partition model assigns each alignment site a given specific model, mixture models do not need this information. A mixture model will compute for each site its probability (or weight) of belonging to each of the mixture classes (also called categories or components). Since the site-to-class assignment is unknown, the site likelihood under mixture models is the weighted sum of site likelihoods per mixture class.</p><p>For example, the <a href="http://www.iqtree.org/doc/Substitution-Models#rate-heterogeneity-across-sites">discrete Gamma rate heterogeneity</a> is a simple mixture model type. It has several rate categories with equal weight. IQ-TREE also supports a number of <a href="http://www.iqtree.org/doc/Substitution-Models#protein-mixture-models">predefined protein mixture models</a> such as the profile mixture models <code>C10</code> to <code>C60</code> (The ML variants of Bayesian <code>CAT</code> models).</p><p>Here, we discuss several possibilities to define new mixture models in IQ-TREE.</p><h3 id="defining-mixture-models">Defining mixture models</h3><p>To start with, the following command:</p><pre><code>iqtree -s example.phy -m "MIX{JC,HKY}"
</code></pre><p>specifies a mixture model (via the <code>MIX</code> keyword in the model string) with two components. The components (1) <code>JC</code> model, and (2) <code>HKY</code> model, are given in curly brackets and separated with a comma. IQ-TREE will then estimate the parameters of both mixture components as well as their weights: the proportion of sites belonging to each component.</p><blockquote><p><strong>NOTE</strong>: Do not forget the double-quotes around model string! They prevent interpretation of the curly brackets by the command line shell, i.e., <code>MIX{JC,HKY}</code> would otherwise be interpreted as <code>MIXJC MIXHKY</code>.</p></blockquote><p>Mixture models can be combined with rate heterogeneity, e.g.:</p><pre><code>iqtree -s example.phy -m "MIX{JC,HKY}+G4"
</code></pre><p>Here, we specify two mixture components and four Gamma rate categories. Effectively, this means that there are eight mixture components. Each site has a probability belonging to either <code>JC</code> or <code>HKY</code> and to one of the four rate categories.</p><h3 id="profile-mixture-models">Profile mixture models</h3><p>Sometimes one only wants to model the changes in nucleotide or amino-acid frequencies along the sequences while keeping the substitution rate matrix the same. This can be specified in IQ-TREE via <code>FMIX{...}</code> model syntax. For convenience the mixture components can be defined in a NEXUS file like this (example corresponds to <a href="http://www.iqtree.org/doc/Substitution-Models#protein-mixture-models">the CF4 model</a> of (<a href="https://doi.org/10.1186/1471-2148-8-331">Wang et al., 2008</a>)):</p><pre><code>#nexus
begin models;
    frequency Fclass1 = 0.02549352 0.01296012 0.005545202 0.006005566 0.01002193 0.01112289 0.008811948 0.001796161 0.004312188 0.2108274 0.2730413 0.01335451 0.07862202 0.03859909 0.005058205 0.008209453 0.03210019 0.002668138 0.01379098 0.2376598;
    frequency Fclass2 = 0.09596966 0.008786096 0.02805857 0.01880183 0.005026264 0.006454635 0.01582725 0.7215719 0.003379354 0.002257725 0.003013483 0.01343441 0.001511657 0.002107865 0.006751404 0.04798539 0.01141559 0.000523736 0.002188483 0.004934972;
    frequency Fclass3 = 0.01726065 0.005467988 0.01092937 0.3627871 0.001046402 0.01984758 0.5149206 0.004145081 0.002563289 0.002955213 0.005286931 0.01558693 0.002693098 0.002075771 0.003006167 0.01263069 0.01082144 0.000253451 0.001144787 0.004573568;
    frequency Fclass4 = 0.1263139 0.09564027 0.07050061 0.03316681 0.02095119 0.05473468 0.02790523 0.009007538 0.03441334 0.005855319 0.008061884 0.1078084 0.009019514 0.05018693 0.07948 0.09447839 0.09258897 0.01390669 0.05367769 0.01230413;

    frequency CF4model = FMIX{empirical,Fclass1,Fclass2,Fclass3,Fclass4};
end;
</code></pre><blockquote><p><strong>NOTE</strong>: The amino-acid order in this file is: A R N D C Q E G H I L K M F P S T W Y V.</p></blockquote><p>Here, the NEXUS file contains a <code>models</code> block to define new models. More explicitly, we define four AA profiles <code>Fclass1</code> to <code>Fclass4</code>, each containing 20 AA frequencies. Then, the frequency mixture is defined with</p><pre><code>FMIX{empirical,Fclass1,Fclass2,Fclass3,Fclass4}
</code></pre><p>This means, we have five components: the first corresponds to empirical AA frequencies to be inferred from the data and the remaining four components are specified in this NEXUS file. Please save this to a file, say, <code>mymodels.nex</code>. One can now start the analysis with:</p><pre><code>iqtree -s some_protein.aln -mdef mymodels.nex -m JTT+CF4model+G
</code></pre><p>The <code>-mdef</code> option specifies the NEXUS file containing user-defined models. Here, the <code>JTT</code> matrix is applied for all alignment sites and one varies the AA profiles along the alignment. One can use the NEXUS syntax to define all other profile mixture models such as <code>C10</code> to <code>C60</code>.</p><h3 id="nexus-model-file">NEXUS model file</h3><p>In fact, IQ-TREE uses this NEXUS model file internally to define all <a href="http://www.iqtree.org/doc/Substitution-Models#protein-mixture-models">protein mixture models</a>. In addition to defining state frequencies, one can specify the entire model with rate matrix and state frequencies together. For example, the LG4M model (<a href="https://doi.org/10.1093/molbev/mss112">Le et al., 2012</a>) can be defined by:</p><pre><code>#nexus
begin models;
    model LG4M1 =
        0.269343
        0.254612 0.150988
        0.236821 0.031863 0.659648
        ....;
    ....
    model LG4M4 = ....;
    
    model LG4M = MIX{LG4M1,LG4M2,LG4M3,LG4M4}*G4;
end;
</code></pre><p>Here, we first define the four matrices <code>LG4M1</code>, <code>LG4M2</code>, <code>LG4M3</code> and <code>LG4M4</code> in PAML format (see <a href="http://www.iqtree.org/doc/Substitution-Models#protein-models">protein models</a>). Then <code>LG4M</code> is defined as mixture model with these four components <em>fused</em> with Gamma rate heterogeneity (via <code>*G4</code> syntax instead of <code>+G4</code>). This means that, in total, we have 4 mixture components instead of 16. The first component <code>LG4M1</code> is rescaled by the rate of the lowest Gamma rate category. The fourth component <code>LG4M4</code> corresponds to the highest rate.</p><p>Note that both <code>frequency</code> and <code>model</code> commands can be embedded into a single model file.</p><h2 id="site-specific-frequency-models">Site-specific frequency models</h2><div class="hline"></div><p>Starting with version 1.5.0, IQ-TREE provides a new posterior mean site frequency (PMSF) model as a rapid approximation to the time and memory consuming profile mixture models <code>C10</code> to <code>C60</code> (<a href="https://doi.org/10.1093/bioinformatics/btn445">Le et al., 2008a</a>; a variant of PhyloBayes’ <code>CAT</code> model). The PMSF are the amino-acid profiles for each alignment site computed from an input mixture model and a guide tree. The PMSF model is much faster and requires much less RAM than <code>C10</code> to <code>C60</code> (see table below), regardless of the number of mixture classes. Our extensive simulations and empirical phylogenomic data analyses demonstrate that the PMSF models can effectively ameliorate long branch attraction artefacts.</p><p>If you use this model in a publication please cite:</p><blockquote><p><strong>H.C. Wang, B.Q. Minh, S. Susko and A.J. Roger</strong> (2018) Modeling site heterogeneity with posterior mean site frequency profiles accelerates accurate phylogenomic estimation. <em>Syst. Biol.</em>, 67:216-235. <a href="https://doi.org/10.1093/sysbio/syx068">https://doi.org/10.1093/sysbio/syx068</a></p></blockquote><p>Here is an example of computation time and RAM usage for an Obazoa data set (68 sequences, 43615 amino-acid sites) from <a href="https://doi.org/10.1098/rspb.2013.1755">Brown et al. (2013)</a> using 16 CPU cores:</p><table><thead><tr><th>Models</th><th style="text-align: right">CPU time</th><th style="text-align: right">Wall-clock time</th><th style="text-align: right">RAM usage</th></tr></thead><tbody><tr><td><code>LG+F+G</code></td><td style="text-align: right">43h:38m:23s</td><td style="text-align: right">3h:37m:23s</td><td style="text-align: right">1.8 GB</td></tr><tr><td><code>LG+C20+F+G</code></td><td style="text-align: right">584h:25m:29s</td><td style="text-align: right">46h:39m:06s</td><td style="text-align: right">38.8 GB</td></tr><tr><td><code>LG+C60+F+G</code></td><td style="text-align: right">1502h:25m:31s</td><td style="text-align: right">125h:15m:29s</td><td style="text-align: right">112.8 GB</td></tr><tr><td><code>LG+PMSF+G</code></td><td style="text-align: right">73h:30m:37s</td><td style="text-align: right">5h:7m:27s</td><td style="text-align: right">2.2 GB</td></tr></tbody></table><h3 id="example-usages">Example usages</h3><p>To use the PMSF model you have to provide a <em>guide tree</em>, which, for example, can be obtained by a quicker analysis under the simpler <code>LG+F+G</code> model. The guide tree can then be specified via <code>-ft</code> option, for example:</p><pre><code>iqtree -s &lt;alignment&gt; -m LG+C20+F+G -ft &lt;guide_tree&gt;
</code></pre><p>Here, IQ-TREE will perform two phases. In the first phase, IQ-TREE estimates mixture model parameters given the guide tree and then infers the site-specific frequency profile (printed to <code>.sitefreq</code> file). In the second phase, IQ-TREE will conduct typical analysis using the inferred frequency model instead of the mixture model to save RAM and running time. Note that without <code>-ft</code> option, IQ-TREE will conduct the analysis under the specified mixture model.</p><p>The PMSF model allows one, for the first time, to conduct nonparametric bootstrap under such complex models, for example (with 100 bootstrap replicates):</p><pre><code>iqtree -s &lt;alignment&gt; -m LG+C20+F+G -ft &lt;guide_tree&gt; -b 100
</code></pre><p>Please note that the first phase still consumes as much RAM as the mixture model. To overcome this, you can perform the first phase in a high-memory server and the second phase in a normal PC as follows:</p><pre><code>iqtree -s &lt;alignment&gt; -m LG+C20+F+G -ft &lt;guide_tree&gt; -n 0
</code></pre><p>This will stop the analysis after the first phase and also write a <code>.sitefreq</code> file. You can now copy this <code>.sitefreq</code> file to another low-memory machine and run with the same alignment:</p><pre><code>iqtree -s &lt;alignment&gt; -m LG+C20+F+G -fs &lt;file.sitefreq&gt; -b 100
</code></pre><p>This will omit the first phase and thus need much less RAM.</p><p>Finally, note that for long (phylogenomic) alignments you can utilize the multicore IQ-TREE version to further save the computing times with, say, 24 cores by:</p><pre><code># For IQ-TREE version &lt;= 1.5.X
iqtree-omp -nt 24 -s &lt;alignment&gt; -m LG+C20+F+G -fs &lt;file.sitefreq&gt;

# For IQ-TREE version &gt;= 1.6.0
iqtree -nt 24 -s &lt;alignment&gt; -m LG+C20+F+G -fs &lt;file.sitefreq&gt;
</code></pre><p>See also <a href="http://www.iqtree.org/doc/Command-Reference#site-specific-frequency-model-options">the list of relevant command line options</a>.</p><h2 id="heterotachy-models">Heterotachy models</h2><div class="hline"></div><p>Sequence data that have evolved under <em>heterotachy</em>, i.e., rate variation across sites and lineages (<a href="http://mbe.oxfordjournals.org/content/19/1/1.full">Lopez, Casane, and Philippe, 2002</a>), are known to mislead phylogenetic inference (<a href="https://doi.org/10.1038/nature02917">Kolaczkowski and Thornton, 2004</a>). To address this issue we introduce the General Heterogeneous evolution On a Single Topology (GHOST) model. More specifically, GHOST is an <em>edge-unlinked mixture model</em> consisting of several site classes, each having a separate set of model parameters and edge lengths on the same tree topology. Thus, GHOST naturally accounts for heterotachous evolution. In contrast to an <a href="http://www.iqtree.org/doc/Complex-Models#partition-models">edge-unlinked partition model</a>, the GHOST model does not require the <em>a priori</em> data partitioning, a possible source of model misspecification.</p><p>Extensive simulations show that the GHOST model can accurately recover the tree topology, branch lengths, substitution rate and base frequency parameters from heterotachously-evolved sequences. Moreover, we compare the GHOST model to the partition model and show that, owing to the minimization of model constraints, the GHOST model is able to offer unique biological insights when applied to empirical data.</p><p>If you use this model in a publication please cite:</p><blockquote><p><strong>S.M. Crotty, B.Q. Minh, N.G. Bean, B.R. Holland, J. Tuke, L.S. Jermiin and A. von Haeseler</strong> (2019) GHOST: Recovering historical signal from heterotachously-evolved sequence alignments. <em>Syst. Biol.</em>, in press. <a href="https://doi.org/10.1093/sysbio/syz051">https://doi.org/10.1093/sysbio/syz051</a></p></blockquote><h3 id="quick-usages">Quick usages</h3><p>Make sure that you have IQ-TREE version 1.6.0 or later. The GHOST model with <code>k</code> mixture classes is executed by adding <code>+Hk</code> to the model option (<code>-m</code>). For example if one wants to fit a GHOST model with 4 classes in conjunction with the <code>GTR</code> model of DNA evolution to sequences contained in <code>data.fst</code>, one would use the following command:</p><pre><code>iqtree -s data.fst -m GTR+H4
</code></pre><p>By default the above command will link GTR parameters across all classes. If you want to unlink GTR parameters, so that IQ-TREE estimates them separately for each class, replace <code>+H4</code> by <code>*H4</code>:</p><pre><code>iqtree -s data.fst -m GTR*H4
</code></pre><p>Note that this infers one set of empirical base frequencies and apply those to all classes. If one wishes to infer separate base frequencies for each class then the <code>+FO</code> option is required:</p><pre><code>iqtree -s data.fst -m GTR+FO*H4
</code></pre><p>The <code>-wspm</code> option will generate a <code>.siteprob</code> output file. This contains the probability of each site belonging to each class:</p><pre><code>iqtree -s data.fst -m GTR+FO*H4 -wspm
</code></pre><div class="spacing"></div></div><div class="col-md-3"><div class="well well-sm"> <a href="http://www.iqtree.org/doc/iqtree-doc.pdf"><i class="fa fa-file-pdf-o fa-lg" aria-hidden="true"></i> Download Tutorials and Manual in one PDF</a></div><div class="well well-sm" id="sidebar"><h4>Documentation</h4><p> <a href="http://www.iqtree.org/doc/Home"> <i class="fa fa-info-circle"></i> Introduction </a></p><p> <a href="http://www.iqtree.org/doc/Quickstart"> <i class="fa fa-info-circle"></i> Getting Started </a></p><p> <a href="http://www.iqtree.org/doc/Web-Server-Tutorial"> <i class="fa fa-info-circle"></i> Web Server Tutorial </a></p><p> <a href="http://www.iqtree.org/doc/Tutorial"> <i class="fa fa-info-circle"></i> Beginner's Tutorial </a></p><p> <a href="http://www.iqtree.org/doc/Advanced-Tutorial"> <i class="fa fa-info-circle"></i> Advanced Tutorial </a></p><p> <a href="http://www.iqtree.org/doc/Assessing-Phylogenetic-Assumptions"> <i class="fa fa-info-circle"></i> Assessing Phylogenetic Assumptions </a></p><p> <a href="http://www.iqtree.org/doc/Concordance-Factor"> <i class="fa fa-info-circle"></i> Concordance Factor </a></p><p> <a href="http://www.iqtree.org/doc/Dating"> <i class="fa fa-info-circle"></i> Phylogenetic Dating </a></p><p> <a href="http://www.iqtree.org/doc/Rootstrap"> <i class="fa fa-info-circle"></i> Rooting phylogenies </a></p><p> <a href="http://www.iqtree.org/doc/Frequently-Asked-Questions"> <i class="fa fa-question-circle"></i> Frequently Asked Questions </a></p><p> <a href="http://www.iqtree.org/doc/Substitution-Models"> <i class="fa fa-book"></i> Substitution Models </a></p><p> <i class="fa fa-book"></i> Complex Models</p><ul><li><a href="http://www.iqtree.org/doc/Complex-Models#partition-models">Partition models</a></li><li><a href="http://www.iqtree.org/doc/Complex-Models#mixture-models">Mixture models</a></li><li><a href="http://www.iqtree.org/doc/Complex-Models#site-specific-frequency-models">Site-specific frequency models</a></li><li><a href="http://www.iqtree.org/doc/Complex-Models#heterotachy-models">Heterotachy models</a></li></ul><p> <a href="http://www.iqtree.org/doc/Polymorphism-Aware-Models"> <i class="fa fa-book"></i> Polymorphism-Aware Models </a></p><p> <a href="http://www.iqtree.org/doc/Command-Reference"> <i class="fa fa-book"></i> Command Reference </a></p><p> <a href="http://www.iqtree.org/doc/Compilation-Guide"> <i class="fa fa-book"></i> Compilation Guide </a></p><p> <a href="http://www.iqtree.org/doc/Developer-Guide"> <i class="fa fa-codepen"></i> Developer Guide </a></p><p> <a href="http://www.iqtree.org/doc/Analyzing-Big-Data"> <i class="fa fa-info-circle"></i> Analyzing Big Data </a></p></div></div></div></div><div id="footerwrap"><div class="container"><div class="row"><div class="col-sm-4"><h4>Software</h4><p> <a href="http://www.iqtree.org/release/">IQ-TREE release notes</a><br> <a href="https://github.com/Cibiv/IQ-TREE">GitHub repository</a><br> <a href="http://www.iqtree.org/about/">Development team</a><br> <a href="http://www.somsubhra.com/github-release-stats/?username=Cibiv&amp;repository=IQ-TREE">Download statistics</a><br></p></div><div class="col-sm-4"><h4>User support</h4><p> <a href="http://www.iqtree.org/doc/Frequently-Asked-Questions">Frequently asked questions</a><br> <a href="http://www.iqtree.org/doc/">User documentation</a><br> <a href="https://groups.google.com/d/forum/iqtree">IQ-TREE Google group</a><br> <a href="http://iqtree.cibiv.univie.ac.at/" target="_blank">IQ-TREE online web service</a><br></p></div><div class="col-sm-4"><h4>Links</h4><p> <a href="https://cs.anu.edu.au/">Research School of Computer Science, ANU</a><br> <a href="http://www.cibiv.at/">Center for Integrative Bioinformatics Vienna</a><br> <a href="http://validator.w3.org/check/referer">Link Checker</a><br></p></div></div></div></div><div class="alert alert-dismissible text-center cookiealert" role="alert"><div class="cookiealert-container"> We use cookies to ensure you get the best experience on our website <button type="button" class="btn btn-primary btn-sm acceptcookies" aria-label="Close"> I agree </button></div></div><script src="./Complex Models_files/jquery.min.js.download"></script> <script src="./Complex Models_files/bootstrap.min.js.download"></script> <script src="./Complex Models_files/retina-1.1.0.js.download"></script> <script src="./Complex Models_files/jquery.hoverdir.js.download"></script> <script src="./Complex Models_files/jquery.isotope.min.js.download"></script> <script src="./Complex Models_files/cookiealert-standalone.js.download"></script> <script> (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){ (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o), m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m) })(window,document,'script','//www.google-analytics.com/analytics.js','ga'); ga('create', 'UA-38693379-1', 'auto'); ga('send', 'pageview'); </script> <script> // Portfolio (function($) { "use strict"; var $container = $('.portfolio'), $items = $container.find('.portfolio-item'), portfolioLayout = 'fitRows'; if( $container.hasClass('portfolio-centered') ) { portfolioLayout = 'masonry'; } $container.isotope({ filter: '*', animationEngine: 'best-available', layoutMode: portfolioLayout, animationOptions: { duration: 750, easing: 'linear', queue: false }, masonry: { } }, refreshWaypoints()); function refreshWaypoints() { setTimeout(function() { }, 1000); } $('nav.portfolio-filter ul a').on('click', function() { var selector = $(this).attr('data-filter'); $container.isotope({ filter: selector }, refreshWaypoints()); $('nav.portfolio-filter ul a').removeClass('active'); $(this).addClass('active'); return false; }); function getColumnNumber() { var winWidth = $(window).width(), columnNumber = 1; if (winWidth > 1200) { columnNumber = 5; } else if (winWidth > 950) { columnNumber = 4; } else if (winWidth > 600) { columnNumber = 3; } else if (winWidth > 400) { columnNumber = 2; } else if (winWidth > 250) { columnNumber = 1; } return columnNumber; } function setColumns() { var winWidth = $(window).width(), columnNumber = getColumnNumber(), itemWidth = Math.floor(winWidth / columnNumber); $container.find('.portfolio-item').each(function() { $(this).css( { width : itemWidth + 'px' }); }); } function setPortfolio() { setColumns(); $container.isotope('reLayout'); } $container.imagesLoaded(function () { setPortfolio(); }); $(window).on('resize', function () { setPortfolio(); }); })(jQuery); </script>
<table cellspacing="0" cellpadding="0" class="gstl_50 gssb_c" style="width: 423px; display: none; top: 94px; position: absolute; left: 710px;"><tbody><tr><td class="gssb_f"></td><td class="gssb_e" style="width: 100%;"></td></tr></tbody></table></body></html>