<!DOCTYPE html>
<!-- saved from url=(0030)http://www.iqtree.org/doc/Home -->
<html lang="en" class="csstransforms csstransforms3d csstransitions"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1"><meta name="description" content=""><meta name="author" content=""><title>Introduction</title><link href="./Introduction_files/bootstrap.min.css" rel="stylesheet"><link href="./Introduction_files/style.css" rel="stylesheet"><link href="./Introduction_files/font-awesome.min.css" rel="stylesheet"><link rel="stylesheet" href="./Introduction_files/cookiealert.css"><script async="" src="./Introduction_files/async-ads.js.download"></script><script src="./Introduction_files/cse_element__en.js.download" type="text/javascript"></script><link type="text/css" rel="stylesheet" href="./Introduction_files/default+en.css"><link type="text/css" rel="stylesheet" href="./Introduction_files/default.css"><style type="text/css">.gsc-control-cse{font-family:times new roman, arial, serif}.gsc-control-cse .gsc-table-result{font-family:times new roman, arial, serif}.gsc-refinementsGradient{background:linear-gradient(to left,rgba(255,255,255,1),rgba(255,255,255,0))}.gsc-control-cse{border-color:#337AB7;background-color:#337AB7}input.gsc-input,.gsc-input-box,.gsc-input-box-hover,.gsc-input-box-focus{border-color:#FFFFFF}.gsc-search-button-v2,.gsc-search-button-v2:hover,.gsc-search-button-v2:focus{border-color:#4CAE4C;background-color:#5CB85C;background-image:none;filter:none}.gsc-search-button-v2 svg{fill:#FFFFFF}.gsc-tabHeader.gsc-tabhActive,.gsc-refinementHeader.gsc-refinementhActive{color:#CCCCCC;border-color:#CCCCCC;background-color:#FFFFFF}.gsc-tabHeader.gsc-tabhInactive,.gsc-refinementHeader.gsc-refinementhInactive{color:#CCCCCC;border-color:#CCCCCC;background-color:#FFFFFF}.gsc-webResult.gsc-result,.gsc-results .gsc-imageResult{border-color:#F5F5F5;background-color:#F5F5F5}.gsc-webResult.gsc-result:hover{border-color:#F5F5F5;background-color:#F5F5F5}.gs-webResult.gs-result a.gs-title:link,.gs-webResult.gs-result a.gs-title:link b,.gs-imageResult a.gs-title:link,.gs-imageResult a.gs-title:link b{color:#0000CC}.gs-webResult.gs-result a.gs-title:visited,.gs-webResult.gs-result a.gs-title:visited b,.gs-imageResult a.gs-title:visited,.gs-imageResult a.gs-title:visited b{color:#0000CC}.gs-webResult.gs-result a.gs-title:hover,.gs-webResult.gs-result a.gs-title:hover b,.gs-imageResult a.gs-title:hover,.gs-imageResult a.gs-title:hover b{color:#0000CC}.gs-webResult.gs-result a.gs-title:active,.gs-webResult.gs-result a.gs-title:active b,.gs-imageResult a.gs-title:active,.gs-imageResult a.gs-title:active b{color:#0000CC}.gsc-cursor-page{color:#0000CC}a.gsc-trailing-more-results:link{color:#0000CC}.gs-webResult:not(.gs-no-results-result):not(.gs-error-result) .gs-snippet,.gs-fileFormatType{color:#000000}.gs-webResult div.gs-visibleUrl{color:#008000}.gs-webResult div.gs-visibleUrl-short{color:#008000}.gsc-cursor-box{border-color:#F5F5F5}.gsc-results .gsc-cursor-box .gsc-cursor-page{border-color:#CCCCCC;background-color:#FFFFFF;color:#CCCCCC}.gsc-results .gsc-cursor-box .gsc-cursor-current-page{border-color:#CCCCCC;background-color:#FFFFFF;color:#CCCCCC}.gsc-webResult.gsc-result.gsc-promotion{border-color:#336699;background-color:#FFFFFF}.gsc-completion-title{color:#0000CC}.gsc-completion-snippet{color:#000000}.gs-promotion a.gs-title:link,.gs-promotion a.gs-title:link *,.gs-promotion .gs-snippet a:link{color:#0000CC}.gs-promotion a.gs-title:visited,.gs-promotion a.gs-title:visited *,.gs-promotion .gs-snippet a:visited{color:#0000CC}.gs-promotion a.gs-title:hover,.gs-promotion a.gs-title:hover *,.gs-promotion .gs-snippet a:hover{color:#0000CC}.gs-promotion a.gs-title:active,.gs-promotion a.gs-title:active *,.gs-promotion .gs-snippet a:active{color:#0000CC}.gs-promotion .gs-snippet,.gs-promotion .gs-title .gs-promotion-title-right,.gs-promotion .gs-title .gs-promotion-title-right *{color:#000000}.gs-promotion .gs-visibleUrl,.gs-promotion .gs-visibleUrl-short{color:#008000}.gcsc-find-more-on-google{color:#0000CC}.gcsc-find-more-on-google-magnifier{fill:#0000CC}</style><style type="text/css">.gscb_a{display:inline-block;font:27px/13px arial,sans-serif}.gsst_a .gscb_a{color:#a1b9ed;cursor:pointer}.gsst_a:hover .gscb_a,.gsst_a:focus .gscb_a{color:#36c}.gsst_a{display:inline-block}.gsst_a{cursor:pointer;padding:0 4px}.gsst_a:hover{text-decoration:none!important}.gsst_b{font-size:16px;padding:0 2px;position:relative;user-select:none;-webkit-user-select:none;white-space:nowrap}.gsst_e{vertical-align:middle;opacity:0.55;}.gsst_a:hover .gsst_e,.gsst_a:focus .gsst_e{opacity:0.72;}.gsst_a:active .gsst_e{opacity:1;}.gsst_f{background:white;text-align:left}.gsst_g{background-color:white;border:1px solid #ccc;border-top-color:#d9d9d9;box-shadow:0 2px 4px rgba(0,0,0,0.2);-webkit-box-shadow:0 2px 4px rgba(0,0,0,0.2);margin:-1px -3px;padding:0 6px}.gsst_h{background-color:white;height:1px;margin-bottom:-1px;position:relative;top:-1px}.gsib_a{width:100%;padding:4px 6px 0}.gsib_a,.gsib_b{vertical-align:top}.gssb_c{border:0;position:absolute;z-index:989}.gssb_e{border:1px solid #ccc;border-top-color:#d9d9d9;box-shadow:0 2px 4px rgba(0,0,0,0.2);-webkit-box-shadow:0 2px 4px rgba(0,0,0,0.2);cursor:default}.gssb_f{visibility:hidden;white-space:nowrap}.gssb_k{border:0;display:block;position:absolute;top:0;z-index:988}.gsdd_a{border:none!important}.gsq_a{padding:0}.gssb_a{padding:0 7px}.gssb_a,.gssb_a td{white-space:nowrap;overflow:hidden;line-height:22px}#gssb_b{font-size:11px;color:#36c;text-decoration:none}#gssb_b:hover{font-size:11px;color:#36c;text-decoration:underline}.gssb_g{text-align:center;padding:8px 0 7px;position:relative}.gssb_h{font-size:15px;height:28px;margin:0.2em;-webkit-appearance:button}.gssb_i{background:#eee}.gss_ifl{visibility:hidden;padding-left:5px}.gssb_i .gss_ifl{visibility:visible}a.gssb_j{font-size:13px;color:#36c;text-decoration:none;line-height:100%}a.gssb_j:hover{text-decoration:underline}.gssb_l{height:1px;background-color:#e5e5e5}.gssb_m{color:#000;background:#fff}.gssb_a{padding:0 9px}.gsib_a{padding:5px 9px 4px 9px}.gscb_a{line-height:27px}.gssb_e{border:0}.gssb_l{margin:5px 0}input.gsc-input::-webkit-input-placeholder{font-size:14px}input.gsc-input:-moz-placeholder{font-size:14px}input.gsc-input::-moz-placeholder{font-size:14px}input.gsc-input:-ms-input-placeholder{font-size:14px}input.gsc-input:focus::-webkit-input-placeholder{color:transparent}input.gsc-input:focus:-moz-placeholder{color:transparent}input.gsc-input:focus::-moz-placeholder{color:transparent}input.gsc-input:focus:-ms-input-placeholder{color:transparent}.gssb_c .gsc-completion-container{position:static}.gssb_c{z-index:5000}.gsc-completion-container table{background:transparent;font-size:inherit;font-family:inherit}.gssb_c > tbody > tr,.gssb_c > tbody > tr > td,.gssb_d,.gssb_d > tbody > tr,.gssb_d > tbody > tr > td,.gssb_e,.gssb_e > tbody > tr,.gssb_e > tbody > tr > td{padding:0;margin:0;border:0}.gssb_a table,.gssb_a table tr,.gssb_a table tr td{padding:0;margin:0;border:0}</style></head><body style=""><div class="navbar navbar-default navbar-fixed-top" role="navigation"><div class="container"><div class="navbar-header"> <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar"></span> <span class="icon-bar"></span> <span class="icon-bar"></span> </button> <a class="navbar-brand" href="http://www.iqtree.org/">IQ-TREE</a></div><div class="navbar-collapse collapse navbar-right"><ul class="nav navbar-nav"><li><a data-toggle="collapse" data-target=".navbar-collapse" href="http://www.iqtree.org/#download">Download</a></li><li><a href="http://www.iqtree.org/news/">News</a></li><li class="dropdown"> <a class="dropdown-toggle" data-toggle="dropdown" href="http://www.iqtree.org/doc/Home#">Web server <span class="caret"></span></a><ul class="dropdown-menu"><li><a href="http://iqtree.cibiv.univie.ac.at/" target="_blank">CIBIV, Austria</a></li><li><a href="https://www.hiv.lanl.gov/content/sequence/IQTREE/iqtree.html" target="_blank">Los Alamos Lab, USA</a></li><li><a href="https://www.phylo.org/" target="_blank">CIPRES Gateway, USA</a></li></ul></li><li><a href="http://www.iqtree.org/doc/">Docs</a></li><li><a href="http://www.iqtree.org/workshop/">Workshop</a></li><li><a href="http://www.iqtree.org/about/">About</a></li></ul></div></div></div><div id="blue"><div class="container"><div class="row"><div class="col-sm-6"><h3> Tutorial</h3></div><div class="col-sm-6"> <script async="" src="./Introduction_files/analytics.js.download"></script><script type="text/javascript" async="" src="./Introduction_files/f.txt"></script><script> (function() { var cx = '003669294240048101214:pace-yz0kou'; var gcse = document.createElement('script'); gcse.type = 'text/javascript'; gcse.async = true; gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') + '//cse.google.com/cse.js?cx=' + cx; var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(gcse, s); })(); </script> <div id="___gcse_0"><div class="gsc-control-cse gsc-control-cse-en"><div class="gsc-control-wrapper-cse" dir="ltr"><form class="gsc-search-box gsc-search-box-tools" accept-charset="utf-8"><table cellspacing="0" cellpadding="0" class="gsc-search-box"><tbody><tr><td class="gsc-input"><div class="gsc-input-box" id="gsc-iw-id1"><table cellspacing="0" cellpadding="0" id="gs_id50" class="gstl_50 gsc-input" style="width: 100%; padding: 0px;"><tbody><tr><td id="gs_tti50" class="gsib_a"><input autocomplete="off" type="text" size="10" class="gsc-input" name="search" title="search" id="gsc-i-id1" dir="ltr" spellcheck="false" style="width: 100%; padding: 0px; border: none; margin: 0px; height: auto; background: url(&quot;https://www.google.com/cse/static/images/1x/en/branding.png&quot;) left center no-repeat rgb(255, 255, 255); outline: none;"></td><td class="gsib_b"><div class="gsst_b" id="gs_st50" dir="ltr"><a class="gsst_a" href="javascript:void(0)" style="display: none;" title="Clear search box" role="button"><span class="gscb_a" id="gs_cb50" aria-hidden="true">×</span></a></div></td></tr></tbody></table></div></td><td class="gsc-search-button"><button class="gsc-search-button gsc-search-button-v2"><svg width="13" height="13" viewBox="0 0 13 13"><title>search</title><path d="m4.8495 7.8226c0.82666 0 1.5262-0.29146 2.0985-0.87438 0.57232-0.58292 0.86378-1.2877 0.87438-2.1144 0.010599-0.82666-0.28086-1.5262-0.87438-2.0985-0.59352-0.57232-1.293-0.86378-2.0985-0.87438-0.8055-0.010599-1.5103 0.28086-2.1144 0.87438-0.60414 0.59352-0.8956 1.293-0.87438 2.0985 0.021197 0.8055 0.31266 1.5103 0.87438 2.1144 0.56172 0.60414 1.2665 0.8956 2.1144 0.87438zm4.4695 0.2115 3.681 3.6819-1.259 1.284-3.6817-3.7 0.0019784-0.69479-0.090043-0.098846c-0.87973 0.76087-1.92 1.1413-3.1207 1.1413-1.3553 0-2.5025-0.46363-3.4417-1.3909s-1.4088-2.0686-1.4088-3.4239c0-1.3553 0.4696-2.4966 1.4088-3.4239 0.9392-0.92727 2.0864-1.3969 3.4417-1.4088 1.3553-0.011889 2.4906 0.45771 3.406 1.4088 0.9154 0.95107 1.379 2.0924 1.3909 3.4239 0 1.2126-0.38043 2.2588-1.1413 3.1385l0.098834 0.090049z"></path></svg></button></td><td class="gsc-clear-button"><div class="gsc-clear-button" title="clear results">&nbsp;</div></td></tr></tbody></table></form><div class="gsc-results-wrapper-overlay"><div class="gsc-results-close-btn" tabindex="0"></div><div class="gsc-positioningWrapper"><div class="gsc-tabsAreaInvisible"><div aria-label="refinement" role="tab" class="gsc-tabHeader gsc-inline-block gsc-tabhActive">Custom Search</div><span class="gs-spacer"> </span></div></div><div class="gsc-positioningWrapper"><div class="gsc-tabsAreaInvisible"></div></div><div class="gsc-above-wrapper-area-invisible"><table cellspacing="0" cellpadding="0" class="gsc-above-wrapper-area-container"><tbody><tr><td class="gsc-result-info-container"><div class="gsc-result-info-invisible"></div></td><td class="gsc-orderby-container"><div class="gsc-orderby-invisible"><div class="gsc-orderby-label gsc-inline-block">Sort by:</div><div class="gsc-option-menu-container gsc-inline-block"><div class="gsc-selected-option-container gsc-inline-block"><div class="gsc-selected-option">Relevance</div><div class="gsc-option-selector"></div></div><div class="gsc-option-menu-invisible"><div class="gsc-option-menu-item gsc-option-menu-item-highlighted"><div class="gsc-option">Relevance</div></div><div class="gsc-option-menu-item"><div class="gsc-option">Date</div></div></div></div></div></td></tr></tbody></table></div><div class="gsc-adBlockInvisible"></div><div class="gsc-wrapper"><div class="gsc-adBlockInvisible"></div><div class="gsc-resultsbox-invisible"><div class="gsc-resultsRoot gsc-tabData gsc-tabdActive"><div><div class="gsc-expansionArea"></div></div></div></div></div></div><div class="gsc-modal-background-image" tabindex="0"></div></div></div></div></div></div></div></div><div class="container mtb"><div class="row"><div class="col-md-9"><p><em>Last update: Dec 1, 2019, Contributors: Jana Trifinopoulos, Minh Bui, Nltung</em></p><h1 id="introduction">Introduction</h1><h2 id="why-iq-tree">Why IQ-TREE?</h2><div class="hline"></div><p>Thanks to the recent advent of next-generation sequencing techniques, the amount of phylogenomic/transcriptomic data have been rapidly accumulated. This extremely facilitates resolving many “deep phylogenetic” questions in the tree of life. At the same time it poses major computational challenges to analyze such big data, where most phylogenetic software cannot handle. Moreover, there is a need to develop more complex probabilistic models to adequately capture realistic aspects of genomic sequence evolution.</p><p>This trends motivated us to develop the IQ-TREE software with a strong emphasis on phylogenomic inference. Our goals are:</p><ul><li><strong>Accuracy</strong>: Proposing novel computational methods that perform better than existing approaches.</li><li><strong>Speed</strong>: Allowing fast analysis on big data sets and utilizing high performance computing platforms.</li><li><strong>Flexibility</strong>: Facilitating the inclusion of new (phylogenomic) models and sequence data types.</li><li><strong>Versatility</strong>: Implementing a broad range of commonly-used maximum likelihood analyses.</li></ul><p>IQ-TREE has been developed since 2011 and freely available at <a href="http://www.iqtree.org/">http://www.iqtree.org/</a> as open-source software under the <a href="http://www.gnu.org/licenses/licenses.en.html">GNU-GPL license version 2</a>. It is actively maintained by the core development team (see below) and a number of collabrators.</p><p>The name IQ-TREE comes from the fact that it is the successor of <a href="http://www.cibiv.at/software/iqpnni/"><strong>IQ</strong>PNNI</a> and <a href="http://www.tree-puzzle.de/"><strong>TREE</strong>-PUZZLE</a> software.</p><h2 id="key-features">Key features</h2><div class="hline"></div><ul><li><strong>Efficient search algorithm</strong>: Fast and effective stochastic algorithm to reconstruct phylogenetic trees by maximum likelihood. IQ-TREE compares favorably to RAxML and PhyML in terms of likelihood while requiring similar amount of computing time (<a href="https://doi.org/10.1093/molbev/msu300">Nguyen et al., 2015</a>).</li><li><strong>Ultrafast bootstrap</strong>: An ultrafast bootstrap approximation (UFBoot) to assess branch supports. UFBoot is 10 to 40 times faster than RAxML rapid bootstrap and obtains less biased support values (<a href="https://doi.org/10.1093/molbev/mst024">Minh et al., 2013</a>; <a href="https://doi.org/10.1093/molbev/msx281">Hoang et al., 2018</a>).</li><li><strong>Ultrafast model selection</strong>: An ultrafast and automatic model selection (ModelFinder) which is 10 to 100 times faster than jModelTest and ProtTest. ModelFinder also finds best-fit partitioning scheme like PartitionFinder.</li><li><strong>Big Data Analysis</strong>: Supporting huge datasets with thousands of sequences or millions of alignment sites via <a href="http://www.iqtree.org/doc/Command-Reference#checkpointing-to-resume-stopped-run">checkpointing</a>, safe numerical and low memory mode. <a href="http://www.iqtree.org/doc/Tutorial#utilizing-multi-core-cpus">Multicore CPUs</a> and <a href="http://www.iqtree.org/doc/Compilation-Guide#compiling-mpi-version">parallel MPI system</a> are utilized to speedup analysis.</li><li><strong>Phylogenetic testing</strong>: Several fast branch tests like SH-aLRT and aBayes test (<a href="https://doi.org/10.1093/sysbio/syr041">Anisimova et al., 2011</a>) and tree topology tests like the approximately unbiased (AU) test (<a href="https://doi.org/10.1080/10635150290069913">Shimodaira, 2002</a>).</li></ul><p>The strength of IQ-TREE is the availability of a wide variety of phylogenetic models:</p><ul><li><strong>Common models</strong>: All <a href="http://www.iqtree.org/doc/Substitution-Models">common substitution models</a> for DNA, protein, codon, binary and morphological data with <a href="http://www.iqtree.org/doc/Substitution-Models#rate-heterogeneity-across-sites">rate heterogeneity among sites</a> and <a href="http://www.iqtree.org/doc/Substitution-Models#ascertainment-bias-correction">ascertainment bias correction</a> for e.g. SNP data.</li><li><strong><a href="http://www.iqtree.org/doc/Complex-Models#partition-models">Partition models</a></strong>: Allowing individual models for different genomic loci (e.g. genes or codon positions), mixed data types, mixed rate heterogeneity types, linked or unlinked branch lengths between partitions.</li><li><strong>Mixture models</strong>: <a href="http://www.iqtree.org/doc/Complex-Models#mixture-models">fully customizable mixture models</a> and <a href="http://www.iqtree.org/doc/Substitution-Models#protein-mixture-models">empirical protein mixture models</a> and.</li><li><strong>Polymorphism-aware models</strong>: Accounting for <em>incomplete lineage sorting</em> to infer species tree from genome-wide population data (<a href="https://doi.org/10.1016/j.jtbi.2016.07.042">Schrempf et al., 2016</a>).</li></ul><h2 id="free-web-server">Free web server</h2><div class="hline"></div><p>For a quick start you can also try the IQ-TREE web server, which performs online computation using a dedicated computing cluster. It is very easy to use with as few as just 3 clicks! Try it out at</p><p><a href="http://iqtree.cibiv.univie.ac.at/">http://iqtree.cibiv.univie.ac.at</a></p><h2 id="user-support">User support</h2><div class="hline"></div><p>Please refer to the <a href="http://www.iqtree.org/doc/">user documentation</a> and <a href="http://www.iqtree.org/doc/Frequently-Asked-Questions">frequently asked questions</a>. If you have further questions, feedback, feature requests, and bug reports, please sign up the following Google group (if not done yet) and post a topic to the</p><p><a href="https://groups.google.com/d/forum/iqtree">https://groups.google.com/d/forum/iqtree</a></p><p><em>The average response time is two working days.</em></p><h2 id="documentation">Documentation</h2><div class="hline"></div><p>IQ-TREE has an extensive documentation with several tutorials and manual:</p><ul><li><a href="http://www.iqtree.org/doc/Quickstart">Getting started guide</a>: recommended for users who just downloaded IQ-TREE.</li><li><a href="http://www.iqtree.org/doc/Web-Server-Tutorial">Web Server Tutorial</a>: A quick starting guide for the IQ-TREE Web Server.</li><li><a href="http://www.iqtree.org/doc/Tutorial">Beginner’s tutorial</a>: recommended for users starting to use IQ-TREE.</li><li><a href="http://www.iqtree.org/doc/Advanced-Tutorial">Advanced tutorial</a>: recommended for more experienced users who want to explore more features of IQ-TREE.</li><li><a href="http://www.iqtree.org/doc/Command-Reference">Command Reference</a>: Comprehensive documentation of command-line options available in IQ-TREE.</li><li><a href="http://www.iqtree.org/doc/Substitution-Models">Substitution Models</a>: All common substitution models and usages.</li><li><a href="http://www.iqtree.org/doc/Complex-Models">Complex Models</a>: Complex models such as partition and mixture models.</li><li><a href="http://www.iqtree.org/doc/Polymorphism-Aware-Models">Polymorphism Aware Models</a>: Polymorphism-aware phylogenetic Models (PoMo) related documentation.</li><li><a href="http://www.iqtree.org/doc/Compilation-Guide">Compilation guide</a>: for advanced users who wants to compile IQ-TREE from source code.</li><li><a href="http://www.iqtree.org/doc/Frequently-Asked-Questions">Frequently asked questions (FAQ)</a>: recommended to have a look before you post a question in the <a href="https://groups.google.com/d/forum/iqtree">IQ-TREE group</a>.</li></ul><h2 id="how-to-cite-iq-tree">How to cite IQ-TREE?</h2><div class="hline"></div><blockquote><p><strong>To maintain IQ-TREE, support users and secure fundings, it is important for us that you cite the following papers, whenever the corresponding features were applied for your analysis.</strong></p><ul><li><p>Example 1: <em>We obtained branch supports with the ultrafast bootstrap (Hoang et al., 2018) implemented in the IQ-TREE software (Nguyen et al., 2015).</em></p></li><li><p>Example 2: <em>We inferred the maximum-likelihood tree using the edge-linked partition model in IQ-TREE (Chernomor et al., 2016; Nguyen et al., 2015).</em></p></li></ul></blockquote><p>If you performed the tests of symmetry, please cite:</p><ul><li><strong>S. Naser-Khdour, B.Q. Minh, W. Zhang, E.A. Stone, R. Lanfear</strong> (2019) The prevalence and pmpact of model violations in phylogenetic analysis, <em>Genome Biol. Evol.</em>, in press. <a href="https://doi.org/10.1093/gbe/evz193">https://doi.org/10.1093/gbe/evz193</a></li></ul><p>If you used the polymorphism-aware models please cite:</p><ul><li><strong>D. Schrempf, B.Q. Minh, A. von Haeseler, and C. Kosiol</strong> (2019) Polymorphism-aware species trees with advanced mutation models, bootstrap, and rate heterogeneity. <em>Mol. Biol. Evol.</em>, 36:1294-1301. <a href="https://doi.org/10.1093/molbev/msz043">https://doi.org/10.1093/molbev/msz043</a></li></ul><p>If you used the heterotachy model (GHOST) please cite:</p><ul><li><strong>S.M. Crotty, B.Q. Minh, N.G. Bean, B.R. Holland, J. Tuke, L.S. Jermiin, A. von Haeseler</strong> (2019) GHOST: Recovering historical signal from heterotachously-evolved sequence alignments. <em>Syst. Biol.</em>, in press. <a href="https://doi.org/10.1093/sysbio/syz051">https://doi.org/10.1093/sysbio/syz051</a></li></ul><p>If you performed the ultrafast bootstrap (UFBoot) please cite:</p><ul><li><strong>D.T. Hoang, O. Chernomor, A. von Haeseler, B.Q. Minh, and L.S. Vinh</strong> (2018) UFBoot2: Improving the ultrafast bootstrap approximation. <em>Mol. Biol. Evol.</em>, 35:518–522. <a href="https://doi.org/10.1093/molbev/msx281">https://doi.org/10.1093/molbev/msx281</a></li></ul><p>If you used posterior mean site frequency model please cite:</p><ul><li><strong>H.C. Wang, B.Q. Minh, S. Susko and A.J. Roger</strong> (2018) Modeling site heterogeneity with posterior mean site frequency profiles accelerates accurate phylogenomic estimation. <em>Syst. Biol.</em>, 67:216-235. <a href="https://doi.org/10.1093/sysbio/syx068">https://doi.org/10.1093/sysbio/syx068</a></li></ul><p>If you used ModelFinder please cite:</p><ul><li><strong>S. Kalyaanamoorthy, B.Q. Minh, T.K.F. Wong, A. von Haeseler, and L.S. Jermiin</strong> (2017) ModelFinder: Fast Model Selection for Accurate Phylogenetic Estimates, <em>Nature Methods</em>, 14:587–589. <a href="https://doi.org/10.1038/nmeth.4285">https://doi.org/10.1038/nmeth.4285</a></li></ul><p>If you performed tree reconstruction please cite:</p><ul><li><strong>L.-T. Nguyen, H.A. Schmidt, A. von Haeseler, and B.Q. Minh</strong> (2015) IQ-TREE: A fast and effective stochastic algorithm for estimating maximum likelihood phylogenies. <em>Mol. Biol. Evol.</em>, 32:268-274. <a href="https://doi.org/10.1093/molbev/msu300">https://doi.org/10.1093/molbev/msu300</a></li></ul><p>If you used partition models e.g., for phylogenomic analysis please cite:</p><ul><li><strong>O. Chernomor, A. von Haeseler, and B.Q. Minh</strong> (2016) Terrace aware data structure for phylogenomic inference from supermatrices. <em>Syst. Biol.</em>, 65:997-1008. <a href="https://doi.org/10.1093/sysbio/syw037">https://doi.org/10.1093/sysbio/syw037</a></li></ul><p>If you used the <a href="http://iqtree.cibiv.univie.ac.at/">IQ-TREE web server</a> please cite:</p><ul><li><strong>J. Trifinopoulos, L.-T. Nguyen, A. von Haeseler, and B.Q. Minh</strong> (2016) W-IQ-TREE: a fast online phylogenetic tool for maximum likelihood analysis. <em>Nucleic Acids Res.</em>, 44 (W1):W232-W235. <a href="https://doi.org/10.1093/nar/gkw256">https://doi.org/10.1093/nar/gkw256</a></li></ul><h2 id="development-team">Development team</h2><div class="hline"></div><p>IQ-TREE is actively developed by:</p><p><strong>Bui Quang Minh</strong>, <em>Team leader</em>, Designs and implements software core, tree search, ultrafast bootstrap, model selection.</p><p><strong>Robert Lanfear</strong>, <em>Co-leader</em>, Model selection.</p><p><strong>Olga Chernomor</strong>, <em>Developer</em>, Implements partition models.</p><p><strong>Heiko A. Schmidt</strong>, <em>Developer</em>, Integrates TREE-PUZZLE features.</p><p><strong>Dominik Schrempf</strong>, <em>Developer</em>, Implements polymorphism-aware models (PoMo).</p><p><strong>Michael Woodhams</strong>, <em>Developer</em>, Implements Lie Markov models.</p><p><strong>Diep Thi Hoang</strong>, <em>Developer</em>, Improves ultrafast bootstrap.</p><p><strong>Arndt von Haeseler</strong>, <em>Advisor</em>.</p><p>Past members:</p><p><strong>Lam Tung Nguyen</strong>, <em>Developer</em>, Implemented tree search algorithm.</p><p><strong>Jana Trifinopoulos</strong>, <em>Developer</em>, Implemented web service.</p><h2 id="credits-and-acknowledgements">Credits and acknowledgements</h2><div class="hline"></div><p>Some parts of the code were taken from the following packages/libraries: <a href="http://www.libpll.org/">Phylogenetic likelihood library</a>, <a href="http://www.tree-puzzle.de/">TREE-PUZZLE</a>, <a href="https://doi.org/10.1093/oxfordjournals.molbev.a025808">BIONJ</a>, <a href="https://doi.org/10.1093/bioinformatics/btg319">Nexus Class Libary</a>, <a href="http://eigen.tuxfamily.org/">Eigen library</a>, <a href="http://www.sprng.org/">SPRNG library</a>, <a href="http://www.zlib.net/">Zlib library</a>, gzstream library, <a href="http://www.agner.org/optimize/">vectorclass library</a>, <a href="https://www.gnu.org/software/gsl/">GNU scientific library</a>.</p><p>IQ-TREE was funded by the <a href="http://www.fwf.ac.at/">Austrian Science Fund</a> (grant no. I760-B17 from 2012-2015 and I 2508-B29 from 2016-2017), the <a href="https://www.univie.ac.at/">University of Vienna</a> (Initiativkolleg I059-N from 2012-2015), the <a href="https://www.anu.edu.au/">Australian National University</a> (2018-onwards), <a href="https://chanzuckerberg.com/">Chan-Zuckerberg Initiative</a> (2020).</p><div class="spacing"></div></div><div class="col-md-3"><div class="well well-sm"> <a href="http://www.iqtree.org/doc/iqtree-doc.pdf"><i class="fa fa-file-pdf-o fa-lg" aria-hidden="true"></i> Download Tutorials and Manual in one PDF</a></div><div class="well well-sm" id="sidebar"><h4>Documentation</h4><p> <i class="fa fa-info-circle"></i> Introduction</p><ul><li><a href="http://www.iqtree.org/doc/Home#why-iq-tree">Why IQ-TREE?</a></li><li><a href="http://www.iqtree.org/doc/Home#key-features">Key features</a></li><li><a href="http://www.iqtree.org/doc/Home#free-web-server">Free web server</a></li><li><a href="http://www.iqtree.org/doc/Home#user-support">User support</a></li><li><a href="http://www.iqtree.org/doc/Home#documentation">Documentation</a></li><li><a href="http://www.iqtree.org/doc/Home#how-to-cite-iq-tree">How to cite IQ-TREE?</a></li><li><a href="http://www.iqtree.org/doc/Home#development-team">Development team</a></li><li><a href="http://www.iqtree.org/doc/Home#credits-and-acknowledgements">Credits and acknowledgements</a></li></ul><p> <a href="http://www.iqtree.org/doc/Quickstart"> <i class="fa fa-info-circle"></i> Getting Started </a></p><p> <a href="http://www.iqtree.org/doc/Web-Server-Tutorial"> <i class="fa fa-info-circle"></i> Web Server Tutorial </a></p><p> <a href="http://www.iqtree.org/doc/Tutorial"> <i class="fa fa-info-circle"></i> Beginner's Tutorial </a></p><p> <a href="http://www.iqtree.org/doc/Advanced-Tutorial"> <i class="fa fa-info-circle"></i> Advanced Tutorial </a></p><p> <a href="http://www.iqtree.org/doc/Assessing-Phylogenetic-Assumptions"> <i class="fa fa-info-circle"></i> Assessing Phylogenetic Assumptions </a></p><p> <a href="http://www.iqtree.org/doc/Concordance-Factor"> <i class="fa fa-info-circle"></i> Concordance Factor </a></p><p> <a href="http://www.iqtree.org/doc/Dating"> <i class="fa fa-info-circle"></i> Phylogenetic Dating </a></p><p> <a href="http://www.iqtree.org/doc/Rootstrap"> <i class="fa fa-info-circle"></i> Rooting phylogenies </a></p><p> <a href="http://www.iqtree.org/doc/Frequently-Asked-Questions"> <i class="fa fa-question-circle"></i> Frequently Asked Questions </a></p><p> <a href="http://www.iqtree.org/doc/Substitution-Models"> <i class="fa fa-book"></i> Substitution Models </a></p><p> <a href="http://www.iqtree.org/doc/Complex-Models"> <i class="fa fa-book"></i> Complex Models </a></p><p> <a href="http://www.iqtree.org/doc/Polymorphism-Aware-Models"> <i class="fa fa-book"></i> Polymorphism-Aware Models </a></p><p> <a href="http://www.iqtree.org/doc/Command-Reference"> <i class="fa fa-book"></i> Command Reference </a></p><p> <a href="http://www.iqtree.org/doc/Compilation-Guide"> <i class="fa fa-book"></i> Compilation Guide </a></p><p> <a href="http://www.iqtree.org/doc/Developer-Guide"> <i class="fa fa-codepen"></i> Developer Guide </a></p><p> <a href="http://www.iqtree.org/doc/Analyzing-Big-Data"> <i class="fa fa-info-circle"></i> Analyzing Big Data </a></p></div></div></div></div><div id="footerwrap"><div class="container"><div class="row"><div class="col-sm-4"><h4>Software</h4><p> <a href="http://www.iqtree.org/release/">IQ-TREE release notes</a><br> <a href="https://github.com/Cibiv/IQ-TREE">GitHub repository</a><br> <a href="http://www.iqtree.org/about/">Development team</a><br> <a href="http://www.somsubhra.com/github-release-stats/?username=Cibiv&amp;repository=IQ-TREE">Download statistics</a><br></p></div><div class="col-sm-4"><h4>User support</h4><p> <a href="http://www.iqtree.org/doc/Frequently-Asked-Questions">Frequently asked questions</a><br> <a href="http://www.iqtree.org/doc/">User documentation</a><br> <a href="https://groups.google.com/d/forum/iqtree">IQ-TREE Google group</a><br> <a href="http://iqtree.cibiv.univie.ac.at/" target="_blank">IQ-TREE online web service</a><br></p></div><div class="col-sm-4"><h4>Links</h4><p> <a href="https://cs.anu.edu.au/">Research School of Computer Science, ANU</a><br> <a href="http://www.cibiv.at/">Center for Integrative Bioinformatics Vienna</a><br> <a href="http://validator.w3.org/check/referer">Link Checker</a><br></p></div></div></div></div><div class="alert alert-dismissible text-center cookiealert" role="alert"><div class="cookiealert-container"> We use cookies to ensure you get the best experience on our website <button type="button" class="btn btn-primary btn-sm acceptcookies" aria-label="Close"> I agree </button></div></div><script src="./Introduction_files/jquery.min.js.download"></script> <script src="./Introduction_files/bootstrap.min.js.download"></script> <script src="./Introduction_files/retina-1.1.0.js.download"></script> <script src="./Introduction_files/jquery.hoverdir.js.download"></script> <script src="./Introduction_files/jquery.isotope.min.js.download"></script> <script src="./Introduction_files/cookiealert-standalone.js.download"></script> <script> (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){ (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o), m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m) })(window,document,'script','//www.google-analytics.com/analytics.js','ga'); ga('create', 'UA-38693379-1', 'auto'); ga('send', 'pageview'); </script> <script> // Portfolio (function($) { "use strict"; var $container = $('.portfolio'), $items = $container.find('.portfolio-item'), portfolioLayout = 'fitRows'; if( $container.hasClass('portfolio-centered') ) { portfolioLayout = 'masonry'; } $container.isotope({ filter: '*', animationEngine: 'best-available', layoutMode: portfolioLayout, animationOptions: { duration: 750, easing: 'linear', queue: false }, masonry: { } }, refreshWaypoints()); function refreshWaypoints() { setTimeout(function() { }, 1000); } $('nav.portfolio-filter ul a').on('click', function() { var selector = $(this).attr('data-filter'); $container.isotope({ filter: selector }, refreshWaypoints()); $('nav.portfolio-filter ul a').removeClass('active'); $(this).addClass('active'); return false; }); function getColumnNumber() { var winWidth = $(window).width(), columnNumber = 1; if (winWidth > 1200) { columnNumber = 5; } else if (winWidth > 950) { columnNumber = 4; } else if (winWidth > 600) { columnNumber = 3; } else if (winWidth > 400) { columnNumber = 2; } else if (winWidth > 250) { columnNumber = 1; } return columnNumber; } function setColumns() { var winWidth = $(window).width(), columnNumber = getColumnNumber(), itemWidth = Math.floor(winWidth / columnNumber); $container.find('.portfolio-item').each(function() { $(this).css( { width : itemWidth + 'px' }); }); } function setPortfolio() { setColumns(); $container.isotope('reLayout'); } $container.imagesLoaded(function () { setPortfolio(); }); $(window).on('resize', function () { setPortfolio(); }); })(jQuery); </script>
<table cellspacing="0" cellpadding="0" class="gstl_50 gssb_c" style="width: 423px; display: none; top: 94px; position: absolute; left: 710px;"><tbody><tr><td class="gssb_f"></td><td class="gssb_e" style="width: 100%;"></td></tr></tbody></table></body></html>