<!DOCTYPE html>
<!-- saved from url=(0035)http://www.iqtree.org/doc/Rootstrap -->
<html lang="en" class="csstransforms csstransforms3d csstransitions"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1"><meta name="description" content=""><meta name="author" content=""><title>Rooting phylogenies</title><link href="./Rooting phylogenies_files/bootstrap.min.css" rel="stylesheet"><link href="./Rooting phylogenies_files/style.css" rel="stylesheet"><link href="./Rooting phylogenies_files/font-awesome.min.css" rel="stylesheet"><link rel="stylesheet" href="./Rooting phylogenies_files/cookiealert.css"><script async="" src="./Rooting phylogenies_files/async-ads.js.download"></script><script async="" src="./Rooting phylogenies_files/analytics.js.download"></script><script src="./Rooting phylogenies_files/cse_element__en.js.download" type="text/javascript"></script><link type="text/css" rel="stylesheet" href="./Rooting phylogenies_files/default+en.css"><link type="text/css" rel="stylesheet" href="./Rooting phylogenies_files/default.css"><style type="text/css">.gsc-control-cse{font-family:times new roman, arial, serif}.gsc-control-cse .gsc-table-result{font-family:times new roman, arial, serif}.gsc-refinementsGradient{background:linear-gradient(to left,rgba(255,255,255,1),rgba(255,255,255,0))}.gsc-control-cse{border-color:#337AB7;background-color:#337AB7}input.gsc-input,.gsc-input-box,.gsc-input-box-hover,.gsc-input-box-focus{border-color:#FFFFFF}.gsc-search-button-v2,.gsc-search-button-v2:hover,.gsc-search-button-v2:focus{border-color:#4CAE4C;background-color:#5CB85C;background-image:none;filter:none}.gsc-search-button-v2 svg{fill:#FFFFFF}.gsc-tabHeader.gsc-tabhActive,.gsc-refinementHeader.gsc-refinementhActive{color:#CCCCCC;border-color:#CCCCCC;background-color:#FFFFFF}.gsc-tabHeader.gsc-tabhInactive,.gsc-refinementHeader.gsc-refinementhInactive{color:#CCCCCC;border-color:#CCCCCC;background-color:#FFFFFF}.gsc-webResult.gsc-result,.gsc-results .gsc-imageResult{border-color:#F5F5F5;background-color:#F5F5F5}.gsc-webResult.gsc-result:hover{border-color:#F5F5F5;background-color:#F5F5F5}.gs-webResult.gs-result a.gs-title:link,.gs-webResult.gs-result a.gs-title:link b,.gs-imageResult a.gs-title:link,.gs-imageResult a.gs-title:link b{color:#0000CC}.gs-webResult.gs-result a.gs-title:visited,.gs-webResult.gs-result a.gs-title:visited b,.gs-imageResult a.gs-title:visited,.gs-imageResult a.gs-title:visited b{color:#0000CC}.gs-webResult.gs-result a.gs-title:hover,.gs-webResult.gs-result a.gs-title:hover b,.gs-imageResult a.gs-title:hover,.gs-imageResult a.gs-title:hover b{color:#0000CC}.gs-webResult.gs-result a.gs-title:active,.gs-webResult.gs-result a.gs-title:active b,.gs-imageResult a.gs-title:active,.gs-imageResult a.gs-title:active b{color:#0000CC}.gsc-cursor-page{color:#0000CC}a.gsc-trailing-more-results:link{color:#0000CC}.gs-webResult:not(.gs-no-results-result):not(.gs-error-result) .gs-snippet,.gs-fileFormatType{color:#000000}.gs-webResult div.gs-visibleUrl{color:#008000}.gs-webResult div.gs-visibleUrl-short{color:#008000}.gsc-cursor-box{border-color:#F5F5F5}.gsc-results .gsc-cursor-box .gsc-cursor-page{border-color:#CCCCCC;background-color:#FFFFFF;color:#CCCCCC}.gsc-results .gsc-cursor-box .gsc-cursor-current-page{border-color:#CCCCCC;background-color:#FFFFFF;color:#CCCCCC}.gsc-webResult.gsc-result.gsc-promotion{border-color:#336699;background-color:#FFFFFF}.gsc-completion-title{color:#0000CC}.gsc-completion-snippet{color:#000000}.gs-promotion a.gs-title:link,.gs-promotion a.gs-title:link *,.gs-promotion .gs-snippet a:link{color:#0000CC}.gs-promotion a.gs-title:visited,.gs-promotion a.gs-title:visited *,.gs-promotion .gs-snippet a:visited{color:#0000CC}.gs-promotion a.gs-title:hover,.gs-promotion a.gs-title:hover *,.gs-promotion .gs-snippet a:hover{color:#0000CC}.gs-promotion a.gs-title:active,.gs-promotion a.gs-title:active *,.gs-promotion .gs-snippet a:active{color:#0000CC}.gs-promotion .gs-snippet,.gs-promotion .gs-title .gs-promotion-title-right,.gs-promotion .gs-title .gs-promotion-title-right *{color:#000000}.gs-promotion .gs-visibleUrl,.gs-promotion .gs-visibleUrl-short{color:#008000}.gcsc-find-more-on-google{color:#0000CC}.gcsc-find-more-on-google-magnifier{fill:#0000CC}</style><style type="text/css">.gscb_a{display:inline-block;font:27px/13px arial,sans-serif}.gsst_a .gscb_a{color:#a1b9ed;cursor:pointer}.gsst_a:hover .gscb_a,.gsst_a:focus .gscb_a{color:#36c}.gsst_a{display:inline-block}.gsst_a{cursor:pointer;padding:0 4px}.gsst_a:hover{text-decoration:none!important}.gsst_b{font-size:16px;padding:0 2px;position:relative;user-select:none;-webkit-user-select:none;white-space:nowrap}.gsst_e{vertical-align:middle;opacity:0.55;}.gsst_a:hover .gsst_e,.gsst_a:focus .gsst_e{opacity:0.72;}.gsst_a:active .gsst_e{opacity:1;}.gsst_f{background:white;text-align:left}.gsst_g{background-color:white;border:1px solid #ccc;border-top-color:#d9d9d9;box-shadow:0 2px 4px rgba(0,0,0,0.2);-webkit-box-shadow:0 2px 4px rgba(0,0,0,0.2);margin:-1px -3px;padding:0 6px}.gsst_h{background-color:white;height:1px;margin-bottom:-1px;position:relative;top:-1px}.gsib_a{width:100%;padding:4px 6px 0}.gsib_a,.gsib_b{vertical-align:top}.gssb_c{border:0;position:absolute;z-index:989}.gssb_e{border:1px solid #ccc;border-top-color:#d9d9d9;box-shadow:0 2px 4px rgba(0,0,0,0.2);-webkit-box-shadow:0 2px 4px rgba(0,0,0,0.2);cursor:default}.gssb_f{visibility:hidden;white-space:nowrap}.gssb_k{border:0;display:block;position:absolute;top:0;z-index:988}.gsdd_a{border:none!important}.gsq_a{padding:0}.gssb_a{padding:0 7px}.gssb_a,.gssb_a td{white-space:nowrap;overflow:hidden;line-height:22px}#gssb_b{font-size:11px;color:#36c;text-decoration:none}#gssb_b:hover{font-size:11px;color:#36c;text-decoration:underline}.gssb_g{text-align:center;padding:8px 0 7px;position:relative}.gssb_h{font-size:15px;height:28px;margin:0.2em;-webkit-appearance:button}.gssb_i{background:#eee}.gss_ifl{visibility:hidden;padding-left:5px}.gssb_i .gss_ifl{visibility:visible}a.gssb_j{font-size:13px;color:#36c;text-decoration:none;line-height:100%}a.gssb_j:hover{text-decoration:underline}.gssb_l{height:1px;background-color:#e5e5e5}.gssb_m{color:#000;background:#fff}.gssb_a{padding:0 9px}.gsib_a{padding:5px 9px 4px 9px}.gscb_a{line-height:27px}.gssb_e{border:0}.gssb_l{margin:5px 0}input.gsc-input::-webkit-input-placeholder{font-size:14px}input.gsc-input:-moz-placeholder{font-size:14px}input.gsc-input::-moz-placeholder{font-size:14px}input.gsc-input:-ms-input-placeholder{font-size:14px}input.gsc-input:focus::-webkit-input-placeholder{color:transparent}input.gsc-input:focus:-moz-placeholder{color:transparent}input.gsc-input:focus::-moz-placeholder{color:transparent}input.gsc-input:focus:-ms-input-placeholder{color:transparent}.gssb_c .gsc-completion-container{position:static}.gssb_c{z-index:5000}.gsc-completion-container table{background:transparent;font-size:inherit;font-family:inherit}.gssb_c > tbody > tr,.gssb_c > tbody > tr > td,.gssb_d,.gssb_d > tbody > tr,.gssb_d > tbody > tr > td,.gssb_e,.gssb_e > tbody > tr,.gssb_e > tbody > tr > td{padding:0;margin:0;border:0}.gssb_a table,.gssb_a table tr,.gssb_a table tr td{padding:0;margin:0;border:0}</style></head><body style=""><div class="navbar navbar-default navbar-fixed-top" role="navigation"><div class="container"><div class="navbar-header"> <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar"></span> <span class="icon-bar"></span> <span class="icon-bar"></span> </button> <a class="navbar-brand" href="http://www.iqtree.org/">IQ-TREE</a></div><div class="navbar-collapse collapse navbar-right"><ul class="nav navbar-nav"><li><a data-toggle="collapse" data-target=".navbar-collapse" href="http://www.iqtree.org/#download">Download</a></li><li><a href="http://www.iqtree.org/news/">News</a></li><li class="dropdown"> <a class="dropdown-toggle" data-toggle="dropdown" href="http://www.iqtree.org/doc/Rootstrap#">Web server <span class="caret"></span></a><ul class="dropdown-menu"><li><a href="http://iqtree.cibiv.univie.ac.at/" target="_blank">CIBIV, Austria</a></li><li><a href="https://www.hiv.lanl.gov/content/sequence/IQTREE/iqtree.html" target="_blank">Los Alamos Lab, USA</a></li><li><a href="https://www.phylo.org/" target="_blank">CIPRES Gateway, USA</a></li></ul></li><li><a href="http://www.iqtree.org/doc/">Docs</a></li><li><a href="http://www.iqtree.org/workshop/">Workshop</a></li><li><a href="http://www.iqtree.org/about/">About</a></li></ul></div></div></div><div id="blue"><div class="container"><div class="row"><div class="col-sm-6"><h3> Tutorial</h3></div><div class="col-sm-6"> <script type="text/javascript" async="" src="./Rooting phylogenies_files/f.txt"></script><script> (function() { var cx = '003669294240048101214:pace-yz0kou'; var gcse = document.createElement('script'); gcse.type = 'text/javascript'; gcse.async = true; gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') + '//cse.google.com/cse.js?cx=' + cx; var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(gcse, s); })(); </script> <div id="___gcse_0"><div class="gsc-control-cse gsc-control-cse-en"><div class="gsc-control-wrapper-cse" dir="ltr"><form class="gsc-search-box gsc-search-box-tools" accept-charset="utf-8"><table cellspacing="0" cellpadding="0" class="gsc-search-box"><tbody><tr><td class="gsc-input"><div class="gsc-input-box" id="gsc-iw-id1"><table cellspacing="0" cellpadding="0" id="gs_id50" class="gstl_50 gsc-input" style="width: 100%; padding: 0px;"><tbody><tr><td id="gs_tti50" class="gsib_a"><input autocomplete="off" type="text" size="10" class="gsc-input" name="search" title="search" id="gsc-i-id1" dir="ltr" spellcheck="false" style="width: 100%; padding: 0px; border: none; margin: 0px; height: auto; background: url(&quot;https://www.google.com/cse/static/images/1x/en/branding.png&quot;) left center no-repeat rgb(255, 255, 255); outline: none;"></td><td class="gsib_b"><div class="gsst_b" id="gs_st50" dir="ltr"><a class="gsst_a" href="javascript:void(0)" style="display: none;" title="Clear search box" role="button"><span class="gscb_a" id="gs_cb50" aria-hidden="true">×</span></a></div></td></tr></tbody></table></div></td><td class="gsc-search-button"><button class="gsc-search-button gsc-search-button-v2"><svg width="13" height="13" viewBox="0 0 13 13"><title>search</title><path d="m4.8495 7.8226c0.82666 0 1.5262-0.29146 2.0985-0.87438 0.57232-0.58292 0.86378-1.2877 0.87438-2.1144 0.010599-0.82666-0.28086-1.5262-0.87438-2.0985-0.59352-0.57232-1.293-0.86378-2.0985-0.87438-0.8055-0.010599-1.5103 0.28086-2.1144 0.87438-0.60414 0.59352-0.8956 1.293-0.87438 2.0985 0.021197 0.8055 0.31266 1.5103 0.87438 2.1144 0.56172 0.60414 1.2665 0.8956 2.1144 0.87438zm4.4695 0.2115 3.681 3.6819-1.259 1.284-3.6817-3.7 0.0019784-0.69479-0.090043-0.098846c-0.87973 0.76087-1.92 1.1413-3.1207 1.1413-1.3553 0-2.5025-0.46363-3.4417-1.3909s-1.4088-2.0686-1.4088-3.4239c0-1.3553 0.4696-2.4966 1.4088-3.4239 0.9392-0.92727 2.0864-1.3969 3.4417-1.4088 1.3553-0.011889 2.4906 0.45771 3.406 1.4088 0.9154 0.95107 1.379 2.0924 1.3909 3.4239 0 1.2126-0.38043 2.2588-1.1413 3.1385l0.098834 0.090049z"></path></svg></button></td><td class="gsc-clear-button"><div class="gsc-clear-button" title="clear results">&nbsp;</div></td></tr></tbody></table></form><div class="gsc-results-wrapper-overlay"><div class="gsc-results-close-btn" tabindex="0"></div><div class="gsc-positioningWrapper"><div class="gsc-tabsAreaInvisible"><div aria-label="refinement" role="tab" class="gsc-tabHeader gsc-inline-block gsc-tabhActive">Custom Search</div><span class="gs-spacer"> </span></div></div><div class="gsc-positioningWrapper"><div class="gsc-tabsAreaInvisible"></div></div><div class="gsc-above-wrapper-area-invisible"><table cellspacing="0" cellpadding="0" class="gsc-above-wrapper-area-container"><tbody><tr><td class="gsc-result-info-container"><div class="gsc-result-info-invisible"></div></td><td class="gsc-orderby-container"><div class="gsc-orderby-invisible"><div class="gsc-orderby-label gsc-inline-block">Sort by:</div><div class="gsc-option-menu-container gsc-inline-block"><div class="gsc-selected-option-container gsc-inline-block"><div class="gsc-selected-option">Relevance</div><div class="gsc-option-selector"></div></div><div class="gsc-option-menu-invisible"><div class="gsc-option-menu-item gsc-option-menu-item-highlighted"><div class="gsc-option">Relevance</div></div><div class="gsc-option-menu-item"><div class="gsc-option">Date</div></div></div></div></div></td></tr></tbody></table></div><div class="gsc-adBlockInvisible"></div><div class="gsc-wrapper"><div class="gsc-adBlockInvisible"></div><div class="gsc-resultsbox-invisible"><div class="gsc-resultsRoot gsc-tabData gsc-tabdActive"><div><div class="gsc-expansionArea"></div></div></div></div></div></div><div class="gsc-modal-background-image" tabindex="0"></div></div></div></div></div></div></div></div><div class="container mtb"><div class="row"><div class="col-md-9"><p><em>Last update: May 18, 2021, Contributors: Minh Bui, Suha Naser</em></p><h1 id="rooting-phylogenetic-trees">Rooting phylogenetic trees</h1><p>Using time-reversible Markov models is a very common practice in phylogenetic analysis, because they provide high computational efficiency. However, these models infer <em>unrooted</em> trees hence lack the ability to infer the root placement of the estimated phylogeny. In order to compensate for the inability of these models to root the tree, many researchers use external information such as using outgroup taxa or additional assumptions such as molecular-clocks.</p><p>This guide provides the outgroup approach and another rooting approach using <em>non-reversible</em> models (<a href="https://doi.org/10.1101/2020.07.31.230144">Naser-Khdour et al., 2021</a>), which will be useful when an outgroup is lacking. Please make sure that you use IQ-TREE <strong>version 2.1.3</strong> or later for full features below and cite this manuscript:</p><blockquote><p>S. Naser-Khdour, B.Q. Minh, R. Lanfear (2021) Assessing Confidence in Root Placement on Phylogenies: An Empirical Study Using Non-Reversible Models. <a href="https://doi.org/10.1101/2020.07.31.230144">https://doi.org/10.1101/2020.07.31.230144</a></p></blockquote><h2 id="inferring-unrooted-tree-with-outgroup">Inferring unrooted tree with outgroup</h2><p>We first demonstrate the outgroup approach to root the Bovidae family of five sampled species (Yak, Cow, Goat, Sheep and Tibetan antelope) using two outgroup species (Pig and Whale). Please download:</p><ul><li>An <a href="http://www.iqtree.org/doc/data/bovidae_outgroup.phy">input DNA alignment file</a> for these 7 species.</li><li>An <a href="http://www.iqtree.org/doc/data/bovidae.nex">input partition file</a> that defines 52 genes in this alignment. This is a subset of the mammal dataset (<a href="https://doi.org/10.1016/j.dib.2018.04.094">Wu et al., 2018</a>).</li></ul><blockquote><p>Choosing a “good” outgroup is an entire topic on its own. In generally, the outgroup must contain taxa that do not belong to the ingroup but are evolutionarily close enough to the ingroup taxa.</p></blockquote><p>To infer an unrooted tree, run:</p><pre><code>iqtree2 -s bovidae_outgroup.phy -p bovidae.nex -B 1000 -T AUTO --prefix rev_dna_outg
</code></pre><p>that will invoke the ultrafast bootstrap with 1000 replicates (<code>-B 1000</code>), detect the optimal number of threads (<code>-T AUTO</code>) and write all output files with the prefix <code>rev_dna_outg</code>.</p><p>The input alignment contains protein-coding genes. We can ask IQ-TREE to translate the alignment into protein sequences using the standard genetic code (<code>-st NT2AA</code>) and perform an amino-acid analysis on the translated alignment with:</p><pre><code>iqtree2 -s bovidae_outgroup.phy -p bovidae.nex -B 1000 -T AUTO -st NT2AA --prefix rev_aa_outg
</code></pre><p>where setting the prefix to <code>rev_aa_outg</code> avoids file overwriting with the previous run. The resulting tree may now look like (extracted from <code>rev_aa_outg.iqtree</code>):</p><pre><code>NOTE: Tree is UNROOTED although outgroup taxon 'Yak' is drawn at root
Numbers in parentheses are  ultrafast bootstrap support (%)

+--Yak
|
+--Cow
|
|          +--Goat
|       +--| (100)
|       |  +--Sheep
|   +---| (100)
|   |   +---Tibetan_antelope
+---| (100)
    |                      +-------------------------------Wild_pig
    +----------------------| (100)
                           +-------------------Minke_whale
</code></pre><p>You can open <code>rev_aa_outg.treefile</code> in a tree viewer software (e.g. FigTree) and re-root the tree on the branch separating the outgroup (<code>Wild_pig</code> and <code>Minke_whale</code>) from the remaining ingroup to obtain an outgroup-rooted tree.</p><p>Finally, if you want you can also perform a non-partition analysis by removing the option <code>-p</code>.</p><h2 id="inferring-rooted-trees-without-outgroup">Inferring rooted trees without outgroup</h2><p>We will now infer a rooted tree using non-reversible models. Please download:</p><ul><li>An <a href="http://www.iqtree.org/doc/data/bovidae.phy">input DNA alignment file</a> for 5 ingroup species (Yak, Cow, Goat, Sheep and Tibetan antelope). This is a sub-alignment of the alignment above. We can re-use the same partition file.</li></ul><p>To speed up the analysis, we will perform two steps. The first step is the same as the run above to infer an unrooted tree using reversible models:</p><pre><code>iqtree2 -s bovidae.phy -p bovidae.nex -B 1000 -T AUTO --prefix rev_dna
</code></pre><p>This run will also write the best partitioning scheme to <code>rev_dna.best_scheme.nex</code> file. In the second step, we will re-use this best scheme but replace the substitution model with the most general non-reversible DNA model, 12.12 or UNREST (see <a href="http://www.iqtree.org/doc/Substitution-Models#lie-markov-models">this doc</a>) to obtain a rooted tree:</p><pre><code>iqtree2 -s bovidae.phy -p rev_dna.best_scheme.nex --model-joint 12.12 -B 1000 -T AUTO --prefix nonrev_dna
</code></pre><p>The option <code>--model-joint 12.12</code> tells IQ-TREE use a linked substitution model 12.12 across all partitions. This is to avoid potential over-parameterization as this is very parameter-rich model with 12 parameters.</p><p>The resulting tree extracted from .iqtree file might look like this:</p><pre><code>NOTE: Tree is ROOTED at virtual root '__root__'
Numbers in parentheses are  ultrafast bootstrap support (%)

       +---Yak
+------| (72)
|      |                                       +----------Goat
|      |                                   +---| (100)
|      |                                   |   +----------Sheep
|      +-----------------------------------| (95)
|                                          +-------------Tibetan_antelope
|
+**Cow
|
+**__root__
</code></pre><p>(You can better visualize the .treefile in a tree viewer software).</p><p>This run will write an additional tree file <code>nonrev_dna.rootstrap.nex</code> with <em>rootstrap</em> support values (see box below for definition) annotated on every branch of the tree. If you open this file in FigTree it may look like this (click on “Branch Labels” and choose <code>rootstrap</code> for “Display” as shown in the figure):</p><p><img src="./Rooting phylogenies_files/nonrev_dna.png" alt="Rooted tree with rootstrap supports for DNA"></p><p>It shows that the tree might be rooted in the branch leading to <code>Cow</code> with a rootstrap support of 72%, which is rather low. The 2nd best branch separating Cow and Yak from the rest has a rootstrap support of 17.9%. So with this dataset the DNA model cannot reliably tell where the root position is, but at least provides some candidates.</p><blockquote><p><strong>Rootstrap</strong>: To compute rootstrap supports, we conduct a bootstrap analysis to obtain a number of rooted bootstrap trees using non-reversible models. We define the rootstrap support for each branch in the maximum likelihood (ML) tree, as the proportion of rooted bootstrap trees that have the root on that branch. The rootstrap support values are computed for all the branches including external branches. The sum of the rootstrap support values along the tree are always smaller than or equal to one. A sum that is smaller than one can occur when one or more bootstrap replicates are rooted on a branch that does not occur in the ML tree.</p></blockquote><p>We will now try the amino-acid model to see if that helps. We again use <code>-st NT2AA</code> option to conveniently perform this analysis:</p><pre><code># step 1: infer unrooted tree with reversible models
iqtree2 -s bovidae.phy -p bovidae.nex -B 1000 -T AUTO -st NT2AA --prefix rev_aa

# step 2: infer rooted tree with linked non-reversible models
iqtree2 -s bovidae.phy -p rev_aa.best_scheme.nex --model-joint NONREV -B 1000 -T AUTO -st NT2AA --prefix nonrev_aa
</code></pre><p>The option <code>--model-joint NONREV</code> tells IQ-TREE to use the most general amino-acid model NONREV and to link the NONREV model parameters across all partitions: NONREV has 379 parameters and linking them across partitions will avoid over-parameterization. The tree extracted from <code>nonrev_aa.iqtree</code> file now may look like:</p><pre><code>NOTE: Tree is ROOTED at virtual root '__root__'
Numbers in parentheses are  ultrafast bootstrap support (%)

                                 +------------Yak
+--------------------------------| (100)
|                                +----------------Cow
|
|                                   +-------------------Goat
|                          +--------| (100)
|                          |        +---------------------Sheep
+--------------------------| (100)
|                          +-----------------------------Tibetan_antelope
|
+**__root__
</code></pre><p>Interestingly, the amino-acid model suggests a different root position compared with the DNA model. But this position agrees with the outgroup rooting approach. And the tree <code>nonrev_aa.rootstrap.nex</code> with rootstrap supports look like:</p><p><img src="./Rooting phylogenies_files/nonrev_aa.png" alt="Rooted tree with rootstrap supports using amino-acid nonreversible model"></p><p>That means, the branch separating Yak and Cow from the rest receives a very high rootstrap support of 99.9%. Therefore, the amino-acid model seems to have a much higher power to detect the root, compared with the DNA model.</p><h2 id="testing-root-positions">Testing root positions</h2><p>The rootstrap introduced above is one way to measure our confidence in the root placement, but it is not a statistical test. Alternatively, we can apply the <a href="http://www.iqtree.org/doc/Advanced-Tutorial#tree-topology-tests">tree topology tests</a> to compare the log-likelihoods of the trees being rooted on every branch of the ML tree. IQ-TREE v2.1.3 provides a convenient option <code>--root-test</code> that will re-root the tree on every branch and perform the test for you. So you can run:</p><pre><code>iqtree2 -s bovidae.phy -p rev_aa.best_scheme.nex --model-joint NONREV -st NT2AA --root-test -zb 1000 -au -te nonrev_aa.treefile --prefix nonrev_aa_test
</code></pre><p><code>-zb 1000 -au</code> is to perform several tree topology tests including the approximately-unbiased (AU) test for the tree found above (<code>-te nonrev_aa.treefile</code>). This run will write a file <code>nonrev_aa_test.roottest.csv</code> which might look like:</p><pre><code># Test results for rooting positions on every branch
# This file can be read in MS Excel or in R with command:
#    dat=read.csv('nonrev_aa_test.roottest.csv',comment.char='#')
# Columns are comma-separated with following meanings:
#    ID:      Branch ID
#    logL:    Log-likelihood of the tree rooted at this branch
#    deltaL:  logL difference from the maximal logl
#    bp-RELL: bootstrap proportion using RELL method (Kishino et al. 1990)
#    p-KH:    p-value of one sided Kishino-Hasegawa test (1989)
#    p-SH:    p-value of Shimodaira-Hasegawa test (2000)
#    c-ELW:   Expected Likelihood Weight (Strimmer &amp; Rambaut 2002)
#    p-AU:    p-value of approximately unbiased (AU) test (Shimodaira, 2002)
ID,logL,deltaL,bp-RELL,p-KH,p-SH,c-ELW,p-AU
1,-90388.66044,0,0.983,0.96,1,0.9695602131,0.9975595105
8,-90401.6833,13.02286164,0.005,0.04,0.19,0.01262108065,0.00558089101
5,-90401.68371,13.0232665,0.01,0.04,0.19,0.01262245766,0.006374455939
3,-90410.10499,21.44455589,0.002,0.016,0.104,0.002397842346,0.001014359781
2,-90410.1084,21.44796542,0,0.016,0.104,0.002389013519,0.0008999725939
6,-90413.04441,24.38397245,0,0.005,0.059,0.0002047272296,0.0004975092439
7,-90413.04797,24.38753181,0,0.005,0.059,0.0002046654974,0.0005061888325
</code></pre><p>The branches are sorted by log-likelihoods in descending order. The last column (p-AU) shows the p-values of the AU test. The branch ID 1 has an AU p-value of 0.9975595105, whereas all other branches has p-values &lt; 0.01. To associate branch ID you can return to the FigTree window for <code>nonrev_aa.rootstrap.nex</code> file and select “Display” to “id” in the “Branch Labels” tab.</p><p>The conclusion from this analysis: we can reject all rooting positions on branches other than branch ID 1, which agrees with the rootstrap measure.</p><p>TIP: These options <code>--root-test -zb 1000 -au</code> can be combined with the rootstrap run in the previous section to calculate the rootstrap support values and the rooting test p-values in one single analysis.</p><div class="spacing"></div></div><div class="col-md-3"><div class="well well-sm"> <a href="http://www.iqtree.org/doc/iqtree-doc.pdf"><i class="fa fa-file-pdf-o fa-lg" aria-hidden="true"></i> Download Tutorials and Manual in one PDF</a></div><div class="well well-sm" id="sidebar"><h4>Documentation</h4><p> <a href="http://www.iqtree.org/doc/Home"> <i class="fa fa-info-circle"></i> Introduction </a></p><p> <a href="http://www.iqtree.org/doc/Quickstart"> <i class="fa fa-info-circle"></i> Getting Started </a></p><p> <a href="http://www.iqtree.org/doc/Web-Server-Tutorial"> <i class="fa fa-info-circle"></i> Web Server Tutorial </a></p><p> <a href="http://www.iqtree.org/doc/Tutorial"> <i class="fa fa-info-circle"></i> Beginner's Tutorial </a></p><p> <a href="http://www.iqtree.org/doc/Advanced-Tutorial"> <i class="fa fa-info-circle"></i> Advanced Tutorial </a></p><p> <a href="http://www.iqtree.org/doc/Assessing-Phylogenetic-Assumptions"> <i class="fa fa-info-circle"></i> Assessing Phylogenetic Assumptions </a></p><p> <a href="http://www.iqtree.org/doc/Concordance-Factor"> <i class="fa fa-info-circle"></i> Concordance Factor </a></p><p> <a href="http://www.iqtree.org/doc/Dating"> <i class="fa fa-info-circle"></i> Phylogenetic Dating </a></p><p> <i class="fa fa-info-circle"></i> Rooting phylogenies</p><ul><li><a href="http://www.iqtree.org/doc/Rootstrap#inferring-unrooted-tree-with-outgroup">Inferring unrooted tree with outgroup</a></li><li><a href="http://www.iqtree.org/doc/Rootstrap#inferring-rooted-trees-without-outgroup">Inferring rooted trees without outgroup</a></li><li><a href="http://www.iqtree.org/doc/Rootstrap#testing-root-positions">Testing root positions</a></li></ul><p> <a href="http://www.iqtree.org/doc/Frequently-Asked-Questions"> <i class="fa fa-question-circle"></i> Frequently Asked Questions </a></p><p> <a href="http://www.iqtree.org/doc/Substitution-Models"> <i class="fa fa-book"></i> Substitution Models </a></p><p> <a href="http://www.iqtree.org/doc/Complex-Models"> <i class="fa fa-book"></i> Complex Models </a></p><p> <a href="http://www.iqtree.org/doc/Polymorphism-Aware-Models"> <i class="fa fa-book"></i> Polymorphism-Aware Models </a></p><p> <a href="http://www.iqtree.org/doc/Command-Reference"> <i class="fa fa-book"></i> Command Reference </a></p><p> <a href="http://www.iqtree.org/doc/Compilation-Guide"> <i class="fa fa-book"></i> Compilation Guide </a></p><p> <a href="http://www.iqtree.org/doc/Developer-Guide"> <i class="fa fa-codepen"></i> Developer Guide </a></p><p> <a href="http://www.iqtree.org/doc/Analyzing-Big-Data"> <i class="fa fa-info-circle"></i> Analyzing Big Data </a></p></div></div></div></div><div id="footerwrap"><div class="container"><div class="row"><div class="col-sm-4"><h4>Software</h4><p> <a href="http://www.iqtree.org/release/">IQ-TREE release notes</a><br> <a href="https://github.com/Cibiv/IQ-TREE">GitHub repository</a><br> <a href="http://www.iqtree.org/about/">Development team</a><br> <a href="http://www.somsubhra.com/github-release-stats/?username=Cibiv&amp;repository=IQ-TREE">Download statistics</a><br></p></div><div class="col-sm-4"><h4>User support</h4><p> <a href="http://www.iqtree.org/doc/Frequently-Asked-Questions">Frequently asked questions</a><br> <a href="http://www.iqtree.org/doc/">User documentation</a><br> <a href="https://groups.google.com/d/forum/iqtree">IQ-TREE Google group</a><br> <a href="http://iqtree.cibiv.univie.ac.at/" target="_blank">IQ-TREE online web service</a><br></p></div><div class="col-sm-4"><h4>Links</h4><p> <a href="https://cs.anu.edu.au/">Research School of Computer Science, ANU</a><br> <a href="http://www.cibiv.at/">Center for Integrative Bioinformatics Vienna</a><br> <a href="http://validator.w3.org/check/referer">Link Checker</a><br></p></div></div></div></div><div class="alert alert-dismissible text-center cookiealert" role="alert"><div class="cookiealert-container"> We use cookies to ensure you get the best experience on our website <button type="button" class="btn btn-primary btn-sm acceptcookies" aria-label="Close"> I agree </button></div></div><script src="./Rooting phylogenies_files/jquery.min.js.download"></script> <script src="./Rooting phylogenies_files/bootstrap.min.js.download"></script> <script src="./Rooting phylogenies_files/retina-1.1.0.js.download"></script> <script src="./Rooting phylogenies_files/jquery.hoverdir.js.download"></script> <script src="./Rooting phylogenies_files/jquery.isotope.min.js.download"></script> <script src="./Rooting phylogenies_files/cookiealert-standalone.js.download"></script> <script> (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){ (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o), m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m) })(window,document,'script','//www.google-analytics.com/analytics.js','ga'); ga('create', 'UA-38693379-1', 'auto'); ga('send', 'pageview'); </script> <script> // Portfolio (function($) { "use strict"; var $container = $('.portfolio'), $items = $container.find('.portfolio-item'), portfolioLayout = 'fitRows'; if( $container.hasClass('portfolio-centered') ) { portfolioLayout = 'masonry'; } $container.isotope({ filter: '*', animationEngine: 'best-available', layoutMode: portfolioLayout, animationOptions: { duration: 750, easing: 'linear', queue: false }, masonry: { } }, refreshWaypoints()); function refreshWaypoints() { setTimeout(function() { }, 1000); } $('nav.portfolio-filter ul a').on('click', function() { var selector = $(this).attr('data-filter'); $container.isotope({ filter: selector }, refreshWaypoints()); $('nav.portfolio-filter ul a').removeClass('active'); $(this).addClass('active'); return false; }); function getColumnNumber() { var winWidth = $(window).width(), columnNumber = 1; if (winWidth > 1200) { columnNumber = 5; } else if (winWidth > 950) { columnNumber = 4; } else if (winWidth > 600) { columnNumber = 3; } else if (winWidth > 400) { columnNumber = 2; } else if (winWidth > 250) { columnNumber = 1; } return columnNumber; } function setColumns() { var winWidth = $(window).width(), columnNumber = getColumnNumber(), itemWidth = Math.floor(winWidth / columnNumber); $container.find('.portfolio-item').each(function() { $(this).css( { width : itemWidth + 'px' }); }); } function setPortfolio() { setColumns(); $container.isotope('reLayout'); } $container.imagesLoaded(function () { setPortfolio(); }); $(window).on('resize', function () { setPortfolio(); }); })(jQuery); </script>
<table cellspacing="0" cellpadding="0" class="gstl_50 gssb_c" style="width: 423px; display: none; top: 94px; position: absolute; left: 710px;"><tbody><tr><td class="gssb_f"></td><td class="gssb_e" style="width: 100%;"></td></tr></tbody></table></body></html>